diff --git a/include/b64/decode.h b/include/b64/decode.h
index 12b16ea..6d79ac5 100644
--- a/include/b64/decode.h
+++ b/include/b64/decode.h
@@ -22,7 +22,7 @@ namespace base64
 		base64_decodestate _state;
 		int _buffersize;
 
-		decoder(int buffersize_in = BUFFERSIZE)
+		decoder(int buffersize_in = BUFSIZ)
 		: _buffersize(buffersize_in)
 		{}
 
diff --git a/include/b64/encode.h b/include/b64/encode.h
index 5d807d9..b842ea4 100644
--- a/include/b64/encode.h
+++ b/include/b64/encode.h
@@ -22,7 +22,7 @@ namespace base64
 		base64_encodestate _state;
 		int _buffersize;
 
-		encoder(int buffersize_in = BUFFERSIZE)
+		encoder(int buffersize_in = BUFSIZ)
 		: _buffersize(buffersize_in)
 		{}
 
