From 0b7ebea31de1fba984e7ad457b76a178f0bb65a5 Mon Sep 17 00:00:00 2001
From: Batuhan Apaydin <batuhan.apaydin@chainguard.dev>
Date: Mon, 24 Feb 2025 10:37:13 +0300
Subject: [PATCH] fix azure plugin spec

Signed-off-by: Batuhan Apaydin <batuhan.apaydin@chainguard.dev>
---
 fluent-plugin-azurestorage.gemspec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fluent-plugin-azurestorage.gemspec b/fluent-plugin-azurestorage.gemspec
index 560c9ce..8c6ce7b 100644
--- a/fluent-plugin-azurestorage.gemspec
+++ b/fluent-plugin-azurestorage.gemspec
@@ -18,7 +18,7 @@ Gem::Specification.new do |gem|
   gem.require_paths = ['lib']
 
   gem.add_dependency "fluentd", [">= 0.14.0", "< 2"]
-  gem.add_dependency "azure", [">= 0.7.1", "<= 0.7.7"]
+  gem.add_dependency "azure"
   gem.add_dependency "uuidtools", ">= 2.1.5"
   gem.add_development_dependency "rake", ">= 0.9.2"
   gem.add_development_dependency "test-unit", ">= 3.0.8"
-- 
2.39.5 (Apple Git-154)

