--- a/packages/http/src/mocker/generator/JSONSchema.ts
+++ b/packages/http/src/mocker/generator/JSONSchema.ts
@@ -2,6 +2,10 @@ import faker from '@faker-js/faker';
 import { cloneDeep } from 'lodash';
 import { JSONSchema } from '../../types';
 
+interface ExtendedJSONSchema extends JSONSchema {
+  __bundled__?: unknown;
+}
+
 import { JSONSchemaFaker } from 'json-schema-faker';
 import * as sampler from '@stoplight/json-schema-sampler';
 import { Either, toError, tryCatch } from 'fp-ts/Either';
@@ -79,7 +83,8 @@ export function generate(
           if (seed) {
             JSONSchemaFaker.option('random', seedrandom(seed))
           }
-          return sortSchemaAlphabetically(JSONSchemaFaker.generate({ ...cloneDeep(updatedSource), __bundled__: bundle }))
+          const schema: ExtendedJSONSchema = { ...cloneDeep(updatedSource), __bundled__: bundle };
+          return sortSchemaAlphabetically(JSONSchemaFaker.generate(schema))
         },
         toError
       )
