From 9ed26447e4d19b9e4d3752463705076a90e4ceec Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.ledkov@surgut.co.uk>
Date: Thu, 5 Feb 2026 21:57:31 +0000
Subject: [PATCH] Link startup files without package-metadata

---
 csu/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/csu/Makefile b/csu/Makefile
index 88fc77662e..802c2249d2 100644
--- a/csu/Makefile
+++ b/csu/Makefile
@@ -122,7 +122,7 @@ include ../Rules
 subdir_lib: $(extra-objs:%=$(objpfx)%)
 
 define link-relocatable
-$(CC) -nostdlib -nostartfiles -r -o $@ $^
+$(CC) -Wl,--package-metadata= -nostdlib -nostartfiles -r -o $@ $^
 endef
 
 ifndef start-installed-name-rule
-- 
2.51.0

