diff --git a/async-http.gemspec b/async-http.gemspec
index 2a698a2..b786310 100644
--- a/async-http.gemspec
+++ b/async-http.gemspec
@@ -11,7 +11,6 @@ Gem::Specification.new do |spec|
 	spec.license = "MIT"
 	
 	spec.cert_chain  = ["release.cert"]
-	spec.signing_key = File.expand_path("~/.gem/release.pem")
 	
 	spec.homepage = "https://github.com/socketry/async-http"
 	
@@ -28,10 +27,10 @@ Gem::Specification.new do |spec|
 	
 	spec.add_dependency "async", ">= 2.35.1"
 	spec.add_dependency "async-pool", "~> 0.12"
-	spec.add_dependency "io-endpoint", "~> 0.18"
+	spec.add_dependency "io-endpoint", ">= 0.17"
 	spec.add_dependency "io-stream", "~> 0.14"
 	spec.add_dependency "protocol-http", "~> 0.66"
 	spec.add_dependency "protocol-http1", "~> 0.41"
-	spec.add_dependency "protocol-http2", "~> 0.28"
+	spec.add_dependency "protocol-http2", ">= 0.27"
 	spec.add_dependency "protocol-url", "~> 0.2"
 end
