diff --git a/CMakeLists.txt b/CMakeLists.txt
index f92ecb0..19a326f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -49,7 +49,12 @@ install(FILES "${GENERATED_CONFIG_BUILD_PATH}"
   COMPONENT lib
 )
 
+option(BUILD_EXAMPLES "Build examples" ON)
+option(BUILD_UNITTESTS "Build unit tests" ON)
+option(BUILD_DOC "Build documentation" ON)
 
+if(BUILD_DOC)
+	add_subdirectory(docs)
 if(DOC_INSTALL_DIR)
 else()
   if(WIN32)
@@ -72,13 +77,6 @@ install(FILES ${PROJECT_BINARY_DIR}/docs/index.html ${PROJECT_BINARY_DIR}/docs/m
   ${PROJECT_BINARY_DIR}/docs/style.css DESTINATION ${DOC_INSTALL_DIR}
   COMPONENT doc
 )
-
-option(BUILD_EXAMPLES "Build examples" ON)
-option(BUILD_UNITTESTS "Build unit tests" ON)
-option(BUILD_DOC "Build documentation" ON)
-
-if(BUILD_DOC)
-	add_subdirectory(docs)
 endif()
 if(BUILD_EXAMPLES)
 	add_subdirectory(examples)
