diff --git a/Makefile.am b/Makefile.am
index 79d3209a5..ea9ee1ca3 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -340,7 +340,7 @@ if BUILD_SERVER
 	rm -f $(DESTDIR)$(localstatedir)/log/stap-server/log
 endif
 
-SUBDIRS = . java python stapdyn staprun stapbpf doc man po httpd stap-exporter interactive-notebook
+SUBDIRS = . java python stapdyn staprun stapbpf doc man po httpd stap-exporter
 # NB: the gcov target above uses this to enumarate linked binaries' build directories
 DIST_SUBDIRS = testsuite $(SUBDIRS)
 EXTRA_DIST = m4/ChangeLog
diff --git a/Makefile.in b/Makefile.in
index 46505a827..37aa815ee 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -806,7 +806,7 @@ LDADD =
 EXAMPLE_SOURCE_DIR = $(srcdir)/testsuite/systemtap.examples
 EXAMPLE_META_FILES = $(EXAMPLE_SOURCE_DIR)/*/*.meta
 TEST_COV_DIR = coverage
-SUBDIRS = . java python stapdyn staprun stapbpf doc man po httpd stap-exporter interactive-notebook
+SUBDIRS = . java python stapdyn staprun stapbpf doc man po httpd stap-exporter 
 # NB: the gcov target above uses this to enumarate linked binaries' build directories
 DIST_SUBDIRS = testsuite $(SUBDIRS)
 EXTRA_DIST = m4/ChangeLog
diff --git a/configure b/configure
index 64137f32c..fcac489e8 100755
--- a/configure
+++ b/configure
@@ -14766,12 +14766,6 @@ ac_config_files="$ac_config_files java/org/systemtap/byteman/helper/HelperSDT.ja
 
 ac_config_files="$ac_config_files python/Makefile"
 
-ac_config_files="$ac_config_files interactive-notebook/Makefile"
-
-ac_config_files="$ac_config_files interactive-notebook/stap-jupyter-container"
-
-ac_config_files="$ac_config_files interactive-notebook/stap-jupyter-install"
-
 ac_config_files="$ac_config_files staprun/Makefile"
 
 ac_config_files="$ac_config_files stapbpf/Makefile"
@@ -15693,9 +15687,6 @@ do
     "java/stapbm") CONFIG_FILES="$CONFIG_FILES java/stapbm" ;;
     "java/org/systemtap/byteman/helper/HelperSDT.java") CONFIG_FILES="$CONFIG_FILES java/org/systemtap/byteman/helper/HelperSDT.java" ;;
     "python/Makefile") CONFIG_FILES="$CONFIG_FILES python/Makefile" ;;
-    "interactive-notebook/Makefile") CONFIG_FILES="$CONFIG_FILES interactive-notebook/Makefile" ;;
-    "interactive-notebook/stap-jupyter-container") CONFIG_FILES="$CONFIG_FILES interactive-notebook/stap-jupyter-container" ;;
-    "interactive-notebook/stap-jupyter-install") CONFIG_FILES="$CONFIG_FILES interactive-notebook/stap-jupyter-install" ;;
     "staprun/Makefile") CONFIG_FILES="$CONFIG_FILES staprun/Makefile" ;;
     "stapbpf/Makefile") CONFIG_FILES="$CONFIG_FILES stapbpf/Makefile" ;;
     "httpd/Makefile") CONFIG_FILES="$CONFIG_FILES httpd/Makefile" ;;
@@ -16515,8 +16506,6 @@ See \`config.log' for more details" "$LINENO" 5; }
     "run-stap":F) chmod +x run-stap ;;
     "dtrace":F) chmod +x dtrace ;;
     "java/stapbm":F) chmod +x java/stapbm ;;
-    "interactive-notebook/stap-jupyter-container":F) chmod +x interactive-notebook/stap-jupyter-container ;;
-    "interactive-notebook/stap-jupyter-install":F) chmod +x interactive-notebook/stap-jupyter-install ;;
     "staprun/run-staprun":F) chmod +x staprun/run-staprun ;;
     "staprun/guest/stapshd":F) chmod +x staprun/guest/stapshd ;;
     "staprun/guest/stapsh-daemon":F) chmod +x staprun/guest/stapsh-daemon ;;
diff --git a/configure.ac b/configure.ac
index 3a08f4ce0..963e05aab 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1032,9 +1032,6 @@ AC_CONFIG_FILES(java/Makefile)
 AC_CONFIG_FILES([java/stapbm], [chmod +x java/stapbm])
 AC_CONFIG_FILES(java/org/systemtap/byteman/helper/HelperSDT.java)
 AC_CONFIG_FILES(python/Makefile)
-AC_CONFIG_FILES(interactive-notebook/Makefile)
-AC_CONFIG_FILES([interactive-notebook/stap-jupyter-container], [chmod +x interactive-notebook/stap-jupyter-container])
-AC_CONFIG_FILES([interactive-notebook/stap-jupyter-install], [chmod +x interactive-notebook/stap-jupyter-install])
 AC_CONFIG_FILES(staprun/Makefile)
 AC_CONFIG_FILES(stapbpf/Makefile)
 AC_CONFIG_FILES([httpd/Makefile httpd/docker/Makefile])
diff --git a/interactive-notebook/Makefile.am b/interactive-notebook/Makefile.am
index f5be22deb..c8210dc59 100644
--- a/interactive-notebook/Makefile.am
+++ b/interactive-notebook/Makefile.am
@@ -5,28 +5,9 @@ install-exec-local:
 	for f in stap-jupyter-container stap-jupyter-install; \
 		do $(INSTALL_PROGRAM) -D $$f $(DESTDIR)$(bindir)/$$f; done
 
-install-data-local:
-	(cd "$(srcdir)"; for f in *.{ipynb,md,txt,json,py}; \
-		do $(INSTALL_DATA) -D $$f $(DESTDIR)$(pkgdatadir)/interactive-notebook/$$f; done)
-	(cd "$(srcdir)/isystemtap"; for f in *.{py,json}; \
-		do $(INSTALL_DATA) -D $$f $(DESTDIR)$(pkgdatadir)/interactive-notebook/isystemtap/$$f; done)
-	(cd "$(srcdir)/container"; for f in *; \
-		do $(INSTALL_DATA) -D $$f $(DESTDIR)$(pkgdatadir)/interactive-notebook/container/$$f; done)
-	($(INSTALL_DATA) -D -m 755 stap-jupyter-install $(DESTDIR)$(pkgdatadir)/interactive-notebook/container/stap-jupyter-install)
-	(cd "$(srcdir)/codemirror"; \
-		for f in jupyterlab-stap-highlighting/__init__.py lib/plugin.js lib/stap.js package.json; \
-		do $(INSTALL_DATA) -D $$f $(DESTDIR)$(pkgdatadir)/interactive-notebook/codemirror/$$f; done)
-	(cd "$(srcdir)/tests"; for f in *; \
-		do $(INSTALL_DATA) -D $$f $(DESTDIR)$(pkgdatadir)/interactive-notebook/tests/$$f; done)
-	@echo "STAP_VERSION    = '$(VERSION)'"               >  "constants.py"
-	@echo "STAP_PKGDATADIR = '$(pkgdatadir)'"            >> "constants.py"
-	@echo "STAP_PREFIX     = '$(prefix)'"                >> "constants.py"
-	$(INSTALL_DATA) -D constants.py $(DESTDIR)$(pkgdatadir)/interactive-notebook/isystemtap/constants.py
-
 clean-local:
 	rm -f constants.py stap-jupyter-container stap-jupyter-install
 
 uninstall-local:
-	rm -rf $(DESTDIR)$(pkgdatadir)/interactive-notebook
 	rm -f $(DESTDIR)$(bindir)/stap-jupyter-container
 	rm -f $(DESTDIR)$(bindir)/stap-jupyter-install
diff --git a/interactive-notebook/Makefile.in b/interactive-notebook/Makefile.in
index 2d96ec9d7..ad75f085a 100644
--- a/interactive-notebook/Makefile.in
+++ b/interactive-notebook/Makefile.in
@@ -89,7 +89,7 @@ POST_UNINSTALL = :
 build_triplet = @build@
 host_triplet = @host@
 target_triplet = @target@
-subdir = interactive-notebook
+subdir = :
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
 am__aclocal_m4_deps = $(top_srcdir)/m4/ax_check_compile_flag.m4 \
 	$(top_srcdir)/m4/ax_cxx_compile_stdcxx.m4 \
@@ -351,9 +351,7 @@ $(srcdir)/Makefile.in: @MAINTAINER_MODE_TRUE@ $(srcdir)/Makefile.am  $(am__confi
 	      exit 1;; \
 	  esac; \
 	done; \
-	echo ' cd $(top_srcdir) && $(AUTOMAKE) --foreign interactive-notebook/Makefile'; \
 	$(am__cd) $(top_srcdir) && \
-	  $(AUTOMAKE) --foreign interactive-notebook/Makefile
 Makefile: $(srcdir)/Makefile.in $(top_builddir)/config.status
 	@case '$?' in \
 	  *config.status*) \
@@ -501,29 +499,10 @@ install-exec-local:
 	for f in stap-jupyter-container stap-jupyter-install; \
 		do $(INSTALL_PROGRAM) -D $$f $(DESTDIR)$(bindir)/$$f; done
 
-install-data-local:
-	(cd "$(srcdir)"; for f in *.{ipynb,md,txt,json,py}; \
-		do $(INSTALL_DATA) -D $$f $(DESTDIR)$(pkgdatadir)/interactive-notebook/$$f; done)
-	(cd "$(srcdir)/isystemtap"; for f in *.{py,json}; \
-		do $(INSTALL_DATA) -D $$f $(DESTDIR)$(pkgdatadir)/interactive-notebook/isystemtap/$$f; done)
-	(cd "$(srcdir)/container"; for f in *; \
-		do $(INSTALL_DATA) -D $$f $(DESTDIR)$(pkgdatadir)/interactive-notebook/container/$$f; done)
-	($(INSTALL_DATA) -D -m 755 stap-jupyter-install $(DESTDIR)$(pkgdatadir)/interactive-notebook/container/stap-jupyter-install)
-	(cd "$(srcdir)/codemirror"; \
-		for f in jupyterlab-stap-highlighting/__init__.py lib/plugin.js lib/stap.js package.json; \
-		do $(INSTALL_DATA) -D $$f $(DESTDIR)$(pkgdatadir)/interactive-notebook/codemirror/$$f; done)
-	(cd "$(srcdir)/tests"; for f in *; \
-		do $(INSTALL_DATA) -D $$f $(DESTDIR)$(pkgdatadir)/interactive-notebook/tests/$$f; done)
-	@echo "STAP_VERSION    = '$(VERSION)'"               >  "constants.py"
-	@echo "STAP_PKGDATADIR = '$(pkgdatadir)'"            >> "constants.py"
-	@echo "STAP_PREFIX     = '$(prefix)'"                >> "constants.py"
-	$(INSTALL_DATA) -D constants.py $(DESTDIR)$(pkgdatadir)/interactive-notebook/isystemtap/constants.py
-
 clean-local:
 	rm -f constants.py stap-jupyter-container stap-jupyter-install
 
 uninstall-local:
-	rm -rf $(DESTDIR)$(pkgdatadir)/interactive-notebook
 	rm -f $(DESTDIR)$(bindir)/stap-jupyter-container
 	rm -f $(DESTDIR)$(bindir)/stap-jupyter-install
 
diff --git a/interactive-notebook/README.md b/interactive-notebook/README.md
index 0ef7ab6bd..7953df0c2 100644
--- a/interactive-notebook/README.md
+++ b/interactive-notebook/README.md
@@ -31,13 +31,6 @@ Run `stap-jupyter-install`. This will install the ISystemtap kernel, syntax high
 
 ## Development Directory Structure
 ```
-interactive-notebook
-├── codemirror                      // Jupyterlab extension for syntax highlighting
-│   ├── jupyterlab-stap-highlighting
-│   │   └── ...
-│   ├── lib
-│   │   └── ...
-│   └── package.json
 │
 ├── container                       // Container which wraps up the pythonic install elements
 │   ├── Dockerfile
diff --git a/systemtap.spec b/systemtap.spec
index a85358bfd..649b55368 100644
--- a/systemtap.spec
+++ b/systemtap.spec
@@ -140,7 +140,6 @@ Release: 1%{?release_override}%{?dist}
 # systemtap-runtime-virtguest udev rules, init scripts/systemd service, req:-runtime
 # systemtap-runtime-python2 HelperSDT python2 module, req:-runtime
 # systemtap-runtime-python3 HelperSDT python3 module, req:-runtime
-# systemtap-jupyter      /usr/bin/stap-jupyter-* interactive-notebook req:systemtap
 #
 # Typical scenarios:
 #
@@ -1299,7 +1298,6 @@ exit 0
 %{_bindir}/stap-jupyter-install
 %{_mandir}/man1/stap-jupyter.1*
 %dir %{_datadir}/systemtap
-%{_datadir}/systemtap/interactive-notebook
 
 # ------------------------------------------------------------------------
 
