diff --git a/docker-entrypoint.sh b/docker-entrypoint.sh
index 544ddef..dd748e0 100755
--- a/docker-entrypoint.sh
+++ b/docker-entrypoint.sh
@@ -10,10 +10,12 @@ if [[ "$1" != "history" ]] && [[ "$1" != "init" ]] && type "$1" > /dev/null 2>&1
   exec "$@"
 else
   if [[ "$*" == *--defaultsFile* ]] || [[ "$*" == *--defaults-file* ]] || [[ "$*" == *--version* ]]; then
+    # Patched to use LIQUIBASE_HOME env
     ## Just run as-is
-    exec /liquibase/liquibase "$@"
+    exec $LIQUIBASE_HOME/liquibase "$@"
   else
+    # Patched to use LIQUIBASE_HOME env
     ## Include standard defaultsFile
-    exec /liquibase/liquibase "--defaultsFile=/liquibase/liquibase.docker.properties" "$@"
+    exec $LIQUIBASE_HOME/liquibase "--defaultsFile=$LIQUIBASE_HOME/liquibase.docker.properties" "$@"
   fi
 fi
\ No newline at end of file
