Update the loopback binary location to be /usr/bin

diff --git a/plugins/cilium-cni/install-plugin.sh b/plugins/cilium-cni/install-plugin.sh
index 90d2a38184..683705d548 100755
--- a/plugins/cilium-cni/install-plugin.sh
+++ b/plugins/cilium-cni/install-plugin.sh
@@ -19,7 +19,7 @@ if [ ! -f "${CNI_DIR}/bin/loopback" ]; then
 	echo "Installing loopback driver..."
 
 	# Don't fail hard if this fails as it is usually not required
-	cp /cni/loopback "${CNI_DIR}/bin/" || true
+	cp /usr/bin/loopback "${CNI_DIR}/bin/" || true
 fi
 
 echo "Installing ${BIN_NAME} to ${CNI_DIR}/bin/ ..."
