diff --git a/tools/chacl.c b/tools/chacl.c
index 525a7ff..8fff875 100644
--- a/tools/chacl.c
+++ b/tools/chacl.c
@@ -320,7 +320,7 @@ walk_dir(acl_t acl, acl_t dacl, const char *fname)
 {
 	int failed = 0;
 	DIR *dir;
-	struct dirent64 *d;
+	struct dirent *d;
 	char *name;
 
 	if ((dir = opendir(fname)) == NULL) {
@@ -332,7 +332,7 @@ walk_dir(acl_t acl, acl_t dacl, const char *fname)
 		return(0);	/* got a file, not an error */
 	}
 
-	while ((d = readdir64(dir)) != NULL) {
+	while ((d = readdir(dir)) != NULL) {
 		/* skip "." and ".." entries */
 		if (strcmp(d->d_name, ".") == 0 || strcmp(d->d_name, "..") == 0)
 			continue;