From: Dermot Bradley <dermot_bradley@yahoo.com>
Date: Fri, 29 May 2020 19:55:23 +0100
Updated: Mon, 30 May 2022 23:42:39 +0000

correct paths in plugins

--- a/audisp/plugins/remote/au-remote.conf
+++ b/audisp/plugins/remote/au-remote.conf
@@ -5,6 +5,6 @@
 
 active = no
-path = /sbin/audisp-remote
+path = /usr/sbin/audisp-remote
 type = always
 #args =
 format = string
--- a/audisp/plugins/syslog/syslog.conf
+++ b/audisp/plugins/syslog/syslog.conf
@@ -8,6 +8,6 @@
 
 active = no
-path = /sbin/audisp-syslog
+path = /usr/sbin/audisp-syslog
 type = always 
 args = LOG_INFO
 format = string
