--- a/pkg/scheduler/fit.go
+++ b/pkg/scheduler/fit.go
@@ -109,12 +109,12 @@
 		return err
 	}

-	plugins := []framework.FilterPlugin{
-		noderesourcesPlugin.(framework.FilterPlugin),
-		schedulablePlugin.(framework.FilterPlugin),
-		nodeaffinityPlugin.(framework.FilterPlugin),
-		// interpodaffinityPlugin.(framework.FilterPlugin),
-		tainttolerationPlugin.(framework.FilterPlugin),
+	plugins := []schedframework.FilterPlugin{
+		noderesourcesPlugin.(schedframework.FilterPlugin),
+		schedulablePlugin.(schedframework.FilterPlugin),
+		nodeaffinityPlugin.(schedframework.FilterPlugin),
+		// interpodaffinityPlugin.(schedframework.FilterPlugin),
+		tainttolerationPlugin.(schedframework.FilterPlugin),
 	}

 	succeededPods := 0
@@ -128,10 +128,10 @@

 		candidateNodes := convertNodeInfos(usableNodes)
 		skippedPlugins := map[string]struct{}{}
-		var preFilterResult *framework.PreFilterResult
+		var preFilterResult *schedframework.PreFilterResult

 		for _, plugin := range plugins {
-			prefilterPlugin, ok := plugin.(framework.PreFilterPlugin)
+			prefilterPlugin, ok := plugin.(schedframework.PreFilterPlugin)
 			if !ok {
 				continue
 			}
