diff --git a/configure.ac b/configure.ac
index 932b9da..b4f85bb 100644
--- a/configure.ac
+++ b/configure.ac
@@ -80,15 +80,10 @@ AC_CHECK_DECLS([program_invocation_name, program_invocation_short_name],
 # Set these flags *last*, or else the test programs won't compile
 if test x$GCC = xyes ; then
   # Using -ggdb3 makes (some versions of) Redhat's gcc-2.96 dump core
-  if "$CC" --version | grep '^2\.96$' 1>/dev/null 2>&1; then
-    true
-  else
-    CFLAGS="$CFLAGS -ggdb3"
-  fi
   CFLAGS="$CFLAGS -Wall -W \
  -Wmissing-prototypes -Wmissing-declarations -Wstrict-prototypes \
  -Waggregate-return \
- -Wpointer-arith -Wbad-function-cast -Wnested-externs"
+ -Wpointer-arith -Wbad-function-cast -Wnested-externs -fgnu89-inline"
 fi
 
 CPPFLAGS="$CPPFLAGS -I$srcdir"