diff --git a/clients/python/pyproject.toml b/clients/python/pyproject.toml
index 0162beb575..ebdb46303e 100644
--- a/clients/python/pyproject.toml
+++ b/clients/python/pyproject.toml
@@ -52,7 +52,7 @@ classifiers = [

 dependencies = [
     "python-dateutil",
-    "urllib3 >= 1.25.3",
+    "urllib3 >= 1.26.19",
 ]

 [project.urls]
@@ -66,7 +66,7 @@ Homepage = "https://airflow.apache.org/"
 [tool.hatch.envs.test]
 dependencies = [
     "pytest-cov>=2.8.1",
-    "urllib3 >= 1.25.3",
+    "urllib3 >= 1.26.19",
 ]

 [tool.hatch.envs.test.scripts]
