Subject: Fix server.js and node paths for FHS compliance
Upstream: No (custom patch for Wolfi/Debian packaging)
Origin: https://salsa.debian.org/mendezr/stremio
Author: Juan Manuel Méndez Rey <vejeta@gmail.com>

---
diff --git a/main.qml b/main.qml
index 6199cd1..870fcc7 100644
--- a/main.qml
+++ b/main.qml
@@ -278,10 +278,10 @@ ApplicationWindow {
         errorDialog.visible = true
     }
     function launchServer() {
-        var node_executable = applicationDirPath + "/node"
+        var node_executable = "/usr/bin/node"
         if (Qt.platform.os === "windows") node_executable = applicationDirPath + "/stremio-runtime.exe"
         streamingServer.start(node_executable, 
-            [applicationDirPath +"/server.js"].concat(Qt.application.arguments.slice(1)), 
+            ["/usr/share/stremio/server.js"].concat(Qt.application.arguments.slice(1)), 
             "EngineFS server started at "
         )
     }
