diff --git a/update-ms-fonts b/update-ms-fonts
index 89d95c3..8c76af5 100755
--- a/update-ms-fonts
+++ b/update-ms-fonts
@@ -196,7 +196,6 @@ EOF
     for ff in $FONTFILES; do
         for ttf in `grep $ff msfonts.info | awk '{print $4}'`; do
             longname=`awk "/$ttf/ { print \\$2 }" msfonts.info`
-            mv -Z $ttf $FONTDIR/$longname
             ln -sf $longname $FONTDIR/$ttf
         done
     done
