From 64ae7397450876ab95d5713bbd2393630e4928f5 Mon Sep 17 00:00:00 2001
From: Leonardo Arena <rnalrd@gmail.com>
Date: Tue, 4 Jan 2022 12:01:39 +0000
Subject: [PATCH] remove debian stuff

---
 update-ms-fonts | 12 +-----------
 1 file changed, 1 insertion(+), 11 deletions(-)

diff --git a/update-ms-fonts b/update-ms-fonts
index 89d95c3..c726c57 100755
--- a/update-ms-fonts
+++ b/update-ms-fonts
@@ -34,7 +34,7 @@ export http_proxy
 mstt_exit_with_error() {
     echo "$1" >&2
     echo "The fonts are NOT installed." >&2
-    echo "Please run 'dpkg-reconfigure ttf-mscorefonts-installer' to perform the installation again" >&2
+    echo "Please run 'update-ms-fonts' to perform the installation again" >&2
     exit 1
 }
 
@@ -188,11 +188,6 @@ EOF
   # Give sane names. These are nearly the same names MS uses for the
   # Macintosh versions
     
-    mkdir -p /usr/share/doc/ttf-mscorefonts-installer $FONTDIR
-    if [ -e licen.txt ] ; then
-        mv licen.txt '/usr/share/doc/ttf-mscorefonts-installer/READ_ME!'
-        gzip -f -9 '/usr/share/doc/ttf-mscorefonts-installer/READ_ME!'
-    fi
     for ff in $FONTFILES; do
         for ttf in `grep $ff msfonts.info | awk '{print $4}'`; do
             longname=`awk "/$ttf/ { print \\$2 }" msfonts.info`
@@ -201,11 +196,6 @@ EOF
         done
     done
     
-  # Make a note of what we installed so we can uninstall it later
-    awk '{print $2}' msfonts.info > /var/lib/msttcorefonts/ms-fonts
-    awk '{print $4}' msfonts.info >> /var/lib/msttcorefonts/ms-fonts
-
-    dpkg-trigger --no-await /usr/share/fonts
 fi
 
 cd /
-- 
2.34.1
