diff --git a/rakelib/artifacts.rake b/rakelib/artifacts.rake
index c4705b5..ad6b693 100644
--- a/rakelib/artifacts.rake
+++ b/rakelib/artifacts.rake
@@ -196,17 +196,11 @@ namespace "artifact" do
 
   desc "Build all (jdk bundled and not) OSS tar.gz and zip of default logstash plugins with all dependencies"
   task "archives_oss" => ["prepare-oss", "generate_build_metadata"] do
-    #with bundled JDKs
-    @bundles_jdk = true
-    license_details = ['APACHE-LICENSE-2.0', "-oss", oss_exclude_paths]
-    create_archive_pack(license_details, "x86_64", "linux", "windows", "darwin")
-    create_archive_pack(license_details, "arm64", "linux", "darwin")
-
     #without JDK
     @bundles_jdk = false
+    license_details = ['APACHE-LICENSE-2.0', "-oss", oss_exclude_paths]
     safe_system("./gradlew bootstrap") #force the build of Logstash jars
     build_tar(*license_details, platform: '-no-jdk')
-    build_zip(*license_details, platform: '-no-jdk')
   end
 
   desc "Build an RPM of logstash with all dependencies"
