diff --git a/pkgs/sqlite3.47.2/generic/tclsqlite3.c b/pkgs/sqlite3.47.2/generic/tclsqlite3.c
index dd73fba..d4b589c 100644
--- a/pkgs/sqlite3.47.2/generic/tclsqlite3.c
+++ b/pkgs/sqlite3.47.2/generic/tclsqlite3.c
@@ -59,6 +59,7 @@
 # include <stdlib.h>
 # include <string.h>
 # include <assert.h>
+# include <stdint.h>
   typedef unsigned char u8;
 # ifndef SQLITE_PTRSIZE
 #   if defined(__SIZEOF_POINTER__)
