From 9e6c63aef64734ca2ded0489509e352666e3048e Mon Sep 17 00:00:00 2001
From: Ankush Pathak <ankush.pathak@chainguard.dev>
Date: Thu, 20 Nov 2025 20:00:57 +0000
Subject: [PATCH] Bump js-yaml to 4.1.1 to remediate GHSA-mh29-5h37-fv8m

Signed-off-by: Ankush Pathak <ankush.pathak@chainguard.dev>
---
 package.json | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/package.json b/package.json
index 8bd1c71b..5276233e 100644
--- a/package.json
+++ b/package.json
@@ -77,7 +77,7 @@
     "http-proxy": "^1.18.1",
     "httpolyglot": "^0.1.2",
     "i18next": "^25.3.0",
-    "js-yaml": "^4.1.0",
+    "js-yaml": "^4.1.1",
     "limiter": "^2.1.0",
     "pem": "^1.14.8",
     "proxy-agent": "^6.3.1",
-- 
2.51.2

