--- a/ci/build/build-vscode.sh
+++ b/ci/build/build-vscode.sh
@@ -112,7 +112,7 @@ EOF
   # this because we have an NPM package that could be installed on any platform.
   # The correct platform dependencies and scripts will be installed as part of
   # the post-install during `npm install` or when building a standalone release.
-  node --max-old-space-size=16384 --optimize-for-size \
+  node --max-old-space-size=16384 --optimize-for-size \
        ./node_modules/gulp/bin/gulp.js \
        "vscode-reh-web-linux-x64${MINIFY:+-min}"

