--- ./lib/vscode/node_modules/v8flags/index.js
+++ ./lib/vscode/node_modules/v8flags/index.js
@@ -16,7 +16,7 @@
 // This number must be incremented whenever the generated cache file changes.
 var CACHE_VERSION = 2;
 
-var configfile = '.v8flags-' + CACHE_VERSION + '-' + process.versions.v8 + '.' + crypto.createHash('md5').update(user).digest('hex') + '.json';
+var configfile = '.v8flags-' + CACHE_VERSION + '-' + process.versions.v8 + '.' + crypto.createHash('sha256').update(user).digest('hex') + '.json';
 
 var failureMessage = [
   'Unable to cache a config file for v8flags to your home directory',


grep -v -E '^(#|$)' $series | (while read patchfile; do
  echo $patchfile
done)