diff --git a/pypy/module/cpyext/include/pyconfig.h b/pypy/module/cpyext/include/pyconfig.h
index a897524b43..9d25072da8 100644
--- a/pypy/module/cpyext/include/pyconfig.h
+++ b/pypy/module/cpyext/include/pyconfig.h
@@ -29,7 +29,9 @@ extern "C" {
 #ifndef _WIN32
 #define VA_LIST_IS_ARRAY
 #ifndef __APPLE__
-#define HAVE_CLOCK_GETTIME
+#ifndef HAVE_CLOCK_GETTIME
+ #define HAVE_CLOCK_GETTIME
+#endif
 #endif
 #endif
 
