--- /usr/lib/spark/bin/load-spark-env.sh
+++ /usr/lib/spark/bin/load-spark-env.sh
@@ -65,6 +65,6 @@ if [ -z "$SPARK_SCALA_VERSION" ]; then
 fi

 # Append jline option to enable the Beeline process to run in background.
-if [[ ( ! $(ps -o stat= -p $$) =~ "+" ) && ! ( -p /dev/stdin ) ]]; then
+if [[ ( ! $(/usr/bin/ps -o stat= -p $$) =~ "+" ) && ! ( -p /dev/stdin ) ]]; then
   export SPARK_BEELINE_OPTS="$SPARK_BEELINE_OPTS -Djline.terminal=jline.UnsupportedTerminal"
 fi