--- a/build.gradle
+++ b/build.gradle
@@ -554,7 +554,7 @@
         entry 'log4j-over-slf4j'
         entry 'slf4j-api'
       }
-      dependency 'org.postgresql:postgresql:42.7.10'
+      dependency 'org.postgresql:postgresql:42.7.11'
       dependency 'org.reflections:reflections:0.10.2'
       dependency 'org.simpleframework:simple:5.1.6'
       dependency 'org.sonarsource.git.blame:git-files-blame:2.0.2.54'
