diff --git a/build.gradle b/build.gradle
index babe4d57c9f..7b500a58c6b 100644
--- a/build.gradle
+++ b/build.gradle
@@ -430,7 +430,7 @@ subprojects {
       dependency 'org.awaitility:awaitility:4.3.0'
       dependency 'org.apache.commons:commons-collections4:4.5.0'
       dependency 'org.apache.commons:commons-csv:1.14.0'
-      dependency 'org.apache.commons:commons-lang3:3.17.0'
+      dependency 'org.apache.commons:commons-lang3:3.18.0'
       dependency 'org.apache.commons:commons-email2-jakarta:2.0.0-M1'
       dependency 'org.apache.commons:commons-exec:1.5.0'
       dependency 'org.apache.commons:commons-text:1.13.1'
