From c16f0449afb93177b54f31fdf8407a591a1c624b Mon Sep 17 00:00:00 2001
From: Alain Kermis <alain.kermis@sonarsource.com>
Date: Mon, 24 Mar 2025 16:15:09 +0100
Subject: [PATCH] SONAR-24448 Upgrade org.eclipse.jgit:org.eclipse.jgit to
 7.2.0.202503040940-r

---
 build.gradle | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build.gradle b/build.gradle
index 29436b2b76f..bb1442b2e2a 100644
--- a/build.gradle
+++ b/build.gradle
@@ -469,7 +469,7 @@ subprojects {
       }
       dependency "org.elasticsearch.plugin:transport-netty4-client:${elasticSearchClientVersion}"
       dependency 'org.elasticsearch:mocksocket:1.2'
-      dependency 'org.eclipse.jgit:org.eclipse.jgit:7.2.0.202503040940-r'
+      dependency 'org.eclipse.jgit:org.eclipse.jgit:7.2.1.202505142326-r'
       dependency "org.codelibs.elasticsearch.module:analysis-common:7.17.22"
       dependency "org.codelibs.elasticsearch.module:reindex:7.17.22"
       dependency 'org.tmatesoft.svnkit:svnkit:1.10.11') {
