diff --git a/genisoimage/genisoimage.c b/genisoimage/genisoimage.c
index 46f0cb7..d7d38ce 100644
--- a/genisoimage/genisoimage.c
+++ b/genisoimage/genisoimage.c
@@ -47,6 +47,7 @@
 
 #include <mconfig.h>
 #include "genisoimage.h"
+#include "checksum.h"
 #include <errno.h>
 #include <timedefs.h>
 #include <fctldefs.h>
diff --git a/genisoimage/jte.c b/genisoimage/jte.c
index 0dff289..770d6ef 100644
--- a/genisoimage/jte.c
+++ b/genisoimage/jte.c
@@ -11,6 +11,7 @@
 
 #include <mconfig.h>
 #include "genisoimage.h"
+#include "md5.h"
 #include <timedefs.h>
 #include <fctldefs.h>
 #include <zlib.h>
