diff --git a/genisoimage/genisoimage.c b/genisoimage/genisoimage.c
index 46f0cb7..f25be73 100644
--- a/genisoimage/genisoimage.c
+++ b/genisoimage/genisoimage.c
@@ -3404,9 +3404,9 @@ if (check_session == 0)
 	goof += sort_tree(root);
 
 	if (goof) {
-		fprintf(stderr, "ISO9660/Rock Ridge tree sort failed.\n");
+		fprintf(stderr, "%s", "ISO9660/Rock Ridge tree sort failed.\n");
 		if(merge_warn_msg)
-			fprintf(stderr, merge_warn_msg);
+			fprintf(stderr, "%s", merge_warn_msg);
 		exit(1);
 	}
 #ifdef UDF
@@ -3417,9 +3417,9 @@ if (check_session == 0)
 		goof += joliet_sort_tree(root);
 	}
 	if (goof) {
-		fprintf(stderr, "Joliet tree sort failed. The -joliet-long switch may help you.\n");
+		fprintf(stderr, "%s", "Joliet tree sort failed. The -joliet-long switch may help you.\n");
 		if(merge_warn_msg)
-			fprintf(stderr, merge_warn_msg);
+			fprintf(stderr, "%s", merge_warn_msg);
 		exit(1);
 	}
 	/*
diff --git a/genisoimage/tree.c b/genisoimage/tree.c
index 7805888..8412cc3 100644
--- a/genisoimage/tree.c
+++ b/genisoimage/tree.c
@@ -647,7 +647,7 @@ got_valid_name:
      fprintf(stderr, "Unable to sort directory %s\n",
            this_dir->whole_name);
      if(merge_warn_msg)
-        fprintf(stderr, merge_warn_msg);
+        fprintf(stderr, "%s", merge_warn_msg);
      exit(1);
 	}
 	/*
