diff --git a/go.mod b/go.mod
index 91838101891..249cafea474 100644
--- a/go.mod
+++ b/go.mod
@@ -5,7 +5,8 @@ go 1.21
 toolchain go1.23.4
 
 replace (
-	github.com/containerd/containerd => github.com/containerd/containerd v1.6.27 // for compatibilty with docker 20.10.x
+	github.com/containerd/containerd => github.com/containerd/containerd v1.6.38 // for compatibilty with docker 20.10.x
+	github.com/containerd/errdefs => github.com/containerd/errdefs v0.1.0 // Force compatible version for containerd 1.6.38
 	github.com/crewjam/saml => github.com/rancher/saml v0.4.14-rancher3
 	github.com/docker/distribution => github.com/docker/distribution v2.8.2+incompatible // rancher-machine requires a replace is set
 	github.com/docker/docker => github.com/docker/docker v20.10.27+incompatible // rancher-machine requires a replace is set