--- a/Documentation/meson.build
+++ b/Documentation/meson.build
@@ -510,17 +510,22 @@ if want_docs != 'false'
     else
         # asciidoctor not found, install pre compiled documetationx
+        fs = import('fs')
         foreach man_set : man_sets
             section = man_set['section']
             foreach adoc : man_set['sources']
                 if want_docs == 'all' or want_docs == 'man'
-                    man = files(adoc + '.' + section)
-                    install_data(man, install_dir: man_set['dir'])
+                    man_file = meson.current_source_dir() / (adoc + '.' + section)
+                    if fs.is_file(man_file)
+                        install_data(man_file, install_dir: man_set['dir'])
+                    endif
                 endif
                 if want_docs == 'all' or want_docs == 'html'
-                    html = files(adoc + '.html')
-                    install_data(html, install_dir: htmldir)
+                    html_file = meson.current_source_dir() / (adoc + '.html')
+                    if fs.is_file(html_file)
+                        install_data(html_file, install_dir: htmldir)
+                    endif
                 endif
             endforeach
         endforeach
     endif
 endif
