From 7855cbb4dc030ac08bf403503edec6f2f80b86f0 Mon Sep 17 00:00:00 2001
From: Batuhan Apaydin <batuhan.apaydin@chainguard.dev>
Date: Sun, 11 Aug 2024 20:42:16 +0300
Subject: [PATCH] re-add accidently removed flag.Parse

Signed-off-by: Batuhan Apaydin <batuhan.apaydin@chainguard.dev>
---
 cmd/fluent-watcher/fluentbit/main.go | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/cmd/fluent-watcher/fluentbit/main.go b/cmd/fluent-watcher/fluentbit/main.go
index ff5215c..a6d90fa 100644
--- a/cmd/fluent-watcher/fluentbit/main.go
+++ b/cmd/fluent-watcher/fluentbit/main.go
@@ -72,6 +72,8 @@ func main() {
 		level.Warn(logger).Log("--flb-timeout is deprecated. Consider setting the terminationGracePeriod field on the `(Cluster)FluentBit` instance.")
 	}
 
+	flag.Parse()
+
 	// First, launch the fluent-bit process.
 	args := []string{"--enable-hot-reload", "-c", configPath}
 	if externalPluginPath != "" {
-- 
2.39.3 (Apple Git-146)

