diff --git a/build.gradle b/build.gradle
index 266db12355..85aa4946ea 100644
--- a/build.gradle
+++ b/build.gradle
@@ -208,6 +208,7 @@ allprojects {
     // that are unrelated to the project dependencies, we should not change them
     if (name != "zinc") {
       resolutionStrategy {
+        force("org.apache.commons:commons-lang3:" + lang3Version)
         force(
           // be explicit about the javassist dependency version instead of relying on the transitive version
           libs.javassist,
diff --git a/gradle.properties b/gradle.properties
index b8d806cc03..cc9d208322 100644
--- a/gradle.properties
+++ b/gradle.properties
@@ -26,6 +26,7 @@ version=8.2.0-22-ccs
 scalaVersion=2.13.16
 # Adding swaggerVersion in gradle.properties to have a single version in place for swagger
 swaggerVersion=2.2.25
+lang3Version=3.18.0
 task=build
 org.gradle.jvmargs=-Xmx2g -Xss100m -XX\:+UseParallelGC
 org.gradle.parallel=true
