From 9eb52f8de4cf58a25a441a594a0dcf8e2bc01cc8 Mon Sep 17 00:00:00 2001
From: Batuhan Apaydin <batuhan.apaydin@chainguard.dev>
Date: Sun, 17 Nov 2024 21:17:01 +0300
Subject: [PATCH] upgrade netty version to fix CVE

Signed-off-by: Batuhan Apaydin <batuhan.apaydin@chainguard.dev>
---
 gradle/dependencies.gradle | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gradle/dependencies.gradle b/gradle/dependencies.gradle
index 2b0f617..0c37c65 100644
--- a/gradle/dependencies.gradle
+++ b/gradle/dependencies.gradle
@@ -113,7 +113,7 @@ versions += [
   mavenArtifact: "3.9.6",
   metrics: "2.2.0",
   mockito: "5.14.2",
-  netty: "4.1.111.Final",
+  netty: "4.1.115.Final",
   opentelemetryProto: "1.0.0-alpha",
   protobuf: "3.25.5", // a dependency of opentelemetryProto
   pcollections: "4.0.1",
-- 
2.39.3 (Apple Git-146)

