diff --git a/Cargo.toml b/Cargo.toml
index 1234567..abcdefg 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -102,7 +102,7 @@
 tracing-opentelemetry = { version = "0.32", optional = true }
 # Newer versions render ANSI escape codes, causing CLI errors to be ugly
 # See also https://github.com/tokio-rs/tracing/issues/3369
-tracing-subscriber = { version = "=0.3.19", features = ["env-filter"] }
+tracing-subscriber = { version = "=0.3.20", features = ["env-filter"] }
 url = "2.4"
 wait-timeout = "0.2"
 xz2 = "0.1.3"
