diff --git a/gradle/libs.versions.toml b/gradle/libs.versions.toml
--- a/gradle/libs.versions.toml
+++ b/gradle/libs.versions.toml
@@ -36,7 +36,7 @@ json_smart        = "2.5.2"
 # when updating the JNA version, also update the version in buildSrc/build.gradle
 jna               = "5.16.0"

-netty             = "4.2.7.Final"
+netty             = "4.2.9.Final"
 joda              = "2.12.7"
 roaringbitmap     = "1.3.0"

