diff --git a/notifications/build.gradle b/notifications/build.gradle
index 1b02718..b32c3c8 100644
--- a/notifications/build.gradle
+++ b/notifications/build.gradle
@@ -85,6 +85,23 @@ allprojects {
         compileTestKotlin.kotlinOptions.jvmTarget = "21"
         compileKotlin.dependsOn ktlint
     }
+
+    // CVE-2025-67735 GHSA-84h7-rjj3-6jx4 - force netty 4.2.8.Final
+    configurations.all {
+        resolutionStrategy {
+            force 'io.netty:netty-buffer:4.2.8.Final'
+            force 'io.netty:netty-codec:4.2.8.Final'
+            force 'io.netty:netty-codec-http:4.2.8.Final'
+            force 'io.netty:netty-codec-http2:4.2.8.Final'
+            force 'io.netty:netty-common:4.2.8.Final'
+            force 'io.netty:netty-handler:4.2.8.Final'
+            force 'io.netty:netty-resolver:4.2.8.Final'
+            force 'io.netty:netty-transport:4.2.8.Final'
+            force 'io.netty:netty-transport-native-unix-common:4.2.8.Final'
+            force 'io.netty:netty-codec-socks:4.2.8.Final'
+            force 'io.netty:netty-handler-proxy:4.2.8.Final'
+        }
+    }
 }
 
 configurations {
