diff --git a/notifications/build.gradle b/notifications/build.gradle
index 1b02718..b32c3c8 100644
--- a/notifications/build.gradle
+++ b/notifications/build.gradle
@@ -85,6 +85,23 @@ allprojects {
         compileTestKotlin.kotlinOptions.jvmTarget = "21"
         compileKotlin.dependsOn ktlint
     }
+
+    // CVE-2025-67735 GHSA-84h7-rjj3-6jx4 - force netty 4.2.8.Final
+    configurations.all {
+        resolutionStrategy {
+            force 'io.netty:netty-buffer:4.2.8.Final'
+            force 'io.netty:netty-codec:4.2.8.Final'
+            force 'io.netty:netty-codec-http:4.2.8.Final'
+            force 'io.netty:netty-codec-http2:4.2.8.Final'
+            force 'io.netty:netty-common:4.2.8.Final'
+            force 'io.netty:netty-handler:4.2.8.Final'
+            force 'io.netty:netty-resolver:4.2.8.Final'
+            force 'io.netty:netty-transport:4.2.8.Final'
+            force 'io.netty:netty-transport-native-unix-common:4.2.8.Final'
+            force 'io.netty:netty-codec-socks:4.2.8.Final'
+            force 'io.netty:netty-handler-proxy:4.2.8.Final'
+        }
+    }
 }
 
 configurations {
diff --git a/notifications/notifications/build.gradle b/notifications/notifications/build.gradle
index 8accfbf..d25aeec 100644
--- a/notifications/notifications/build.gradle
+++ b/notifications/notifications/build.gradle
@@ -146,15 +146,16 @@ configurations.all {
         force "com.amazonaws:aws-java-sdk-s3:${aws_version}"
         force "com.amazonaws:aws-java-sdk-kms:${aws_version}"
         force "joda-time:joda-time:2.12.7"
-        force "io.netty:netty-codec-http:${versions.netty}"
-        force "io.netty:netty-codec-http2:${versions.netty}"
-        force "io.netty:netty-codec:${versions.netty}"
-        force "io.netty:netty-transport:${versions.netty}"
-        force "io.netty:netty-common:${versions.netty}"
-        force "io.netty:netty-buffer:${versions.netty}"
-        force "io.netty:netty-handler:${versions.netty}"
-        force "io.netty:netty-resolver:${versions.netty}"
-        force "io.netty:netty-transport-native-unix-common:${versions.netty}"
+        // CVE-2025-67735 GHSA-84h7-rjj3-6jx4 - force netty 4.2.8.Final
+        force 'io.netty:netty-codec-http:4.2.8.Final'
+        force 'io.netty:netty-codec-http2:4.2.8.Final'
+        force 'io.netty:netty-codec:4.2.8.Final'
+        force 'io.netty:netty-transport:4.2.8.Final'
+        force 'io.netty:netty-common:4.2.8.Final'
+        force 'io.netty:netty-buffer:4.2.8.Final'
+        force 'io.netty:netty-handler:4.2.8.Final'
+        force 'io.netty:netty-resolver:4.2.8.Final'
+        force 'io.netty:netty-transport-native-unix-common:4.2.8.Final'
         force "org.dafny:DafnyRuntime:4.9.0"
     }
 }
