diff --git a/build.gradle b/build.gradle
index 1ca7f99..60c37a1 100644
--- a/build.gradle
+++ b/build.gradle
@@ -278,7 +278,7 @@ dependencies {
 
     def log4jVersion = "${versions.log4j}"
     def protobufVersion = "${versions.protobuf}"
-    def nettyVersion = "${versions.netty}"
+    def nettyVersion = "4.2.11.Final"
 
     configurations {
         // jarHell reports class name conflicts between securemock and mockito-core
