diff --git a/build.gradle b/build.gradle
index faa971e..6f1376c 100644
--- a/build.gradle
+++ b/build.gradle
@@ -277,7 +277,7 @@ dependencies {
 
     def log4jVersion = "${versions.log4j}"
     def protobufVersion = "${versions.protobuf}"
-    def nettyVersion = "${versions.netty}"
+    def nettyVersion = "4.2.8.Final"
 
     configurations {
         // jarHell reports class name conflicts between securemock and mockito-core
@@ -303,6 +303,16 @@ dependencies {
                 force "org.slf4j:slf4j-api:2.0.0"
                 force "org.apache.bcel:bcel:6.6.0" // This line should be removed once Spotbugs is upgraded to 4.7.4
                 force "com.google.guava:guava:${guavaVersion}"
+                // CVE-2025-67735 GHSA-84h7-rjj3-6jx4 - force netty 4.2.8.Final
+                force "io.netty:netty-buffer:${nettyVersion}"
+                force "io.netty:netty-codec:${nettyVersion}"
+                force "io.netty:netty-codec-http:${nettyVersion}"
+                force "io.netty:netty-codec-http2:${nettyVersion}"
+                force "io.netty:netty-common:${nettyVersion}"
+                force "io.netty:netty-handler:${nettyVersion}"
+                force "io.netty:netty-resolver:${nettyVersion}"
+                force "io.netty:netty-transport:${nettyVersion}"
+                force "io.netty:netty-transport-native-unix-common:${nettyVersion}"
             }
         }
     }
