diff --git a/build.gradle b/build.gradle
index af480ed..9ee3d4e 100644
--- a/build.gradle
+++ b/build.gradle
@@ -486,11 +486,16 @@ configurations {
             force "com.fasterxml.jackson.core:jackson-core:${versions.jackson}"
             force "com.fasterxml.jackson.datatype:jackson-datatype-jdk8:${versions.jackson}"
             force "com.fasterxml.jackson.core:jackson-databind:${versions.jackson_databind}"
-            force "io.netty:netty-buffer:${versions.netty}"
-            force "io.netty:netty-common:${versions.netty}"
-            force "io.netty:netty-handler:${versions.netty}"
-            force "io.netty:netty-transport:${versions.netty}"
-            force "io.netty:netty-transport-native-unix-common:${versions.netty}"
+            // CVE-2025-67735 GHSA-84h7-rjj3-6jx4 - force netty 4.2.8.Final
+            force 'io.netty:netty-buffer:4.2.8.Final'
+            force 'io.netty:netty-codec:4.2.8.Final'
+            force 'io.netty:netty-codec-http:4.2.8.Final'
+            force 'io.netty:netty-codec-http2:4.2.8.Final'
+            force 'io.netty:netty-common:4.2.8.Final'
+            force 'io.netty:netty-handler:4.2.8.Final'
+            force 'io.netty:netty-resolver:4.2.8.Final'
+            force 'io.netty:netty-transport:4.2.8.Final'
+            force 'io.netty:netty-transport-native-unix-common:4.2.8.Final'
             force "com.github.luben:zstd-jni:${versions.zstd}"
             force "org.xerial.snappy:snappy-java:1.1.10.8"
             force "com.google.guava:guava:${guava_version}"
