diff --git a/async-query/build.gradle b/async-query/build.gradle
index 53fdcbe29..30567555d 100644
--- a/async-query/build.gradle
+++ b/async-query/build.gradle
@@ -28,7 +28,7 @@ dependencies {
     implementation group: 'org.json', name: 'json', version: '20231013'
     api group: 'com.amazonaws', name: 'aws-java-sdk-emr', version: "${aws_java_sdk_version}"
     api group: 'com.amazonaws', name: 'aws-java-sdk-emrserverless', version: "${aws_java_sdk_version}"
-    implementation group: 'commons-io', name: 'commons-io', version: '2.14.0'
+    implementation group: 'commons-io', name: 'commons-io', version: '2.17.0'
 
     testImplementation(platform("org.junit:junit-bom:5.9.3"))
 
diff --git a/datasources/build.gradle b/datasources/build.gradle
index 9bd233e1f..334bdaf96 100644
--- a/datasources/build.gradle
+++ b/datasources/build.gradle
@@ -21,7 +21,7 @@ dependencies {
     implementation group: 'org.opensearch', name: 'opensearch', version: "${opensearch_version}"
     implementation group: 'org.opensearch', name: 'opensearch-x-content', version: "${opensearch_version}"
     implementation group: 'org.opensearch', name: 'common-utils', version: "${opensearch_build}"
-    implementation group: 'commons-io', name: 'commons-io', version: '2.14.0'
+    implementation group: 'commons-io', name: 'commons-io', version: '2.17.0'
     // FIXME. upgrade aws-encryption-sdk-java once the bouncycastle dependency update to 1.78.
     implementation ('com.amazonaws:aws-encryption-sdk-java:2.4.1') {
 
diff --git a/spark/build.gradle b/spark/build.gradle
index d9d5c9641..a35b1d190 100644
--- a/spark/build.gradle
+++ b/spark/build.gradle
@@ -21,7 +21,7 @@ dependencies {
     implementation group: 'org.json', name: 'json', version: '20231013'
     api group: 'com.amazonaws', name: 'aws-java-sdk-emr', version: "${aws_java_sdk_version}"
     api group: 'com.amazonaws', name: 'aws-java-sdk-emrserverless', version: "${aws_java_sdk_version}"
-    implementation group: 'commons-io', name: 'commons-io', version: '2.14.0'
+    implementation group: 'commons-io', name: 'commons-io', version: '2.17.0'
 
     testImplementation(platform("org.junit:junit-bom:5.9.3"))
 
