From f49bceaa07161102d6952a0fe4e3c15548c2c1c7 Mon Sep 17 00:00:00 2001
Date: Sun, 7 Dec 2025 23:38:19 +0300
Subject: [PATCH] dont build with stable abi

---
 pyproject.toml | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index ad663a5..d20a8a0 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -66,11 +66,6 @@ sdist.exclude = [
     "third-party/faiss/tutorial"
 ]
 
-[[tool.scikit-build.overrides]]
-if.platform-system = "^(darwin|linux)"  # NOTE: Windows ABI3 results in broken DLLs.
-if.abi-flags = "^$"  # NOTE: Free-threaded Python does not support stable ABI.
-wheel.py-api = "cp310"
-
 [tool.cibuildwheel]
 archs = "native"
 # NOTE: We do not support cp310 on Windows ARM64
-- 
2.50.1 (Apple Git-155)

