diff --git a/faiss/faiss/python/loader.py b/faiss/faiss/python/loader.py
index caef9e551..2e7546776 100644
--- a/faiss/faiss/python/loader.py
+++ b/faiss/faiss/python/loader.py
@@ -7,6 +7,7 @@ import platform
 import subprocess
 import logging
 import os
+import sys
 
 
 def Version(v):
@@ -39,6 +40,9 @@ def supported_instruction_sets():
         import numpy
         if Version(numpy.__version__) >= Version("2.0"):
             return False
+        # numpy.distutils deprecated after Python 3.11
+        if sys.version_info >= (3, 12, 0):
+            return False
         # platform-dependent legacy fallback using numpy.distutils.cpuinfo
         import numpy.distutils.cpuinfo
         return "sve" in numpy.distutils.cpuinfo.cpu.info[0].get('Features', "").split()
