diff --git a/setup.py b/setup.py
index 8fc598f..6445d1b 100644
--- a/setup.py
+++ b/setup.py
@@ -65,7 +65,11 @@ def linux_options(
     swig_opts: List[str],
 ) -> dict:
     """Linux options."""
-    default_link_args = ["-l:libfaiss.a", "-l:libopenblas.a", "-lgfortran"]
+    default_link_args = [
+        "-lfaiss",
+        "-lopenblas",
+        "-lgfortran"
+    ]
     if FAISS_ENABLE_GPU:
         default_link_args += [
             "-lcublas_static",
