diff --git a/setup.py b/setup.py
index 561c8e2..6c0d228 100644
--- a/setup.py
+++ b/setup.py
@@ -64,9 +64,9 @@ def linux_options(
 ) -> dict:
     """Linux options."""
     default_link_args = [
-        "-l:libfaiss.a",
-        "-l:libopenblas.a",
-        "-lgfortran",
+        "-lfaiss",
+        "-lopenblas",
+        "-lgfortran"
     ]
     if FAISS_ENABLE_GPU:
         default_link_args += [
