--- a/runtime/bin/ffi_test/ffi_test_functions_generated.cc
+++ b/runtime/bin/ffi_test/ffi_test_functions_generated.cc
@@ -11,6 +11,7 @@
 #include <sys/types.h>
 
 #include <cmath>
+#include <cstdint>
 #include <iostream>
 #include <limits>
 
--- a/runtime/bin/ffi_test/ffi_test_functions.cc
+++ b/runtime/bin/ffi_test/ffi_test_functions.cc
@@ -13,6 +13,7 @@
 #include <sys/types.h>
 
 #include <cmath>
+#include <cstdint>
 #include <iostream>
 #include <limits>
 
--- a/runtime/bin/ffi_test/ffi_test_dynamic_library.cc
+++ b/runtime/bin/ffi_test/ffi_test_dynamic_library.cc
@@ -2,6 +2,7 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
+#include <cstdint>
 #include <stdlib.h>
 #include <string.h>
 #include <iostream>
--- ./third_party/perfetto/include/perfetto/ext/base/uuid.h.orig
+++ ./third_party/perfetto/include/perfetto/ext/base/uuid.h
@@ -18,6 +18,7 @@
 #define INCLUDE_PERFETTO_EXT_BASE_UUID_H_
 
 #include <array>
+#include <cstdint>
 #include <string>
 
 #include "perfetto/ext/base/optional.h"
