diff --git a/pkg/kube/client_factory.go b/pkg/kube/client_factory.go
index 9d4bd88f3e..7658403312 100644
--- a/pkg/kube/client_factory.go
+++ b/pkg/kube/client_factory.go
@@ -92,7 +92,7 @@ func newClientFactory(clientConfig clientcmd.ClientConfig, diskCache bool) *clie
 		if err != nil {
 			return nil, err
 		}
-		return restmapper.NewShortcutExpander(mapper, discoveryClient), nil
+		return restmapper.NewShortcutExpander(mapper, discoveryClient, func(string) {}), nil
 	})
 	return out
 }
diff --git a/pkg/kube/spdy.go b/pkg/kube/spdy.go
index 40d4276a1c..8770da2203 100644
--- a/pkg/kube/spdy.go
+++ b/pkg/kube/spdy.go
@@ -41,9 +41,15 @@ func roundTripperFor(restConfig *rest.Config) (http.RoundTripper, spdy.Upgrader,
 
 	var upgrader *spdyStream.SpdyRoundTripper
 	if restConfig.Proxy != nil {
-		upgrader = spdyStream.NewRoundTripperWithProxy(tlsConfig, restConfig.Proxy)
+		upgrader, err = spdyStream.NewRoundTripperWithProxy(tlsConfig, restConfig.Proxy)
+		if err != nil {
+			return nil, nil, err
+		}
 	} else {
-		upgrader = spdyStream.NewRoundTripper(tlsConfig)
+		upgrader, err = spdyStream.NewRoundTripper(tlsConfig)
+		if err != nil {
+			return nil, nil, err
+		}
 	}
 	wrapper, err := rest.HTTPWrappersForConfig(restConfig, upgrader)
 	if err != nil {
