diff --git a/scripts/xdg-open.in b/scripts/xdg-open.in
index 630e63e..40cc239 100644
--- a/scripts/xdg-open.in
+++ b/scripts/xdg-open.in
@@ -72,7 +72,7 @@ get_key()
 is_file_url_or_path()
 {
     if echo "$1" | grep -q '^file://' \
-            || ! echo "$1" | egrep -q '^[[:alpha:]+\.\-]+:'; then
+            || ! echo "$1" | grep -E -q '^[[:alpha:]+\.\-]+:'; then
         return 0
     else
         return 1
