--- ./scripts/xdg-screensaver.in.orig
+++ ./scripts/xdg-screensaver.in
@@ -26,18 +26,10 @@
 
 #@xdg-utils-common@
 
-# Check if we can use "mv -T"
-if mv -T ... ... 2>&1 | grep '\.\.\.' > /dev/null ; then
-   # We can securely move files in /tmp with mv -T
-   DEBUG 1 "mv -T available"
-   MV="mv -T"
-   screensaver_file="/tmp/xdg-screensaver-$USER-"`echo $DISPLAY | sed 's/:/-/g'`
-else
-   # No secure moves available, use home dir
-   DEBUG 1 "mv -T not available"
-   MV="mv"
-   screensaver_file="$HOME/.xdg-screensaver-"`echo $(hostname)-$DISPLAY | sed 's/:/-/g'`
-fi
+# No secure moves available, use home dir
+DEBUG 1 "mv -T not available"
+MV="mv"
+screensaver_file="$HOME/.xdg-screensaver-"`echo $(hostname)-$DISPLAY | sed 's/:/-/g'`
 lockfile_command=`which lockfile 2> /dev/null`
 
 lockfile()
