From 50d7c545876bd29e1f434de5f9e766e947b58d3b Mon Sep 17 00:00:00 2001
From: Ariadne Conill <ariadne@dereferenced.org>
Date: Mon, 18 Sep 2023 13:16:30 -0700
Subject: [PATCH 4/5] boring: always enable access to boring.Enabled() function

Signed-off-by: Ariadne Conill <ariadne@dereferenced.org>
---
 src/crypto/boring/boring.go | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/src/crypto/boring/boring.go b/src/crypto/boring/boring.go
index 47618fe3c6..e171df3871 100644
--- a/src/crypto/boring/boring.go
+++ b/src/crypto/boring/boring.go
@@ -2,15 +2,10 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build boringcrypto
-
 // Package boring exposes functions that are only available when building with
 // Go+BoringCrypto. This package is available on all targets as long as the
 // Go+BoringCrypto toolchain is used. Use the Enabled function to determine
 // whether the BoringCrypto core is actually in use.
-//
-// Any time the Go+BoringCrypto toolchain is used, the "boringcrypto" build tag
-// is satisfied, so that applications can tag files that use this package.
 package boring
 
 import boring "crypto/internal/backend"
-- 
2.39.2 (Apple Git-143)

