diff --git a/listeners.go b/listeners.go
index 84a32e45..c3bd1d8b 100644
--- a/listeners.go
+++ b/listeners.go
@@ -474,10 +474,6 @@ func (na NetworkAddress) ListenQUIC(ctx context.Context, portOffset uint, config
 		quicTlsConfig := &tls.Config{GetConfigForClient: sqs.getConfigForClient}
 		earlyLn, err := quic.ListenEarly(h3ln, http3.ConfigureTLSConfig(quicTlsConfig), &quic.Config{
 			Allow0RTT: true,
-			RequireAddressValidation: func(clientAddr net.Addr) bool {
-				// TODO: make tunable?
-				return sqs.getActiveRequests() > 1000
-			},
 		})
 		if err != nil {
 			return nil, err
@@ -513,10 +509,6 @@ func ListenQUIC(ln net.PacketConn, tlsConf *tls.Config, activeRequests *int64) (
 		quicTlsConfig := &tls.Config{GetConfigForClient: sqs.getConfigForClient}
 		earlyLn, err := quic.ListenEarly(ln, http3.ConfigureTLSConfig(quicTlsConfig), &quic.Config{
 			Allow0RTT: true,
-			RequireAddressValidation: func(clientAddr net.Addr) bool {
-				// TODO: make tunable?
-				return sqs.getActiveRequests() > 1000
-			},
 		})
 		if err != nil {
 			return nil, err
