From 0d622c6c7e19d350e964fac6b1e38bdcc59e9f63 Mon Sep 17 00:00:00 2001
From: Batuhan Apaydin <batuhan.apaydin@chainguard.dev>
Date: Sun, 8 Dec 2024 12:27:20 +0300
Subject: [PATCH] bump pprof to 0.14

Signed-off-by: Batuhan Apaydin <batuhan.apaydin@chainguard.dev>
---
 Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Cargo.toml b/Cargo.toml
index de26bcc..55e2cb4 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -110,7 +110,7 @@ derivative = "2.2.0"
 
 [target.'cfg(target_os = "linux")'.dependencies]
 netns-rs = "0.1"
-pprof = { version = "0.13", features = ["protobuf", "protobuf-codec", "criterion"] }
+pprof = { version = "0.14", features = ["protobuf", "protobuf-codec", "criterion"] }
 
 [build-dependencies]
 tonic-build = { version = "0.12", default-features = false, features = ["prost"] }
-- 
2.39.3 (Apple Git-146)

