--- a/pkg/scalers/metrics_api_scaler.go	2026-05-20 17:00:27.714206335 -0400
+++ b/pkg/scalers/metrics_api_scaler.go	2026-05-20 17:00:44.939755975 -0400
@@ -147,7 +147,7 @@
 
 // getValueFromPrometheusResponse uses provided valueLocation to access the numeric value in provided body
 func getValueFromPrometheusResponse(body []byte, valueLocation string) (float64, error) {
-	matchers, err := parser.ParseMetricSelector(valueLocation)
+	matchers, err := parser.NewParser(parser.Options{}).ParseMetricSelector(valueLocation)
 	if err != nil {
 		return 0, err
 	}
