diff --git a/crates/core/Cargo.toml b/crates/core/Cargo.toml
index 96999925..53cb902e 100644
--- a/crates/core/Cargo.toml
+++ b/crates/core/Cargo.toml
@@ -94,7 +94,7 @@ wasm_core = [
   "marzano-util/external_functions_ffi",
   "dep:wasm-bindgen",
 ]
-grit_tracing = ["dep:tracing-opentelemetry"]
+grit_tracing = ["tracing-opentelemetry"]
 language-parsers = ["marzano-language/builtin-parser"]
 grit-parser = ["marzano-language/grit-parser"]
 absolute_filename = []
