diff --git a/crates/cli/Cargo.toml b/crates/cli/Cargo.toml
index 78cbae47..327722d4 100644
--- a/crates/cli/Cargo.toml
+++ b/crates/cli/Cargo.toml
@@ -131,7 +131,7 @@ grit_tracing = [
   "dep:opentelemetry-otlp",
   "dep:opentelemetry",
   "dep:opentelemetry_sdk",
-  "dep:tracing-opentelemetry",
+  "tracing-opentelemetry",
   "dep:tracing-subscriber",
   "dep:tracing-log",
   "marzano-core/grit_tracing",
