Author: n/a
Description: Fixes some unsafe traps.

--- a/exp_trap.c
+++ b/exp_trap.c
@@ -264,6 +264,18 @@
 #if defined(SIGSTOP)
 	traps[SIGSTOP].reserved = TRUE;
 #endif
+#if defined(SIGSEGV)
+	traps[SIGSEGV].reserved = TRUE;
+#endif
+#if defined(SIGBUS)
+	traps[SIGBUS].reserved = TRUE;
+#endif
+#if defined(SIGILL)
+	traps[SIGILL].reserved = TRUE;
+#endif
+#if defined(SIGFPE)
+	traps[SIGFPE].reserved = TRUE;
+#endif
 
 	async_handler = Tcl_AsyncCreate(tophalf,(ClientData)0);
 
