--- a/exp_clib.c
+++ b/exp_clib.c
@@ -1483,8 +1483,8 @@
      char *str;
 {
   if (exp_is_debugging) {
-    fprintf(stderr,str);
-    if (exp_logfile) fprintf(exp_logfile,str);
+    fprintf(stderr,"%s",str);
+    if (exp_logfile) fprintf(exp_logfile,"%s",str);
   }
 }
 
--- a/exp_command.c
+++ b/exp_command.c
@@ -995,7 +995,7 @@
 
 	    exp_slave_control(master,1);
 
-	    sprintf(value,"%d",esPtr->fd_slave);
+	    sprintf(value,"%ld",esPtr->fd_slave);
 	    Tcl_SetVar2(interp,SPAWN_OUT,"slave,fd",value,0);
 	}
 	Tcl_SetObjResult (interp, Tcl_NewIntObj (EXP_NOPID));
