diff --git a/pkg/config/loader.go b/pkg/config/loader.go
index 4d0b5f9a..7b09bb1f 100644
--- a/pkg/config/loader.go
+++ b/pkg/config/loader.go
@@ -90,7 +90,13 @@ func (cfg *Config) Resolve() {
 	// Figure out what address we will tell pods to dial for aggregation
 	if cfg.Aggregation.AdvertiseAddress == "" {
 		if ip := os.Getenv("SONOBUOY_ADVERTISE_IP"); ip != "" {
-			cfg.Aggregation.AdvertiseAddress = fmt.Sprintf("[%v]:%d", ip, cfg.Aggregation.BindPort)
+			if strings.Contains(ip, ":") {
+				// IPv6 address needs brackets
+				cfg.Aggregation.AdvertiseAddress = fmt.Sprintf("[%v]:%d", ip, cfg.Aggregation.BindPort)
+			} else {
+				// IPv4 address should not have brackets
+				cfg.Aggregation.AdvertiseAddress = fmt.Sprintf("%v:%d", ip, cfg.Aggregation.BindPort)
+			}
 		} else {
 			hostname, _ := os.Hostname()
 			if hostname != "" {
