diff --git a/bin/apisix b/bin/apisix
index f4c75fbb..11c780d9 100755
--- a/bin/apisix
+++ b/bin/apisix
@@ -38,11 +38,12 @@ if [[ -e $OR_EXEC && "$OR_VER" -ge 119 ]]; then
     # OpenResty version is >= 1.19, use luajit by default
     ROOT=$(${OR_EXEC} -V 2>&1 | grep prefix | grep -Eo 'prefix=(.*)/nginx\s+--' | grep -Eo '/.*/')
     # find the luajit binary of openresty
-    LUAJIT_BIN="$ROOT"/luajit/bin/luajit
-
+    LUAJIT_BIN="/usr/bin/luajit"
+    LUA_EXTRA_PATH="package.path = '/usr/local/apisix/deps/share/lua/5.1/?/init.lua;' .. package.path"
+ 
     # use the luajit of openresty
-    echo "$LUAJIT_BIN $APISIX_LUA $*"
-    exec $LUAJIT_BIN $APISIX_LUA $*
+    echo "$LUAJIT_BIN -e "$LUA_EXTRA_PATH" $APISIX_LUA $*"
+    exec $LUAJIT_BIN -e "$LUA_EXTRA_PATH" $APISIX_LUA $*
 else
     echo "ERROR: Please check the version of OpenResty and Lua, OpenResty 1.19+ + LuaJIT is required for Apache APISIX."
 fi