From f394e9467b5dd8884f51d02d1ff1cd149bd47e4b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Furkan=20T=C3=BCrkal?= <furkan.turkal@chainguard.dev>
Date: Mon, 18 Dec 2023 23:29:55 +0300
Subject: [PATCH] remove signing key
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Furkan Türkal <furkan.turkal@chainguard.dev>
---
 manticore.gemspec | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/manticore.gemspec b/manticore.gemspec
index dd80365..f76299e 100644
--- a/manticore.gemspec
+++ b/manticore.gemspec
@@ -21,12 +21,6 @@ Gem::Specification.new do |spec|
 
   spec.required_ruby_version = '>= 2.3' # JRuby >= 9.1
 
-  private_key = File.expand_path("~/.gemcert/gem-private_key.pem")
-  if File.exists? private_key
-    spec.signing_key = private_key
-    spec.cert_chain  = ['gem-public_cert.pem']
-  end
-
   spec.add_dependency "openssl_pkcs8_pure"
 
   spec.add_development_dependency "jar-dependencies", "~> 0.4.1"
-- 
2.39.3 (Apple Git-145)

