From af9c3c16d159bacb5f1c0af99f061bc197929575 Mon Sep 17 00:00:00 2001
From: Rushmore Mushambi <rushmore@surrealdb.com>
Date: Thu, 28 Mar 2024 09:57:25 +0200
Subject: [PATCH] Stabilise sql2 and jwks

---
 lib/src/lib.rs | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/lib/src/lib.rs b/lib/src/lib.rs
index 918d4d0bd59..809f29afa9d 100644
--- a/lib/src/lib.rs
+++ b/lib/src/lib.rs
@@ -120,14 +120,6 @@ compile_error!(
 	"`ml2` is currently unstable. You need to enable the `surrealdb_unstable` flag to use it."
 );
 
-#[cfg(all(not(surrealdb_unstable), feature = "jwks"))]
-compile_error!("`jwks` depends on a currently unstable feature, `sql2`. You need to enable the `surrealdb_unstable` flag to use it.");
-
-#[cfg(all(not(surrealdb_unstable), feature = "sql2"))]
-compile_error!(
-	"`sql2` is currently unstable. You need to enable the `surrealdb_unstable` flag to use it."
-);
-
 #[cfg(all(not(surrealdb_unstable), feature = "kv-surrealkv"))]
 compile_error!(
 	"`kv-surrealkv` is currently unstable. You need to enable the `surrealdb_unstable` flag to use it."
