diff --git a/gems/bundled_gems b/gems/bundled_gems
index 806af42076..29ba48cdd4 100644
--- a/gems/bundled_gems
+++ b/gems/bundled_gems
@@ -9,10 +9,8 @@ minitest        5.20.0  https://github.com/minitest/minitest
 power_assert    2.0.3   https://github.com/ruby/power_assert
 rake            13.1.0  https://github.com/ruby/rake
 test-unit       3.6.1   https://github.com/test-unit/test-unit
-rexml           3.3.9   https://github.com/ruby/rexml
 rss             0.3.1   https://github.com/ruby/rss
 net-ftp         0.3.4   https://github.com/ruby/net-ftp
-net-imap        0.4.9.1   https://github.com/ruby/net-imap
 net-pop         0.1.2   https://github.com/ruby/net-pop
 net-smtp        0.4.0.1   https://github.com/ruby/net-smtp
 matrix          0.4.2   https://github.com/ruby/matrix
