diff --git a/pyproject.toml b/pyproject.toml
index 93f11603..4c7bc717 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -59,6 +59,8 @@ Issues = "https://github.com/jupyterhub/kubespawner/issues"
 artifacts = [
     "kubespawner/templates/*"
 ]
+# include is required since the project name doesn't match the folder name
+include = ["kubespawner"]
 
 # black is used for autoformatting Python code
 #
