diff --git a/scripts/build b/scripts/build
index 9acc17a2a2..0172e9aaef 100755
--- a/scripts/build
+++ b/scripts/build
@@ -134,7 +134,7 @@ if [ ! -x ${INSTALLBIN}/cni ]; then
 fi
 
 echo Building k3s
-CGO_ENABLED=1 "${GO}" build $BLDFLAGS -tags "$TAGS" -gcflags="all=${GCFLAGS}" -ldflags "$VERSIONFLAGS $LDFLAGS $STATIC" -o bin/k3s ./cmd/server/main.go
+CGO_ENABLED=1 "${GO}" build $BLDFLAGS -tags "$TAGS" -gcflags="all=${GCFLAGS}" -ldflags "$VERSIONFLAGS $LDFLAGS $STATIC" -o bin/k3s ./cmd/server
 ln -s k3s ./bin/containerd
 ln -s k3s ./bin/crictl
 ln -s k3s ./bin/ctr
