diff --git a/go.mod b/go.mod
index 4598f9d0e1..d9f4786f59 100644
--- a/go.mod
+++ b/go.mod
@@ -26,7 +26,7 @@ replace (
 	github.com/quic-go/quic-go => github.com/quic-go/quic-go v0.38.2
 	github.com/rancher/wrangler => github.com/rancher/wrangler v1.1.1-0.20230818201331-3604a6be798d
 	github.com/ugorji/go => github.com/ugorji/go v1.2.11
-	github.com/xenitab/spegel => github.com/k3s-io/spegel v0.0.18-k3s4
+	github.com/spegel-org/spegel => github.com/k3s-io/spegel v0.0.20-k3s1
 	go.etcd.io/etcd/api/v3 => github.com/k3s-io/etcd/api/v3 v3.5.9-k3s1
 	go.etcd.io/etcd/client/pkg/v3 => github.com/k3s-io/etcd/client/pkg/v3 v3.5.9-k3s1
 	go.etcd.io/etcd/client/v2 => github.com/k3s-io/etcd/client/v2 v2.305.9-k3s1
@@ -151,7 +151,6 @@ require (
 	github.com/stretchr/testify v1.8.4
 	github.com/urfave/cli v1.22.14
 	github.com/vishvananda/netlink v1.2.1-beta.2
-	github.com/xenitab/spegel v1.0.18
 	github.com/yl2chen/cidranger v1.0.2
 	go.etcd.io/etcd/api/v3 v3.5.10
 	go.etcd.io/etcd/client/pkg/v3 v3.5.10
diff --git a/pkg/spegel/bootstrap.go b/pkg/spegel/bootstrap.go
index 479bd9bec1..6d3af649ec 100644
--- a/pkg/spegel/bootstrap.go
+++ b/pkg/spegel/bootstrap.go
@@ -15,7 +15,7 @@ import (
 	"github.com/pkg/errors"
 	"github.com/rancher/wrangler/pkg/merr"
 	"github.com/sirupsen/logrus"
-	"github.com/xenitab/spegel/pkg/routing"
+	"github.com/spegel-org/spegel/pkg/routing"
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
diff --git a/pkg/spegel/spegel.go b/pkg/spegel/spegel.go
index 8c42eea1e1..7d0a6a56cb 100644
--- a/pkg/spegel/spegel.go
+++ b/pkg/spegel/spegel.go
@@ -32,11 +32,11 @@ import (
 	"github.com/libp2p/go-libp2p/p2p/host/peerstore/pstoreds"
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
-	"github.com/xenitab/spegel/pkg/metrics"
-	"github.com/xenitab/spegel/pkg/oci"
-	"github.com/xenitab/spegel/pkg/registry"
-	"github.com/xenitab/spegel/pkg/routing"
-	"github.com/xenitab/spegel/pkg/state"
+	"github.com/spegel-org/spegel/pkg/metrics"
+	"github.com/spegel-org/spegel/pkg/oci"
+	"github.com/spegel-org/spegel/pkg/registry"
+	"github.com/spegel-org/spegel/pkg/routing"
+	"github.com/spegel-org/spegel/pkg/state"
 	"k8s.io/component-base/metrics/legacyregistry"
 )
 
@@ -264,7 +264,7 @@ func (c *Config) peerInfo() http.HandlerFunc {
 		logrus.Debugf("Serving p2p peer addr %s to client at %s", info, client)
 		resp.WriteHeader(http.StatusOK)
 		resp.Header().Set("Content-Type", "text/plain")
-		fmt.Fprintf(resp, "%s/p2p/%s", info.Addrs[0].String(), info.ID.Pretty())
+		fmt.Fprintf(resp, "%s/p2p/%s", info.Addrs[0].String(), info.ID.String())
 	})
 }
 
