diff --git a/concurrent-ruby.gemspec b/concurrent-ruby.gemspec
index 863201b5..8e0d3caf 100644
--- a/concurrent-ruby.gemspec
+++ b/concurrent-ruby.gemspec
@@ -16,7 +16,6 @@ Gem::Specification.new do |s|
                         *Dir['ext/concurrent-ruby/**/*'] & git_files,
                         'Rakefile',
                         'Gemfile',
-                        'lib/concurrent-ruby/concurrent/concurrent_ruby.jar'
   ]
   s.extra_rdoc_files = Dir['README*', 'LICENSE*', 'CHANGELOG*']
   s.require_paths    = ['lib/concurrent-ruby']
