From 94343c11a9508f6da42a870abc77066e6a6d3d0e Mon Sep 17 00:00:00 2001
From: David <12832280+David-Else@users.noreply.github.com>
Date: Thu, 6 Nov 2025 17:18:48 +0000
Subject: [PATCH 1/4] Inject HTML into gotmpl

---
 runtime/queries/gotmpl/injections.scm | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/runtime/queries/gotmpl/injections.scm b/runtime/queries/gotmpl/injections.scm
index 321c90add371..3b3d074c3c58 100644
--- a/runtime/queries/gotmpl/injections.scm
+++ b/runtime/queries/gotmpl/injections.scm
@@ -1,2 +1,5 @@
 ((comment) @injection.content
  (#set! injection.language "comment"))
+((text) @injection.content
+ (#set! injection.language "html")
+ (#set! injection.combined))

From d1cea669a50fc1d1834b7b9b3e77fbf9a7ba96de Mon Sep 17 00:00:00 2001
From: David <12832280+David-Else@users.noreply.github.com>
Date: Tue, 11 Nov 2025 17:43:20 +0000
Subject: [PATCH 2/4] Switch to latest and the original tree-sitter-go-template
 repo and update highlights

---
 languages.toml                        |  2 +-
 runtime/queries/gotmpl/highlights.scm | 51 +++++++++++++++++++++------
 2 files changed, 41 insertions(+), 12 deletions(-)

diff --git a/languages.toml b/languages.toml
index 09b05968a8ea..58b1e813d4a4 100644
--- a/languages.toml
+++ b/languages.toml
@@ -858,7 +858,7 @@ indent = { tab-width = 2, unit = " " }
 
 [[grammar]]
 name = "gotmpl"
-source = { git = "https://github.com/dannylongeuay/tree-sitter-go-template", rev = "395a33e08e69f4155156f0b90138a6c86764c979" }
+source = { git = "https://github.com/ngalaiko/tree-sitter-go-template", rev = "ca26229bafcd3f37698a2496c2a5efa2f07e86bc" }
 
 [[language]]
 name = "gowork"
diff --git a/runtime/queries/gotmpl/highlights.scm b/runtime/queries/gotmpl/highlights.scm
index 2f47f4bdc61a..11928b435db2 100644
--- a/runtime/queries/gotmpl/highlights.scm
+++ b/runtime/queries/gotmpl/highlights.scm
@@ -19,6 +19,7 @@
 ; Operators
 
 "|" @operator
+"=" @operator
 ":=" @operator
 
 ; Builtin functions
@@ -38,16 +39,43 @@
 ")" @punctuation.bracket
 "(" @punctuation.bracket
 
-; Keywords
-
-"else" @keyword
-"if" @keyword
-"range" @keyword
-"with" @keyword
-"end" @keyword
-"template" @keyword
-"define" @keyword
-"block" @keyword
+; Actions
+
+(if_action
+  [
+    "if"
+    "else"
+    "end"
+  ] @keyword.control.conditional)
+
+(range_action
+  [
+    "range"
+    "else"
+    "end"
+  ] @keyword.control.conditional)
+
+(template_action
+  "template" @function.builtin)
+
+(block_action
+  [
+    "block"
+    "end"
+  ] @keyword.directive)
+
+(define_action
+  [
+    "define"
+    "end"
+  ] @keyword.directive)
+
+(with_action
+  [
+    "with"
+    "else"
+    "end"
+  ] @keyword.control.conditional)
 
 ; Literals
 
@@ -61,10 +89,11 @@
 
 [
   (int_literal)
-  (float_literal)
   (imaginary_literal)
 ] @constant.numeric.integer
 
+(float_literal) @constant.numeric.float
+
 [
   (true)
   (false)

From 52455aef918a523ac8c7a09e8ef79783e7d673a6 Mon Sep 17 00:00:00 2001
From: David <12832280+David-Else@users.noreply.github.com>
Date: Tue, 11 Nov 2025 18:28:37 +0000
Subject: [PATCH 3/4] Add locals.scm unmodified

---
 runtime/queries/gotmpl/locals.scm | 12 ++++++++++++
 1 file changed, 12 insertions(+)
 create mode 100644 runtime/queries/gotmpl/locals.scm

diff --git a/runtime/queries/gotmpl/locals.scm b/runtime/queries/gotmpl/locals.scm
new file mode 100644
index 000000000000..528e9fabbb58
--- /dev/null
+++ b/runtime/queries/gotmpl/locals.scm
@@ -0,0 +1,12 @@
+[
+  (if_action)
+  (range_action)
+  (block_action)
+  (with_action)
+  (define_action)
+] @local.scope
+
+(variable_definition
+  variable: (variable) @local.definition.var)
+
+(variable) @local.reference

From 879d24dee955312fb49c5f6c0680073a325601c9 Mon Sep 17 00:00:00 2001
From: Michael Davis <mcarsondavis@gmail.com>
Date: Tue, 11 Nov 2025 20:08:51 -0600
Subject: [PATCH 4/4] Update runtime/queries/gotmpl/locals.scm

---
 runtime/queries/gotmpl/locals.scm | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/runtime/queries/gotmpl/locals.scm b/runtime/queries/gotmpl/locals.scm
index 528e9fabbb58..0112db5e095c 100644
--- a/runtime/queries/gotmpl/locals.scm
+++ b/runtime/queries/gotmpl/locals.scm
@@ -7,6 +7,6 @@
 ] @local.scope
 
 (variable_definition
-  variable: (variable) @local.definition.var)
+  variable: (variable) @local.definition.variable)
 
 (variable) @local.reference
