From 6d168eda275deb23b0c643aecd746af3f4cc9937 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Bla=C5=BE=20Hrastnik?= <blaz@mxxn.io>
Date: Tue, 28 Nov 2023 14:38:15 +0900
Subject: [PATCH] fix CI: tree-sitter-gemini user renamed

---
 languages.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/languages.toml b/languages.toml
index ef4687af8a71..33b27626e720 100644
--- a/languages.toml
+++ b/languages.toml
@@ -2916,7 +2916,7 @@ file-types = ["gmi"]
 
 [[grammar]]
 name = "gemini"
-source = { git = "https://git.sr.ht/~sfr/tree-sitter-gemini", rev = "3cc5e4bdf572d5df4277fc2e54d6299bd59a54b3" }
+source = { git = "https://git.sr.ht/~nbsp/tree-sitter-gemini", rev = "3cc5e4bdf572d5df4277fc2e54d6299bd59a54b3" }
 
 [[language]]
 name = "templ"
