From 90f43eefa87de224783ed4bf567e23e354fff2e4 Mon Sep 17 00:00:00 2001
From: ajayk <ajaykemparaj@gmail.com>
Date: Fri, 9 Feb 2024 13:57:22 -0800
Subject: [PATCH] fix correct sha for istio/envoy

---
 WORKSPACE | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/WORKSPACE b/WORKSPACE
index 7196544007..c077d68947 100644
--- a/WORKSPACE
+++ b/WORKSPACE
@@ -36,7 +36,7 @@ bind(
 # Commit date: 2024-01-18
 ENVOY_SHA = "bd2cc6e71a9416bd87673ffc79b3245492825d97"
 
-ENVOY_SHA256 = "0500062c242838d49d6f6ace522346f7ec1969630658f2090070f72ce5c63c5d"
+ENVOY_SHA256 = "3454c991671313c174df1c3a7985bde6c86f87362f09a9656b9c9ac0e6729404"
 
 ENVOY_ORG = "istio"
 
