diff --git a/chronic_duration.gemspec b/chronic_duration.gemspec
index c94711a..abd5fcc 100644
--- a/chronic_duration.gemspec
+++ b/chronic_duration.gemspec
@@ -19,7 +19,7 @@ Gem::Specification.new do |gem|
   gem.test_files    = gem.files.grep(%r{^(test|spec|features)/})
   gem.require_paths = ["lib"]
 
-  gem.add_runtime_dependency "numerizer", "~> 0.1.1"
+  gem.add_runtime_dependency "numerizer"
 
   gem.add_development_dependency "rake", "~> 10.0.3"
   gem.add_development_dependency "rspec", "~> 2.12.0"
