From 27e74baf97c4669e14b8c690044ab979dc34b2ef Mon Sep 17 00:00:00 2001
From: Petr Fedchenkov <giggsoff@gmail.com>
Date: Tue, 28 Jun 2022 10:54:24 +0300
Subject: [PATCH] Fix type to not hit endless getopt loop

Signed-off-by: Petr Fedchenkov <giggsoff@gmail.com>
---
 dhcping.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dhcping.c b/dhcping.c
index 7eb5ae6..cdce51c 100644
--- a/dhcping.c
+++ b/dhcping.c
@@ -70,7 +70,7 @@ unsigned char serveridentifier[4];
 int maxwait=3;
 
 void doargs(int argc,char **argv) {
-    char ch;
+    int ch;
 
     inform=request=verbose=VERBOSE=quiet=0;
     ci=gi=server="0.0.0.0";
-- 
2.34.1
