diff --git a/go.mod b/go.mod
index b6eff2b66..205bcfab8 100644
--- a/go.mod
+++ b/go.mod
@@ -143,7 +143,6 @@ replace (
 	github.com/chzyer/logex => github.com/chzyer/logex v1.2.1
 
 	// https://www.whitesourcesoftware.com/vulnerability-database/CVE-2020-26160
-	github.com/dgrijalva/jwt-go => github.com/golang-jwt/jwt/v4 v4.4.0
 	github.com/golang-jwt/jwt/v4 => github.com/golang-jwt/jwt/v4 v4.5.0
 
 	// opentelemetry cannot update to 1.25.0 according to the dependencies of google.golang.org/grpc
