--- a/opt/bitnami/scripts/libldapclient.sh
+++ b/opt/bitnami/scripts/libldapclient.sh
@@ -35,33 +35,20 @@
 export LDAP_SEARCH_MAP="${LDAP_SEARCH_MAP:-}"
 
 EOF
-    if [[ "$OS_FLAVOUR" =~ ^debian-.*$ ]]; then
-        cat <<"EOF"
 export LDAP_NSLCD_GROUP="nslcd"
-EOF
-    elif [[ "$OS_FLAVOUR" =~ ^(photon)-.*$ ]]; then
-        cat <<"EOF"
-export LDAP_NSLCD_GROUP="ldap"
-EOF
-    fi
 }
 
 ########################
 # Return LDAP config file path depending on distro
 # Globals:
-#   OS_FLAVOUR
+#   None
 # Arguments:
 #   None
 # Returns:
 #   (String) LDAP config file path
 #########################
 ldap_openldap_config_path() {
-    local openldap_config
-    case "$OS_FLAVOUR" in
-    debian-* | ubuntu-*) openldap_config=/etc/ldap/ldap.conf ;;
-    photon-* | redhatubi-*) openldap_config=/etc/openldap/ldap.conf ;;
-    *) error "Unsupported OS flavor ${OS_FLAVOUR}" && exit 1 ;;
-    esac
+    local openldap_config=/etc/ldap/ldap.conf
     echo "$openldap_config"
 }
 
