From 8808bf6f304a1ecac0f13e51aabcddf864189426 Mon Sep 17 00:00:00 2001
From: Batuhan Apaydin <batuhan.apaydin@chainguard.dev>
Date: Tue, 28 Nov 2023 21:09:34 +0300
Subject: [PATCH] upgrade snakeyaml and jackson-dataformat-cbor deps

Signed-off-by: Batuhan Apaydin <batuhan.apaydin@chainguard.dev>
---
 build.gradle | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/build.gradle b/build.gradle
index 3948c51..9125dad 100644
--- a/build.gradle
+++ b/build.gradle
@@ -92,7 +92,8 @@ dependencies {
     implementation "org.clojure:core.match:0.3.0-alpha5"
     implementation "org.clojure:core.memoize:0.5.9"
     implementation "slingshot:slingshot:0.12.2"
-    implementation "org.yaml:snakeyaml:1.23"
+    implementation "org.yaml:snakeyaml:2.0"
+    implementation "com.fasterxml.jackson.dataformat:jackson-dataformat-cbor:2.16.0"
     implementation "cheshire:cheshire:5.8.0"
     implementation ("selmer:selmer:1.12.27") { // templating lib
         exclude module: 'hiccups'
-- 
2.39.3 (Apple Git-145)

