From f4e055b0665564f048ab052817d87582fbca9c1f Mon Sep 17 00:00:00 2001
From: Philippe Deslauriers <philde@chainguard.dev>
Date: Thu, 14 Mar 2024 16:13:43 -0700
Subject: [PATCH] Fix CVE

Signed-off-by: Philippe Deslauriers <philde@chainguard.dev>
---
 build.gradle | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/build.gradle b/build.gradle
index 3948c51..824b28b 100644
--- a/build.gradle
+++ b/build.gradle
@@ -88,11 +88,12 @@ configurations {
 }

 dependencies {
-    implementation "org.clojure:clojure:1.10.1"
+    implementation "org.clojure:clojure:1.11.2"
     implementation "org.clojure:core.match:0.3.0-alpha5"
     implementation "org.clojure:core.memoize:0.5.9"
     implementation "slingshot:slingshot:0.12.2"
-    implementation "org.yaml:snakeyaml:1.23"
+    implementation "org.yaml:snakeyaml:2.0"
+    implementation "com.fasterxml.jackson.dataformat:jackson-dataformat-cbor:2.16.0"
     implementation "cheshire:cheshire:5.8.0"
     implementation ("selmer:selmer:1.12.27") { // templating lib
         exclude module: 'hiccups'
--
2.43.0

