diff --git a/datadog_checks_base/pyproject.toml b/datadog_checks_base/pyproject.toml
index 8ff7f6b..860bb68 100644
--- a/datadog_checks_base/pyproject.toml
+++ b/datadog_checks_base/pyproject.toml
@@ -36,7 +36,7 @@ db = [
 deps = [
     "binary==1.0.1",
     "cachetools==5.5.0",
-    "cryptography==43.0.1",
+    "cryptography==44.0.1",
     "ddtrace==2.10.6",
     "importlib-metadata==2.1.3; python_version < '3.8'",
     "jellyfish==1.1.0",
@@ -58,7 +58,7 @@ http = [
     "botocore==1.35.10",
     "oauthlib==3.2.2",
     "pyjwt==2.9.0",
-    "pyopenssl==24.2.1",
+    "pyopenssl==24.3.0",
     "pysocks==1.7.1",
     "requests-kerberos==0.15.0",
     "requests-ntlm==1.3.0",
