diff --git a/agent_requirements.in b/agent_requirements.in
index 859d088..11f529c 100644
--- a/agent_requirements.in
+++ b/agent_requirements.in
@@ -65,7 +65,7 @@ securesystemslib[crypto,pynacl]==0.28.0
 semver==3.0.2
 service-identity[idna]==24.1.0
 simplejson==3.19.3
-snowflake-connector-python==3.12.1
+snowflake-connector-python==3.12.3; python_version > '3.0'
 supervisor==4.2.5
 tuf==4.0.0
 uptime==3.0.1