diff --git a/datadog_checks_base/pyproject.toml b/datadog_checks_base/pyproject.toml
index cc9875d566..d30ea68a8d 100644
--- a/datadog_checks_base/pyproject.toml
+++ b/datadog_checks_base/pyproject.toml
@@ -36,7 +36,7 @@ db = [
 deps = [
     "binary==1.0.1",
     "cachetools==5.5.1",
-    "cryptography==43.0.1",
+    "cryptography==44.0.1",
     "ddtrace==2.10.6",
     "jellyfish==1.1.3",
     "prometheus-client==0.21.1",
@@ -57,7 +57,7 @@ http = [
     "botocore==1.36.16",
     "oauthlib==3.2.2",
     "pyjwt==2.10.1",
-    "pyopenssl==24.2.1",
+    "pyopenssl==24.3.0",
     "pysocks==1.7.1",
     "requests-kerberos==0.15.0",
     "requests-ntlm==1.3.0",
