diff --git a/cmd/agent/subcommands/integrations/command.go b/cmd/agent/subcommands/integrations/command.go
index 9bcf4b7dc6..e54157e49c 100644
--- a/cmd/agent/subcommands/integrations/command.go
+++ b/cmd/agent/subcommands/integrations/command.go
@@ -184,6 +184,8 @@ You must specify a version of the package to install using the syntax: <package>
 func loadPythonInfo(config config.Component, cliParams *cliParams) error {
 	rootDir, _ = executable.Folder()
 	for {
+		fmt.Printf("Looking for requirements-agent-release.txt in %s\n", filepath.Join(rootDir, reqAgentReleaseFile))
+
 		agentReleaseFile := filepath.Join(rootDir, reqAgentReleaseFile)
 		if _, err := os.Lstat(agentReleaseFile); err == nil {
 			reqAgentReleasePath = agentReleaseFile
@@ -284,7 +286,7 @@ func getCommandPython(pythonMajorVersion string, useSysPython bool) (string, err
 
 	if _, err := os.Stat(pyPath); err != nil {
 		if os.IsNotExist(err) {
-			return pyPath, errors.New("unable to find python executable")
+			return pyPath, errors.New(fmt.Sprintf("unable to find python executable in %s", pyPath))
 		}
 	}
 
