--- a/tasks/system_probe.py
+++ b/tasks/system_probe.py
@@ -1389,7 +1389,7 @@ def build_object_files(

         java_dir = os.path.join("pkg", "network", "protocols", "tls", "java")
         ctx.run(f"{sudo} mkdir -p {EMBEDDED_SHARE_JAVA_DIR}")
-        ctx.run(f"{sudo} install -m644 -oroot -groot {java_dir}/agent-usm.jar {EMBEDDED_SHARE_JAVA_DIR}/agent-usm.jar")
+        ctx.run(f"{sudo} install -m644 {java_dir}/agent-usm.jar {EMBEDDED_SHARE_JAVA_DIR}/agent-usm.jar")

         if ctx.run("command -v rsync >/dev/null 2>&1", warn=True, hide=True).ok:
             rsync_filter = "--filter='+ */' --filter='+ *.o' --filter='+ *.c' --filter='- *'"
@@ -1491,7 +1491,7 @@ def generate_lookup_tables(ctx):


 def is_root():
-    return os.getuid() == 0
+    return True


 def check_for_ninja(ctx):
