From 47816251c4877230711c8d8afb6e929828ad7c37 Mon Sep 17 00:00:00 2001
From: Ariadne Conill <ariadne@dereferenced.org>
Date: Tue, 15 Aug 2023 09:52:05 -0700
Subject: [PATCH 2/3] add zetasql hackfix

Signed-off-by: Ariadne Conill <ariadne@dereferenced.org>
---
 WORKSPACE | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/WORKSPACE b/WORKSPACE
index 303f0ca..2b6c712 100644
--- a/WORKSPACE
+++ b/WORKSPACE
@@ -249,13 +249,13 @@ http_archive(
 )
 
 # BEGIN IFNDEF_WIN
-ZETASQL_COMMIT = "ac37cf5c0d80b5605176fc0f29e87b12f00be693" # 08/10/2022  # windows
+ZETASQL_COMMIT = "c9fc70e7b4b144eb59e0e5f138353592eb8b9199" # 08/10/2022  # windows
 http_archive(  # windows
     name = "com_google_zetasql",  # windows
-    urls = ["https://github.com/google/zetasql/archive/%s.zip" % ZETASQL_COMMIT],  # windows
+    urls = ["https://github.com/kaniini/zetasql/archive/%s.zip" % ZETASQL_COMMIT],  # windows
     strip_prefix = "zetasql-%s" % ZETASQL_COMMIT,  # windows
     #patches = ["//ml_metadata/third_party:zetasql.patch"],  # windows
-    sha256 = '651a768cd51627f58aa6de7039aba9ddab22f4b0450521169800555269447840'  # windows
+    sha256 = 'a9727483ee40d0bf94e36b5a275a890a69ec4c77dc0d47bfcc8d3a20b23d312d'  # windows
 )  # windows
 
 load("@com_google_zetasql//bazel:zetasql_deps_step_1.bzl", "zetasql_deps_step_1")  # windows
-- 
2.39.2 (Apple Git-143)

