From a4ec3831025538d8cacd0b4f81f8edca80631b0a Mon Sep 17 00:00:00 2001
From: Ariadne Conill <ariadne@dereferenced.org>
Date: Tue, 15 Aug 2023 10:15:58 -0700
Subject: [PATCH 3/3] postgresql BUILD: disable x86-64 specific optimizations

Signed-off-by: Ariadne Conill <ariadne@dereferenced.org>
---
 ml_metadata/postgresql.BUILD | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/ml_metadata/postgresql.BUILD b/ml_metadata/postgresql.BUILD
index d6a792d..8f525f4 100644
--- a/ml_metadata/postgresql.BUILD
+++ b/ml_metadata/postgresql.BUILD
@@ -1144,7 +1144,7 @@ genrule(
             "/* #undef HAVE_X509_GET_SIGNATURE_NID */",
             "",
             "/* Define to 1 if the assembler supports X86_64's POPCNTQ instruction. */",
-            "#define HAVE_X86_64_POPCNTQ 1",
+            "/* #undef HAVE_X86_64_POPCNTQ */",
             "",
             "/* Define to 1 if the system has the type `_Bool'. */",
             "#define HAVE__BOOL 1",
@@ -1186,7 +1186,7 @@ genrule(
             "/* #undef HAVE__CPUID */",
             "",
             "/* Define to 1 if you have __get_cpuid. */",
-            "#define HAVE__GET_CPUID 1",
+            "/* #undef HAVE__GET_CPUID */",
             "",
             "/* Define to 1 if your compiler understands _Static_assert. */",
             "#define HAVE__STATIC_ASSERT 1",
-- 
2.39.2 (Apple Git-143)

