--- a/bazel/zetasql_deps_step_1.bzl
+++ b/bazel/zetasql_deps_step_1.bzl
@@ -33,13 +33,19 @@
     llvm_toolchain(
         name = "llvm_toolchain",
         llvm_versions = {
-            "": "16.0.0",
+            "": "16.0.4",
             # The LLVM repo stops providing pre-built binaries for the MacOS x86_64
             # architecture for versions >= 16.0.0: https://github.com/llvm/llvm-project/releases,
             # but our Kokoro MacOS tests are still using x86_64 (ventura).
             # TODO: Upgrade the MacOS version to sonoma-slcn.
             "darwin-x86_64": "15.0.7",
         },
+        strip_prefix = {
+            "linux-x86_64": "clang+llvm-16.0.4-x86_64-linux-gnu-ubuntu-22.04",
+        },
+        urls = {
+            "linux-x86_64": ["https://github.com/llvm/llvm-project/releases/download/llvmorg-16.0.4/clang+llvm-16.0.4-x86_64-linux-gnu-ubuntu-22.04.tar.xz"],
+        },
     )
 
     http_archive(
