diff --git a/java/com/google/zetasql/BUILD b/java/com/google/zetasql/BUILD
index dec4bef..3bd4fab 100644
--- a/java/com/google/zetasql/BUILD
+++ b/java/com/google/zetasql/BUILD
@@ -198,7 +198,7 @@ javadoc_library(
 java_library(
     name = "jni_channel_linux",
     resources = select({
-        "@bazel_tools//src/conditions:linux_x86_64": ["//zetasql/local_service:liblocal_service_jni.so"],
+        "@bazel_tools//src/conditions:linux": ["//zetasql/local_service:liblocal_service_jni.so"],
         "//conditions:default": [],
     }),
     tags = ["maven_coordinates=com.google.zetasql:zetasql-jni-channel-linux:${project.version}"],
diff --git a/zetasql/jdk/BUILD b/zetasql/jdk/BUILD
index 56c30f5..41b5235 100644
--- a/zetasql/jdk/BUILD
+++ b/zetasql/jdk/BUILD
@@ -25,13 +25,13 @@ cc_library(
         "@local_jdk//:jni_header",
     ] + select({
         "@bazel_tools//src/conditions:darwin": ["@local_jdk//:jni_md_header-darwin"],
-        "@bazel_tools//src/conditions:linux_x86_64": ["@local_jdk//:jni_md_header-linux"],
+        "@bazel_tools//src/conditions:linux": ["@local_jdk//:jni_md_header-linux"],
     }),
     includes = [
         "../../external/local_jdk/include",
     ] + select({
         "@bazel_tools//src/conditions:darwin": ["../../external/local_jdk/include/darwin"],
-        "@bazel_tools//src/conditions:linux_x86_64": ["../../external/local_jdk/include/linux"],
+        "@bazel_tools//src/conditions:linux": ["../../external/local_jdk/include/linux"],
     }),
     linkstatic = 1,
 )