diff --git a/java/com/google/zetasql/BUILD b/java/com/google/zetasql/BUILD
index 3ca01857..d9cbc35a 100644
--- a/java/com/google/zetasql/BUILD
+++ b/java/com/google/zetasql/BUILD
@@ -204,7 +204,7 @@ javadoc_library(
 java_library(
     name = "jni_channel_linux",
     resources = select({
-        "@bazel_tools//src/conditions:linux_x86_64": ["//zetasql/local_service:liblocal_service_jni.so"],
+        "@bazel_tools//src/conditions:linux": ["//zetasql/local_service:liblocal_service_jni.so"],
         "//conditions:default": [],
     }),
     tags = ["maven_coordinates=com.google.zetasql:zetasql-jni-channel-linux:${project.version}"],
