diff --git a/bazel/boost.BUILD b/bazel/boost.BUILD
index 1c757ee4..ee1b0617 100644
--- a/bazel/boost.BUILD
+++ b/bazel/boost.BUILD
@@ -20,7 +20,7 @@ load("@rules_foreign_cc//foreign_cc:defs.bzl", "boost_build")
 
 filegroup(
     name = "all_srcs",
-    srcs = glob(["**"]),
+    srcs = glob(["**"], exclude=["libs/wave/test/testwave/testfiles/**"]),
     visibility = ["//visibility:private"],
 )
 
diff --git a/bazel/civetweb.BUILD b/bazel/civetweb.BUILD
index 5cc6271d..efb1728a 100644
--- a/bazel/civetweb.BUILD
+++ b/bazel/civetweb.BUILD
@@ -24,7 +24,7 @@ load("@rules_foreign_cc//foreign_cc:defs.bzl", "cmake")
 
 filegroup(
     name = "all",
-    srcs = glob(["**"]),
+    srcs = glob(["**"], exclude=["test/nonlatin/*"]),
 )
 
 cmake(
