diff --git a/src/wasm-traversal.h b/src/wasm-traversal.h
index 550a930..0f3ad06 100644
--- a/src/wasm-traversal.h
+++ b/src/wasm-traversal.h
@@ -394,7 +394,7 @@ struct PostWalker : public Walker<SubType, VisitorType> {
     // If we do *not* have an empty visitor, we can still optimize in the case
     // of a leaf: leaves have no children, so we can just call doVisit* rather
     // than push that task, pop it later, and call that.
-#if defined(__GNUC__) && !defined(__clang__) && __GNUC__ <= 11
+#if defined(__GNUC__) && !defined(__clang__) && __GNUC__ <= 13
 #define DELEGATE_START(id)                                                     \
   if (&SubType::visit##id !=                                                   \
         &Visitor<SubType, typename SubType::ReturnType>::visit##id ||          \
