From 16b88aca2ea222b264171535b3a3735031b9f1d2 Mon Sep 17 00:00:00 2001
From: Paolo Di Tommaso <paolo.ditommaso@gmail.com>
Date: Sat, 24 May 2025 21:59:34 +0200
Subject: [PATCH] Bump jgit:7.1.1.202505221757-r [ci fast]

Signed-off-by: Paolo Di Tommaso <paolo.ditommaso@gmail.com>
---
 modules/nextflow/build.gradle | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/nextflow/build.gradle b/modules/nextflow/build.gradle
index 57dfa8aa33..ae3dc903f8 100644
--- a/modules/nextflow/build.gradle
+++ b/modules/nextflow/build.gradle
@@ -44,7 +44,7 @@ dependencies {
     api "com.beust:jcommander:1.35"
     api("com.esotericsoftware.kryo:kryo:2.24.0") { exclude group: 'com.esotericsoftware.minlog', module: 'minlog' }
     api('org.iq80.leveldb:leveldb:0.12')
-    api('org.eclipse.jgit:org.eclipse.jgit:7.1.0.202411261347-r')
+    api('org.eclipse.jgit:org.eclipse.jgit:7.1.1.202505221757-r')
     api ('javax.activation:activation:1.1.1')
     api ('javax.mail:mail:1.4.7')
     api ('org.yaml:snakeyaml:2.2')
