diff --git a/modules/nextflow/build.gradle b/modules/nextflow/build.gradle
index 7dac6cece..0204482c2 100644
--- a/modules/nextflow/build.gradle
+++ b/modules/nextflow/build.gradle
@@ -34,8 +34,8 @@ dependencies {
     api "org.slf4j:jcl-over-slf4j:2.0.17"
     api "org.slf4j:jul-to-slf4j:2.0.17"
     api "org.slf4j:log4j-over-slf4j:2.0.17"
-    api "ch.qos.logback:logback-classic:1.5.18"
-    api "ch.qos.logback:logback-core:1.5.18"
+    api "ch.qos.logback:logback-classic:1.5.19"
+    api "ch.qos.logback:logback-core:1.5.19"
     api "org.codehaus.gpars:gpars:1.2.1"
     api("ch.artecat.grengine:grengine:3.0.2") { exclude group: 'org.codehaus.groovy' }
     api "commons-lang:commons-lang:2.6"
