diff --git a/modules/nextflow/build.gradle b/modules/nextflow/build.gradle
index 48fc98215..daa7fc584 100644
--- a/modules/nextflow/build.gradle
+++ b/modules/nextflow/build.gradle
@@ -29,8 +29,8 @@ dependencies {
     api "org.slf4j:jcl-over-slf4j:2.0.7"
     api "org.slf4j:jul-to-slf4j:2.0.7"
     api "org.slf4j:log4j-over-slf4j:2.0.7"
-    api "ch.qos.logback:logback-classic:1.4.14"
-    api "ch.qos.logback:logback-core:1.4.14"
+    api "ch.qos.logback:logback-classic:1.5.13"
+    api "ch.qos.logback:logback-core:1.5.13"
     api "org.codehaus.gpars:gpars:1.2.1"
     api("ch.artecat.grengine:grengine:3.0.2") { exclude group: 'org.codehaus.groovy' }
     api "commons-lang:commons-lang:2.6"
-- 
2.47.0