diff --git a/modules/nextflow/build.gradle b/modules/nextflow/build.gradle
index d7e40cf9d..13d7ea9b0 100644
--- a/modules/nextflow/build.gradle
+++ b/modules/nextflow/build.gradle
@@ -30,7 +30,7 @@ dependencies {
     api "org.slf4j:jul-to-slf4j:2.0.7"
     api "org.slf4j:log4j-over-slf4j:2.0.7"
     api "ch.qos.logback:logback-classic:1.4.14"
-    api "ch.qos.logback:logback-core:1.4.14"
+    api "ch.qos.logback:logback-core:1.5.13"
     api "org.codehaus.gpars:gpars:1.2.1"
     api("ch.artecat.grengine:grengine:3.0.2") { exclude group: 'org.codehaus.groovy' }
     api "commons-lang:commons-lang:2.6"
-- 
2.47.0
