diff --git a/pandas/_libs/meson.build b/pandas/_libs/meson.build
index b4662d6..34bfb2b 100644
--- a/pandas/_libs/meson.build
+++ b/pandas/_libs/meson.build
@@ -62,11 +62,11 @@ libs_sources = {
     # Dict of extension name -> dict of {sources, include_dirs, and deps}
     # numpy include dir is implicitly included
     'algos': {'sources': ['algos.pyx', _algos_common_helper, _algos_take_helper, _khash_primitive_helper]},
-    'arrays': {'sources': ['arrays.pyx']},
+    'arrays': {'sources': ['arrays.pyx', _khash_primitive_helper]},
     'groupby': {'sources': ['groupby.pyx']},
     'hashing': {'sources': ['hashing.pyx']},
     'hashtable': {'sources': ['hashtable.pyx', _khash_primitive_helper, _hashtable_class_helper, _hashtable_func_helper]},
-    'index': {'sources': ['index.pyx', _index_class_helper]},
+    'index': {'sources': ['index.pyx', _khash_primitive_helper, _index_class_helper]},
     'indexing': {'sources': ['indexing.pyx']},
     'internals': {'sources': ['internals.pyx']},
     'interval': {'sources': ['interval.pyx', _intervaltree_helper],
