diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6abf48a6..d9bea856 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -445,11 +445,12 @@ endif ()
 
 enable_testing() # Enable for tests without binary
 
-if (ARCH_S390X)
-    set(ENABLE_OPENSSL_DYNAMIC_DEFAULT ON)
-else ()
-    set(ENABLE_OPENSSL_DYNAMIC_DEFAULT OFF)
-endif ()
+# if (ARCH_S390X)
+#     set(ENABLE_OPENSSL_DYNAMIC_DEFAULT ON)
+# else ()
+#     set(ENABLE_OPENSSL_DYNAMIC_DEFAULT OFF)
+# endif ()
+set(ENABLE_OPENSSL_DYNAMIC_DEFAULT ON)
 option(ENABLE_OPENSSL_DYNAMIC "This option removes SSL from ClickHouse and will link to the OpenSSL version supplied by OS." ${ENABLE_OPENSSL_DYNAMIC_DEFAULT})
 
 # when installing to /usr - place configs to /etc but for /usr/local place to /usr/local/etc
