--- a/go.mod	2026-05-20 17:03:37.721588605 -0400
+++ b/go.mod	2026-05-20 17:03:37.748587998 -0400
@@ -8,10 +8,10 @@
 
 	// pin k8s.io to v0.34.3 & sigs.k8s.io/controller-runtime to v0.22.4
 	github.com/google/cel-go => github.com/google/cel-go v0.22.0
-	github.com/prometheus/client_golang => github.com/prometheus/client_golang v1.23.0
+	github.com/prometheus/client_golang => github.com/prometheus/client_golang v1.23.2
 	github.com/prometheus/client_model => github.com/prometheus/client_model v0.6.2
-	github.com/prometheus/common => github.com/prometheus/common v0.62.0
-	github.com/prometheus/prometheus => github.com/prometheus/prometheus v0.304.2
+	github.com/prometheus/common => github.com/prometheus/common v0.67.5
+	github.com/prometheus/prometheus => github.com/prometheus/prometheus v0.311.3
 	k8s.io/api => k8s.io/api v0.34.3
 	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.34.3
 	k8s.io/apimachinery => k8s.io/apimachinery v0.34.3
