diff --git a/pom.xml b/pom.xml
index 8ca0225d13..12c2398c04 100644
--- a/pom.xml
+++ b/pom.xml
@@ -45,8 +45,8 @@
         <jboss.snapshots.repo.id>jboss-snapshots-repository</jboss.snapshots.repo.id>
         <jboss.snapshots.repo.url>https://s01.oss.sonatype.org/content/repositories/snapshots/</jboss.snapshots.repo.url>
 
-        <quarkus.version>3.8.3</quarkus.version>
-        <quarkus.build.version>3.8.3</quarkus.build.version>
+        <quarkus.version>3.9.2</quarkus.version>
+        <quarkus.build.version>3.9.2</quarkus.build.version>
 
         <project.build-time>${timestamp}</project.build-time>
 
diff --git a/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/PropertyMapper.java b/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/PropertyMapper.java
index 3f64eb1755..71dab0d2a2 100644
--- a/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/PropertyMapper.java
+++ b/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/PropertyMapper.java
@@ -238,7 +238,7 @@ public class PropertyMapper<T> {
     }
 
     private ConfigValue transformValue(String name, Optional<String> value, ConfigSourceInterceptorContext context, String configSourceName) {
-        if (value == null) {
+        if (value == null || value.isEmpty()) {
             return null;
         }
 
