diff --git a/pear/Makefile.frag b/pear/Makefile.frag
index 9408757a..53f98762 100644
--- a/pear/Makefile.frag
+++ b/pear/Makefile.frag
@@ -1,7 +1,8 @@
 peardir=$(PEAR_INSTALLDIR)

 # Skip all php.ini files altogether
-PEAR_INSTALL_FLAGS = -n -dshort_open_tag=0 -dopen_basedir= -derror_reporting=1803 -dmemory_limit=-1
+PEAR_INSTALL_XML_FLAGS = -d extension="$(top_builddir)/modules/xml.so" -d extension="$(top_builddir)/modules/phar.so"
+PEAR_INSTALL_FLAGS = -n -dshort_open_tag=0 -dopen_basedir= -derror_reporting=1803 -dmemory_limit=-1 $(PEAR_INSTALL_XML_FLAGS)

 WGET = `which wget 2>/dev/null`
 FETCH = `which fetch 2>/dev/null`


