diff --git a/hatch_build.py b/hatch_build.py
index 13fa2a4..5a3ec09 100644
--- a/hatch_build.py
+++ b/hatch_build.py
@@ -425,7 +444,7 @@ DEPENDENCIES = [
     "connexion[flask]>=2.14.2,<3.0",
     "cron-descriptor>=1.2.24",
     "croniter>=2.0.2",
-    "cryptography>=41.0.0",
+    "cryptography>=44.0.1",
     "deprecated>=1.2.13",
     "dill>=0.2.2",
     # Required for python 3.8 and 3.9 to work with new annotations styles. Check package
