diff --git a/userspace/falco/CMakeLists.txt b/userspace/falco/CMakeLists.txt
index a67d8ab4..a8d627ab 100644
--- a/userspace/falco/CMakeLists.txt
+++ b/userspace/falco/CMakeLists.txt
@@ -68,7 +68,7 @@ set(FALCO_INCLUDE_DIRECTORIES
 )
 
 set(FALCO_DEPENDENCIES cxxopts)
-set(FALCO_LIBRARIES falco_engine sinsp yaml-cpp)
+set(FALCO_LIBRARIES falco_engine sinsp yaml-cpp "${JSONCPP_LIB}" "${TBB_LIB}" "${RE2_LIB}")
 
 if(USE_JEMALLOC)
        list(APPEND FALCO_DEPENDENCIES jemalloc)
