From 74d0bc25eac4cb078253404ffa5095ac8246be09 Mon Sep 17 00:00:00 2001
From: Melange Build <melange-build@cgr.dev>
Date: Mon, 26 Aug 2024 17:06:24 +0000
Subject: [PATCH] include-deps

---
 userspace/falco/CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/userspace/falco/CMakeLists.txt b/userspace/falco/CMakeLists.txt
index ca21186..52675a5 100644
--- a/userspace/falco/CMakeLists.txt
+++ b/userspace/falco/CMakeLists.txt
@@ -76,6 +76,9 @@ set(
   FALCO_LIBRARIES
   falco_engine
   sinsp
+  "${JSONCPP_LIB}"
+  "${TBB_LIB}"
+  "${RE2_LIB}"
   yaml-cpp
 )
 
-- 
2.46.0
