diff --git a/userspace/falco/CMakeLists.txt b/userspace/falco/CMakeLists.txt
index 13849e6f..43bbd7fd 100644
--- a/userspace/falco/CMakeLists.txt
+++ b/userspace/falco/CMakeLists.txt
@@ -69,7 +69,7 @@ set(FALCO_INCLUDE_DIRECTORIES
 
 set(FALCO_DEPENDENCIES cxxopts)
 
-set(FALCO_LIBRARIES falco_engine sinsp yaml-cpp)
+set(FALCO_LIBRARIES falco_engine sinsp yaml-cpp "${JSONCPP_LIB}" "${TBB_LIB}" "${RE2_LIB}")
 
 if(NOT WIN32)
        target_sources(falco_application PRIVATE outputs_program.cpp outputs_syslog.cpp)
