diff --git a/src/crypto/internal/backend/nobackend.go b/src/crypto/internal/backend/nobackend.go
index ac90ba299f..88be5de522 100644
--- a/src/crypto/internal/backend/nobackend.go
+++ b/src/crypto/internal/backend/nobackend.go
@@ -10,15 +10,15 @@ package backend
 import (
 	"crypto"
 	"crypto/cipher"
-	"crypto/internal/boring/sig"
-	"math/big"
 	bbig "crypto/internal/boring"
+	"crypto/internal/boring/sig"
 	"hash"
 	"io"
+	"math/big"
 )
 
 func init() {
-       strictFIPSNonCompliantBinaryCheck()
+	strictFIPSNonCompliantBinaryCheck()
 }
 
 var enabled = false
@@ -32,6 +32,10 @@ func Unreachable() {
 	sig.StandardCrypto()
 }
 
+func IsStrictFIPSMode() bool {
+	return false
+}
+
 // UnreachableExceptTests marks code that should be unreachable
 // when BoringCrypto is in use. It is a no-op without BoringCrypto.
 func UnreachableExceptTests() {}
diff --git a/src/crypto/internal/backend/not_strict_fips.go b/src/crypto/internal/backend/not_strict_fips.go
index f8e8fd6869..806b035aa8 100644
--- a/src/crypto/internal/backend/not_strict_fips.go
+++ b/src/crypto/internal/backend/not_strict_fips.go
@@ -3,6 +3,8 @@
 
 package backend
 
+var isStrictFIPS bool = false
+
 func strictFIPSOpenSSLRuntimeCheck() {
 }
 
diff --git a/src/crypto/internal/backend/openssl.go b/src/crypto/internal/backend/openssl.go
index 69a1c2bd0c..0e9fec07b7 100644
--- a/src/crypto/internal/backend/openssl.go
+++ b/src/crypto/internal/backend/openssl.go
@@ -71,6 +71,10 @@ func init() {
 	sig.BoringCrypto()
 }
 
+func IsStrictFIPSMode() bool {
+	return isStrictFIPS
+}
+
 func Enabled() bool {
 	return enabled
 }
diff --git a/src/crypto/internal/backend/strict_fips.go b/src/crypto/internal/backend/strict_fips.go
index 894eeca942..c1bda67f12 100644
--- a/src/crypto/internal/backend/strict_fips.go
+++ b/src/crypto/internal/backend/strict_fips.go
@@ -8,6 +8,8 @@ import (
 	"os"
 )
 
+var isStrictFIPS bool = true
+
 func strictFIPSOpenSSLRuntimeCheck() {
 	if hostFIPSModeEnabled() && !Enabled() {
 		fmt.Fprintln(os.Stderr, "FIPS mode is enabled, but the required OpenSSL backend is unavailable")
