diff --git a/src/crypto/internal/backend/openssl.go b/src/crypto/internal/backend/openssl.go
index 07d7692277..8812475b92 100644
--- a/src/crypto/internal/backend/openssl.go
+++ b/src/crypto/internal/backend/openssl.go
@@ -43,6 +43,9 @@ func init() {
 			version = fallbackVersion
 		}
 	}
+	if version == "" {
+		strictFIPSOpenSSLRuntimeCheck()
+	}
 	if err := openssl.Init(version); err != nil {
 		panic("opensslcrypto: can't initialize OpenSSL " + version + ": " + err.Error())
 	}
