diff --git a/src/crypto/x509/verify_test.go b/src/crypto/x509/verify_test.go
index 3551b470ce..6f9573d066 100644
--- a/src/crypto/x509/verify_test.go
+++ b/src/crypto/x509/verify_test.go
@@ -8,6 +8,7 @@ import (
 	"crypto"
 	"crypto/ecdsa"
 	"crypto/elliptic"
+	boring "crypto/internal/backend"
 	"crypto/rand"
 	"crypto/x509/pkix"
 	"encoding/asn1"
@@ -37,6 +38,8 @@ type verifyTest struct {
 
 	errorCallback  func(*testing.T, error)
 	expectedChains [][]string
+
+	boringSkip bool
 }
 
 var verifyTests = []verifyTest{
@@ -143,6 +146,7 @@ var verifyTests = []verifyTest{
 		// The StartCom root is not trusted by Windows when the default
 		// ServerAuth EKU is requested.
 		systemSkip: true,
+		boringSkip: true,
 
 		expectedChains: [][]string{
 			{"dnssec-exp", "StartCom Class 1", "StartCom Certification Authority"},
@@ -155,6 +159,7 @@ var verifyTests = []verifyTest{
 		roots:         []string{startComRoot},
 		currentTime:   1302726541,
 		keyUsages:     []ExtKeyUsage{ExtKeyUsageAny},
+		boringSkip:    true,
 
 		expectedChains: [][]string{
 			{"dnssec-exp", "StartCom Class 1", "StartCom Certification Authority"},
@@ -167,6 +172,7 @@ var verifyTests = []verifyTest{
 		roots:         []string{startComRoot},
 		currentTime:   1302726541,
 		systemSkip:    true, // see dnssec-exp test
+		boringSkip:    true,
 
 		expectedChains: [][]string{
 			{"dnssec-exp", "StartCom Class 1", "StartCom Certification Authority"},
@@ -228,6 +234,7 @@ var verifyTests = []verifyTest{
 		roots:         []string{globalSignRoot},
 		currentTime:   1382387896,
 		dnsName:       "secure.iddl.vt.edu",
+		boringSkip:    true,
 
 		expectedChains: [][]string{
 			{
@@ -557,6 +564,9 @@ func TestGoVerify(t *testing.T) {
 
 	for _, test := range verifyTests {
 		t.Run(test.name, func(t *testing.T) {
+			if test.boringSkip && boring.Enabled() {
+				t.Skip("skipping test with BoringCrypto")
+			}
 			testVerify(t, test, false)
 		})
 	}
