diff --git a/src/crypto/tls/boring.go b/src/crypto/tls/boring.go
index 66252067f2..5be6dcea23 100644
--- a/src/crypto/tls/boring.go
+++ b/src/crypto/tls/boring.go
@@ -22,6 +22,10 @@ func needFIPS() bool {
 	return fipstls.Required()
 }
 
+func supportsHKDF() bool {
+	return boring.SupportsHKDF()
+}
+
 // fipsMinVersion replaces c.minVersion in FIPS-only mode.
 func fipsMinVersion(c *Config) uint16 {
 	// FIPS requires TLS 1.2 or later.
@@ -33,7 +37,7 @@ func fipsMaxVersion(c *Config) uint16 {
 	// FIPS requires TLS 1.2 or later.
 	if boring.SupportsHKDF() {
 		return VersionTLS13
-	}  else {
+	} else {
 		return VersionTLS12
 	}
 }
diff --git a/src/crypto/tls/boring_test.go b/src/crypto/tls/boring_test.go
index 8943422ae0..10a4bc296c 100644
--- a/src/crypto/tls/boring_test.go
+++ b/src/crypto/tls/boring_test.go
@@ -77,7 +77,7 @@ func isBoringCipherSuite(id uint16) bool {
 		TLS_RSA_WITH_AES_128_GCM_SHA256,
 		TLS_RSA_WITH_AES_256_GCM_SHA384,
 		TLS_AES_128_GCM_SHA256,
-		TLS_AES_256_GCM_SHA384,
+		TLS_AES_256_GCM_SHA384:
 		return true
 	}
 	return false
diff --git a/src/crypto/tls/handshake_client.go b/src/crypto/tls/handshake_client.go
index f016e01b4b..0ecdd6c1cc 100644
--- a/src/crypto/tls/handshake_client.go
+++ b/src/crypto/tls/handshake_client.go
@@ -139,7 +141,9 @@ func (c *Conn) makeClientHello() (*clientHelloMsg, *ecdh.PrivateKey, error) {
 		if len(hello.supportedVersions) == 1 {
 			hello.cipherSuites = nil
 		}
-		if hasAESGCMHardwareSupport {
+		if needFIPS() {
+			hello.cipherSuites = append(hello.cipherSuites, defaultFIPSCipherSuitesTLS13...)
+		} else if hasAESGCMHardwareSupport {
 			hello.cipherSuites = append(hello.cipherSuites, defaultCipherSuitesTLS13...)
 		} else {
 			hello.cipherSuites = append(hello.cipherSuites, defaultCipherSuitesTLS13NoAES...)
diff --git a/src/crypto/tls/handshake_client_tls13.go b/src/crypto/tls/handshake_client_tls13.go
index fb79939f56..e2c2e8842e 100644
--- a/src/crypto/tls/handshake_client_tls13.go
+++ b/src/crypto/tls/handshake_client_tls13.go
@@ -41,7 +41,7 @@ type clientHandshakeStateTLS13 struct {
 func (hs *clientHandshakeStateTLS13) handshake() error {
 	c := hs.c
 
-	if needFIPS() && !boring.SupportsHKDF() {
+	if needFIPS() && !supportsHKDF() {
 		return errors.New("tls: internal error: TLS 1.3 reached in FIPS mode")
 	}
 
diff --git a/src/crypto/tls/handshake_server_tls13.go b/src/crypto/tls/handshake_server_tls13.go
index 816ca10858..90256aa27c 100644
--- a/src/crypto/tls/handshake_server_tls13.go
+++ b/src/crypto/tls/handshake_server_tls13.go
@@ -159,6 +161,9 @@ func (hs *serverHandshakeStateTLS13) processClientHello() error {
 	if !hasAESGCMHardwareSupport || !aesgcmPreferred(hs.clientHello.cipherSuites) {
 		preferenceList = defaultCipherSuitesTLS13NoAES
 	}
+	if needFIPS() {
+		preferenceList = defaultFIPSCipherSuitesTLS13
+	}
 	for _, suiteID := range preferenceList {
 		hs.suite = mutualCipherSuiteTLS13(hs.clientHello.cipherSuites, suiteID)
 		if hs.suite != nil {
