diff --git a/Makefile.am b/Makefile.am
index 5a5715d1..b1f9a8a6 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -427,7 +427,6 @@ libcmds_la_SOURCES = \
 	src/utils_cmd_getthreshold.c \
 	src/utils_cmd_getthreshold.h \
 	src/utils_cmd_getval.c \
-	src/utils_cmd_putinsight.c \
 	src/utils_cmd_getval.h \
 	src/utils_cmd_listval.c \
 	src/utils_cmd_listval.h \
diff --git a/configure.ac b/configure.ac
index d5d6a8b5..c54a44c9 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1,7 +1,6 @@
 dnl Process this file with autoconf to produce a configure script.
 AC_PREREQ([2.60])
 AC_INIT([collectd],[m4_esyscmd(./version-gen.sh)])
-AC_CONFIG_MACRO_DIR([./aclocal])
 AC_CONFIG_SRCDIR(src/target_set.c)
 AC_CONFIG_HEADERS(src/config.h)
 AC_CONFIG_AUX_DIR([build-aux])
