From 8bb1d8aceac49212f3035f7bc497f9d1535375f1 Mon Sep 17 00:00:00 2001
From: uti <uti.edun@chainguard.dev>
Date: Mon, 24 Feb 2025 07:31:21 +0000
Subject: [PATCH] s3select: error: cannot convert 'std::shared_ptr<parquet::InternalFileDecryptor>' to 'const parquet::ReaderProperties&'
 object

---
 include/s3select_parquet_intrf.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/s3select/include/s3select_parquet_intrf.h b/include/s3select_parquet_intrf.h
index df04e161..0dd05c33 100644
--- a/src/s3select/include/s3select_parquet_intrf.h
+++ b/src/s3select/include/s3select_parquet_intrf.h
@@ -1245,8 +1245,8 @@ void SerializedFile::ParseMetaDataOfEncryptedFileWithEncryptedFooter(
   }
 
   file_metadata_ =
-	FileMetaData::Make(metadata_buffer->data(), &metadata_len, file_decryptor_);
-      	//FileMetaData::Make(metadata_buffer->data(), &metadata_len, default_reader_properties(), file_decryptor_); //version>9
+	//FileMetaData::Make(metadata_buffer->data(), &metadata_len, file_decryptor_);
+      	FileMetaData::Make(metadata_buffer->data(), &metadata_len, default_reader_properties(), file_decryptor_); //version>9
 }
 
 void SerializedFile::ParseMetaDataOfEncryptedFileWithPlaintextFooter(
-- 
2.48.1

