diff --git a/racc.gemspec b/racc.gemspec
index c25c0c2..92a7551 100644
--- a/racc.gemspec
+++ b/racc.gemspec
@@ -30,7 +30,7 @@ DESC
     "lib/racc/debugflags.rb", "lib/racc/exception.rb",
     "lib/racc/grammar.rb", "lib/racc/grammarfileparser.rb",
     "lib/racc/info.rb", "lib/racc/iset.rb",
-    "lib/racc/logfilegenerator.rb", "lib/racc/parser-text.rb",
+    "lib/racc/logfilegenerator.rb",
     "lib/racc/parser.rb", "lib/racc/parserfilegenerator.rb",
     "lib/racc/sourcetext.rb",
     "lib/racc/state.rb", "lib/racc/statetransitiontable.rb",
