From ed1acc546676751e2c6d72e754166de385950a76 Mon Sep 17 00:00:00 2001
From: Markus Boehme <markus.boehme@mailbox.org>
Date: Wed, 27 Aug 2025 22:49:29 +0200
Subject: [PATCH] pkgconf: expose variables for system-alloc-uid-min

Signed-off-by: Markus Boehme <markus.boehme@mailbox.org>
---
 src/core/systemd.pc.in | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/core/systemd.pc.in b/src/core/systemd.pc.in
index f3b85b0190..952669bd4c 100644
--- a/src/core/systemd.pc.in
+++ b/src/core/systemd.pc.in
@@ -87,6 +87,8 @@ modulesloaddir=${modules_load_dir}
 catalog_dir=${prefix}/lib/systemd/catalog
 catalogdir=${catalog_dir}
 
+system_alloc_uid_min={{SYSTEM_ALLOC_UID_MIN}}
+systemallocuidmin=${system_alloc_uid_min}
 system_uid_max={{SYSTEM_UID_MAX}}
 systemuidmax=${system_uid_max}
 system_gid_max={{SYSTEM_GID_MAX}}
-- 
2.50.1

