diff --git a/charts/build-chart.sh b/charts/build-chart.sh
index db9ecc80..915bc665 100755
--- a/charts/build-chart.sh
+++ b/charts/build-chart.sh
@@ -8,8 +8,8 @@ set -eux -o pipefail
 : "${CHART_PACKAGE:="${CHART_NAME%%-crd}"}"
 : "${TAR_OPTS:=--owner=0 --group=0 --mode=gou-s+r --numeric-owner --no-acls --no-selinux --no-xattrs}"
 : "${CHART_URL:="${CHART_REPO:="https://rke2-charts.rancher.io"}/assets/${CHART_PACKAGE}/${CHART_NAME}-${CHART_VERSION:="0.0.0"}.tgz"}"
-: "${CHART_TMP:=$(mktemp --suffix .tar)}"
-: "${YAML_TMP:=$(mktemp --suffix .yaml)}"
+: "${CHART_TMP:=$(mktemp /tmp/tmpchart.XXXXXX).tar}"
+: "${YAML_TMP:=$(mktemp /tmp/tmpvalues.XXXXXX).yaml}"
 
 cleanup() {
   exit_code=$?
@@ -29,8 +29,16 @@ curl -fsSL "${CHART_URL}" | pigz -dc > "${CHART_TMP}"
 # Extract out Chart.yaml, inject a version requirement and bundle-id annotation, and delete/replace the one in the original tarball
 tar -xOf ${CHART_TMP} ${CHART_NAME}/Chart.yaml > ${YAML_TMP}
 yq -i e ".kubeVersion = \">= ${KUBERNETES_VERSION}\" | .annotations.\"fleet.cattle.io/bundle-id\" = \"rke2\"" ${YAML_TMP}
-tar --delete -b 8192 -f ${CHART_TMP} ${CHART_NAME}/Chart.yaml
-tar --transform="s|.*|${CHART_NAME}/Chart.yaml|" ${TAR_OPTS} -vrf ${CHART_TMP} ${YAML_TMP}
+# BEGIN BusyBox tar-compatible repack workaround
+TMP_REPACK_DIR="$(mktemp -d)"
+tar -xf "${CHART_TMP}" -C "${TMP_REPACK_DIR}"
+CHART_NAME="$(basename "${CHART_FILE}" .yaml)"
+cp "${YAML_TMP}" "${TMP_REPACK_DIR}/${CHART_NAME}/Chart.yaml"
+tar -cf "${CHART_TMP}" -C "${TMP_REPACK_DIR}" .
+rm -rf "${TMP_REPACK_DIR}"
+# END repack workaround
+# Chart.yaml was already replaced inside the archive, no need to re-add it with transform
+#tar --transform="s|.*|${CHART_NAME}/Chart.yaml|" ${TAR_OPTS} -vrf ${CHART_TMP} ${YAML_TMP}
 
 pigz -11 ${CHART_TMP}
 
