Description: Add tinfo to LDLIB for cgdisk command
Author: James Page <james.page@chainguard.dev>

diff --git a/Makefile b/Makefile
index af9701c..87021cd 100644
--- a/Makefile
+++ b/Makefile
@@ -72,7 +72,7 @@ LDLIBS+=-luuid #-licuio -licuuc
 FATBINFLAGS=
 THINBINFLAGS=
 SGDISK_LDLIBS=-lpopt
-CGDISK_LDLIBS=-lncursesw
+CGDISK_LDLIBS=-lncursesw -ltinfo
 LIB_NAMES=crc32 support guid gptpart mbrpart basicmbr mbr gpt bsd parttypes attributes diskio diskio-unix
 MBR_LIBS=support diskio diskio-unix basicmbr mbrpart
 ALL=gdisk cgdisk sgdisk fixparts
