Description: Correct include path for ncurses.h
Origin: https://git.alpinelinux.org/aports/plain/main/gptfdisk/

diff --git a/gptcurses.cc b/gptcurses.cc
index 1b18cf2..4ebfde1 100644
--- a/gptcurses.cc
+++ b/gptcurses.cc
@@ -23,11 +23,7 @@
 #include <iostream>
 #include <string>
 #include <sstream>
-#if defined (__APPLE__) || (__FreeBSD__)
 #include <ncurses.h>
-#else
-#include <ncursesw/ncurses.h>
-#endif
 #include "gptcurses.h"
 #include "support.h"
 
