From e7f90a0931b3e07cc8f44a82e33893a862359f0d Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.ledkov@surgut.co.uk>
Date: Wed, 3 Dec 2025 11:18:32 +0000
Subject: [PATCH] jitter-instrument

---
 .../rands/seed_src_jitter.c.in                | 21 ++++++++++++++++++-
 1 file changed, 20 insertions(+), 1 deletion(-)

diff --git a/providers/implementations/rands/seed_src_jitter.c.in b/providers/implementations/rands/seed_src_jitter.c.in
index 4d73f07574..c674588a54 100644
--- a/providers/implementations/rands/seed_src_jitter.c.in
+++ b/providers/implementations/rands/seed_src_jitter.c.in
@@ -22,6 +22,7 @@ use OpenSSL::paramnames qw(produce_param_decoder);
 #include <openssl/proverr.h>
 #include <openssl/self_test.h>
 #include "internal/common.h"
+#include "internal/cryptlib.h"
 #include "prov/implementations.h"
 #include "prov/provider_ctx.h"
 #include "prov/providercommon.h"
@@ -102,6 +103,14 @@ static size_t get_jitter_random_value(PROV_JITTER *s,
         result = jent_read_entropy(jitter_ec, (char *) buf, len);
         jent_entropy_collector_free(jitter_ec);
 
+#ifdef FIPS_MODULE
+	if (ossl_safe_getenv("FIPS_FAIL_GET_JITTER_RANDOM_VALUE"))
+	  result = -6;
+#else
+	if (ossl_safe_getenv("NONFIPS_FAIL_GET_JITTER_RANDOM_VALUE"))
+	  result = -6;
+#endif
+
         /*
          * Permanent Failure
          * https://github.com/smuellerDD/jitterentropy-library/blob/master/doc/jitterentropy.3#L234
@@ -156,7 +165,17 @@ static int jitter_instantiate(void *vseed, unsigned int strength,
     PROV_JITTER *s = (PROV_JITTER *)vseed;
     int ret;
 
-    if ((ret = jent_entropy_init_ex(0, JENT_FORCE_FIPS)) != 0) {
+    ret = jent_entropy_init_ex(0, JENT_FORCE_FIPS);
+
+#ifdef FIPS_MODULE
+	if (ossl_safe_getenv("FIPS_FAIL_JITTER_INSTANTIATE"))
+	  ret = 1;
+#else
+	if (ossl_safe_getenv("NONFIPS_FAIL_JITTER_INSTANTIATE"))
+	  ret = 1;
+#endif
+
+    if (ret != 0) {
         ERR_raise_data(ERR_LIB_RAND, RAND_R_ERROR_RETRIEVING_ENTROPY,
                        "jent_entropy_init_ex (%d)", ret);
         s->state = EVP_RAND_STATE_ERROR;
-- 
2.51.0

