diff --git a/crypto/provider_core.c b/crypto/provider_core.c
index 490991b5e5..3cfe24ed9d 100644
--- a/crypto/provider_core.c
+++ b/crypto/provider_core.c
@@ -2421,7 +2421,7 @@ static void core_self_test_get_callback(OPENSSL_CORE_CTX *libctx,
     OSSL_SELF_TEST_get_callback((OSSL_LIB_CTX *)libctx, cb, cbarg);
 }
 
-# ifdef OPENSSL_NO_FIPS_JITTER
+# ifdef OPENSSL_NO_JITTER
 static size_t rand_get_entropy(const OSSL_CORE_HANDLE *handle,
                                unsigned char **pout, int entropy,
                                size_t min_len, size_t max_len)
diff --git a/providers/implementations/rands/seed_src_jitter.c b/providers/implementations/rands/seed_src_jitter.c
index ac57f1c14b..b6f2849069 100644
--- a/providers/implementations/rands/seed_src_jitter.c
+++ b/providers/implementations/rands/seed_src_jitter.c
@@ -295,7 +295,6 @@ static size_t jitter_get_seed(void *vseed, unsigned char **pout,
     return ret;
 }
 
-# ifndef OPENSSL_NO_FIPS_JITTER
 size_t ossl_rand_jitter_get_seed(unsigned char **pout, int entropy, size_t min_len, size_t max_len)
 {
     size_t ret = 0;
@@ -311,7 +310,6 @@ size_t ossl_rand_jitter_get_seed(unsigned char **pout, int entropy, size_t min_l
     jitter_free(s);
     return ret;
 }
-# endif
 
 static void jitter_clear_seed(ossl_unused void *vdrbg,
                               unsigned char *out, size_t outlen)
