diff --git a/providers/defltprov.c b/providers/defltprov.c
index a446e888f5..6ce3334827 100644
--- a/providers/defltprov.c
+++ b/providers/defltprov.c
@@ -150,7 +150,7 @@ static const OSSL_ALGORITHM deflt_digests[] = {
 #endif /* OPENSSL_NO_SM3 */
 
 #ifndef OPENSSL_NO_MD5
-    { PROV_NAMES_MD5, "provider=default", ossl_md5_functions },
+    { PROV_NAMES_MD5, "provider=default,fips=yes", ossl_md5_functions },
     { PROV_NAMES_MD5_SHA1, "provider=default", ossl_md5_sha1_functions },
 #endif /* OPENSSL_NO_MD5 */
 
