diff --git a/Ghidra/Debug/Debugger-rmi-trace/Module.manifest b/Ghidra/Debug/Debugger-rmi-trace/Module.manifest
index fefac7bf9c..eb1c5ba2d0 100644
--- a/Ghidra/Debug/Debugger-rmi-trace/Module.manifest
+++ b/Ghidra/Debug/Debugger-rmi-trace/Module.manifest
@@ -1,4 +1,4 @@
 MODULE FILE LICENSE: pypkg/dist/protobuf-3.20.3-py2.py3-none-any.whl BSD-3-GOOGLE
 MODULE FILE LICENSE: pypkg/dist/psutil-5.9.8.tar.gz BSD-3-PSUTIL
-MODULE FILE LICENSE: pypkg/dist/setuptools-68.0.0-py3-none-any.whl MIT
-MODULE FILE LICENSE: pypkg/dist/wheel-0.37.1-py2.py3-none-any.whl MIT
+MODULE FILE LICENSE: pypkg/dist/setuptools-75.8.0-py3-none-any.whl MIT
+MODULE FILE LICENSE: pypkg/dist/wheel-0.45.1-py3-none-any.whl MIT
diff --git a/Ghidra/Debug/Debugger-rmi-trace/build.gradle b/Ghidra/Debug/Debugger-rmi-trace/build.gradle
index 8b6589f035..af44dd6f4c 100644
--- a/Ghidra/Debug/Debugger-rmi-trace/build.gradle
+++ b/Ghidra/Debug/Debugger-rmi-trace/build.gradle
@@ -65,5 +65,5 @@ tasks.assemblePyPackage {
 
 distributePyDep("protobuf-3.20.3-py2.py3-none-any.whl")
 distributePyDep("psutil-5.9.8.tar.gz")
-distributePyDep("setuptools-68.0.0-py3-none-any.whl")
-distributePyDep("wheel-0.37.1-py2.py3-none-any.whl")
+distributePyDep("setuptools-75.8.0-py3-none-any.whl")
+distributePyDep("wheel-0.45.1-py3-none-any.whl")
diff --git a/Ghidra/Debug/Framework-AsyncComm/Module.manifest b/Ghidra/Debug/Framework-AsyncComm/Module.manifest
index 89174007c9..6b95c47ba8 100644
--- a/Ghidra/Debug/Framework-AsyncComm/Module.manifest
+++ b/Ghidra/Debug/Framework-AsyncComm/Module.manifest
@@ -1 +1 @@
-MODULE FILE LICENSE: lib/protobuf-java-3.21.8.jar BSD-3-GOOGLE
+MODULE FILE LICENSE: lib/protobuf-java-3.25.5.jar BSD-3-GOOGLE
diff --git a/Ghidra/Debug/Framework-AsyncComm/build.gradle b/Ghidra/Debug/Framework-AsyncComm/build.gradle
index a84e156808..de4ac1d993 100644
--- a/Ghidra/Debug/Framework-AsyncComm/build.gradle
+++ b/Ghidra/Debug/Framework-AsyncComm/build.gradle
@@ -22,7 +22,7 @@ apply plugin: 'eclipse'
 eclipse.project.name = 'Debug Framework-AsyncComm'
 
 dependencies {
-	api 'com.google.protobuf:protobuf-java:3.21.8'
+	api 'com.google.protobuf:protobuf-java:3.25.5'
 	api project(':Generic')
 	api project(':Graph')
 	api project(':ProposedUtils')
diff --git a/Ghidra/Extensions/MachineLearning/Module.manifest b/Ghidra/Extensions/MachineLearning/Module.manifest
index b819a8578c..019c2db18f 100644
--- a/Ghidra/Extensions/MachineLearning/Module.manifest
+++ b/Ghidra/Extensions/MachineLearning/Module.manifest
@@ -1,6 +1,6 @@
 MODULE FILE LICENSE: lib/olcut-config-protobuf-5.2.0.jar BSD-2-ORACLE
 MODULE FILE LICENSE: lib/olcut-core-5.2.0.jar BSD-2-ORACLE
-MODULE FILE LICENSE: lib/protobuf-java-3.17.3.jar BSD-3-GOOGLE
+MODULE FILE LICENSE: lib/protobuf-java-3.25.5.jar BSD-3-GOOGLE
 MODULE FILE LICENSE: lib/tribuo-classification-core-4.2.0.jar Apache License 2.0
 MODULE FILE LICENSE: lib/tribuo-classification-tree-4.2.0.jar Apache License 2.0
 MODULE FILE LICENSE: lib/tribuo-common-tree-4.2.0.jar Apache License 2.0
diff --git a/Ghidra/Extensions/MachineLearning/build.gradle b/Ghidra/Extensions/MachineLearning/build.gradle
index caaa610a7d..039f0a616e 100644
--- a/Ghidra/Extensions/MachineLearning/build.gradle
+++ b/Ghidra/Extensions/MachineLearning/build.gradle
@@ -26,7 +26,7 @@ dependencies {
 
     api "com.oracle.labs.olcut:olcut-config-protobuf:5.2.0" //{exclude group: "com.google.protobuf", module: "protobuf-java"}
     api ("com.oracle.labs.olcut:olcut-core:5.2.0") {exclude group: "org.jline"}
-    api "com.google.protobuf:protobuf-java:3.17.3"   //only needed for running junits
+    api "com.google.protobuf:protobuf-java:3.25.5"   //only needed for running junits
     api "org.tribuo:tribuo-classification-core:4.2.0"
     api "org.tribuo:tribuo-classification-tree:4.2.0"
     api "org.tribuo:tribuo-common-tree:4.2.0"
diff --git a/Ghidra/Features/GhidraServer/Module.manifest b/Ghidra/Features/GhidraServer/Module.manifest
index 070fe00556..8f9897a3e2 100644
--- a/Ghidra/Features/GhidraServer/Module.manifest
+++ b/Ghidra/Features/GhidraServer/Module.manifest
@@ -1,2 +1,2 @@
-DATA SEARCH IGNORE DIR: yajsw-stable-13.12
+DATA SEARCH IGNORE DIR: yajsw-stable-13.13
 EXCLUDE FROM GHIDRA JAR: true
diff --git a/Ghidra/Features/GhidraServer/build.gradle b/Ghidra/Features/GhidraServer/build.gradle
index abfa67a369..6fed007127 100644
--- a/Ghidra/Features/GhidraServer/build.gradle
+++ b/Ghidra/Features/GhidraServer/build.gradle
@@ -21,7 +21,7 @@ apply plugin: 'eclipse'
 
 eclipse.project.name = 'Features GhidraServer'
 
-def yajswRelease = "yajsw-stable-13.12"
+def yajswRelease = "yajsw-stable-13.13"
 
 configurations {
 	runGhidraServer
diff --git a/Ghidra/Features/GnuDemangler/src/main/java/ghidra/app/util/demangler/gnu/GnuDemanglerNativeProcess.java b/Ghidra/Features/GnuDemangler/src/main/java/ghidra/app/util/demangler/gnu/GnuDemanglerNativeProcess.java
index 73cfa127ee..9977e0c9ee 100644
--- a/Ghidra/Features/GnuDemangler/src/main/java/ghidra/app/util/demangler/gnu/GnuDemanglerNativeProcess.java
+++ b/Ghidra/Features/GnuDemangler/src/main/java/ghidra/app/util/demangler/gnu/GnuDemanglerNativeProcess.java
@@ -238,7 +238,7 @@ public class GnuDemanglerNativeProcess {
 			List<String> errorLines = IOUtils.readLines(err, Charset.defaultCharset());
 			error = StringUtils.join(errorLines, '\n');
 		}
-		catch (IOException e) {
+		catch (Exception e) {
 			throw new IOException("Unable to read process error stream: ", e);
 		}
 
diff --git a/Ghidra/Features/PyGhidra/Module.manifest b/Ghidra/Features/PyGhidra/Module.manifest
index c048c52a90..b3e2ff8655 100644
--- a/Ghidra/Features/PyGhidra/Module.manifest
+++ b/Ghidra/Features/PyGhidra/Module.manifest
@@ -21,5 +21,5 @@ MODULE FILE LICENSE: pypkg/dist/jpype1-1.5.2-cp39-cp39-manylinux_2_17_aarch64.ma
 MODULE FILE LICENSE: pypkg/dist/jpype1-1.5.2-cp39-cp39-macosx_10_9_x86_64.whl Apache License 2.0
 MODULE FILE LICENSE: pypkg/dist/jpype1-1.5.2.tar.gz Apache License 2.0
 MODULE FILE LICENSE: pypkg/dist/packaging-23.2-py3-none-any.whl Apache License 2.0
-MODULE FILE LICENSE: pypkg/dist/setuptools-68.0.0-py3-none-any.whl MIT
-MODULE FILE LICENSE: pypkg/dist/wheel-0.37.1-py2.py3-none-any.whl MIT
+MODULE FILE LICENSE: pypkg/dist/setuptools-75.8.0-py3-none-any.whl MIT
+MODULE FILE LICENSE: pypkg/dist/wheel-0.45.1-py3-none-any.whl MIT
diff --git a/Ghidra/Features/PyGhidra/build.gradle b/Ghidra/Features/PyGhidra/build.gradle
index 9017400fc4..2a366f4f1f 100644
--- a/Ghidra/Features/PyGhidra/build.gradle
+++ b/Ghidra/Features/PyGhidra/build.gradle
@@ -56,8 +56,8 @@ distributePyDep("jpype1-1.5.2-cp39-cp39-manylinux_2_17_aarch64.manylinux2014_aar
 distributePyDep("jpype1-1.5.2-cp39-cp39-macosx_10_9_x86_64.whl")
 distributePyDep("jpype1-1.5.2.tar.gz")
 distributePyDep("packaging-23.2-py3-none-any.whl")
-distributePyDep("setuptools-68.0.0-py3-none-any.whl")
-distributePyDep("wheel-0.37.1-py2.py3-none-any.whl")
+distributePyDep("setuptools-75.8.0-py3-none-any.whl")
+distributePyDep("wheel-0.45.1-py3-none-any.whl")
 
 // Install JPype into the development virtual environment
 task installJPype(type: Exec) {
diff --git a/Ghidra/Framework/Generic/Module.manifest b/Ghidra/Framework/Generic/Module.manifest
index d42b2be056..21b7a68703 100644
--- a/Ghidra/Framework/Generic/Module.manifest
+++ b/Ghidra/Framework/Generic/Module.manifest
@@ -1,4 +1,4 @@
-MODULE FILE LICENSE: lib/commons-compress-1.21.jar Apache License 2.0
+MODULE FILE LICENSE: lib/commons-compress-1.26.0.jar Apache License 2.0
 MODULE FILE LICENSE: lib/guava-32.1.3-jre.jar Apache License 2.0
 MODULE FILE LICENSE: lib/failureaccess-1.0.1.jar Apache License 2.0
 MODULE FILE LICENSE: lib/jdom-legacy-1.1.3.jar JDOM License
@@ -6,9 +6,9 @@ MODULE FILE LICENSE: lib/log4j-api-2.17.1.jar Apache License 2.0
 MODULE FILE LICENSE: lib/log4j-core-2.17.1.jar Apache License 2.0
 MODULE FILE LICENSE: lib/commons-collections4-4.1.jar Apache License 2.0
 MODULE FILE LICENSE: lib/commons-lang3-3.12.0.jar Apache License 2.0
-MODULE FILE LICENSE: lib/commons-io-2.11.0.jar Apache License 2.0
+MODULE FILE LICENSE: lib/commons-io-2.18.0.jar Apache License 2.0
 MODULE FILE LICENSE: lib/commons-text-1.10.0.jar Apache License 2.0
 MODULE FILE LICENSE: lib/gson-2.9.0.jar Apache License 2.0
-MODULE FILE LICENSE: lib/bcpkix-jdk15on-1.69.jar Bouncy Castle License
-MODULE FILE LICENSE: lib/bcprov-jdk15on-1.69.jar Bouncy Castle License
-MODULE FILE LICENSE: lib/bcutil-jdk15on-1.69.jar Bouncy Castle License
+MODULE FILE LICENSE: lib/bcpkix-jdk18on-1.78.jar Bouncy Castle License
+MODULE FILE LICENSE: lib/bcprov-jdk18on-1.78.jar Bouncy Castle License
+MODULE FILE LICENSE: lib/bcutil-jdk18on-1.78.jar Bouncy Castle License
diff --git a/Ghidra/Framework/Generic/build.gradle b/Ghidra/Framework/Generic/build.gradle
index 105095fa02..f7f6d5e91f 100644
--- a/Ghidra/Framework/Generic/build.gradle
+++ b/Ghidra/Framework/Generic/build.gradle
@@ -37,14 +37,14 @@ dependencies {
 	api "org.apache.logging.log4j:log4j-api:2.17.1"
 	api "org.apache.logging.log4j:log4j-core:2.17.1"
 	api "org.apache.commons:commons-collections4:4.1"
-	api "org.apache.commons:commons-compress:1.21"
+	api "org.apache.commons:commons-compress:1.26.0"
 	api "org.apache.commons:commons-lang3:3.12.0"
 	api "org.apache.commons:commons-text:1.10.0"
-	api "commons-io:commons-io:2.11.0"	
+	api "commons-io:commons-io:2.18.0"
 	api "com.google.code.gson:gson:2.9.0"
-	api "org.bouncycastle:bcpkix-jdk15on:1.69" // requires bcutil and bcprov
-	api "org.bouncycastle:bcprov-jdk15on:1.69"
-	api "org.bouncycastle:bcutil-jdk15on:1.69"
+	api "org.bouncycastle:bcpkix-jdk18on:1.78" // requires bcutil and bcprov
+	api "org.bouncycastle:bcprov-jdk18on:1.78"
+	api "org.bouncycastle:bcutil-jdk18on:1.78"
 
 	compileOnly "junit:junit:4.12"
 	compileOnly "org.hamcrest:hamcrest:2.2"
diff --git a/gradle/support/fetchDependencies.gradle b/gradle/support/fetchDependencies.gradle
index eee5bb1870..81e49f2302 100644
--- a/gradle/support/fetchDependencies.gradle
+++ b/gradle/support/fetchDependencies.gradle
@@ -74,9 +74,9 @@ ext.deps = [
 		destination: FLAT_REPO_DIR
 	],
 	[
-		name: "yajsw-stable-13.12.zip",
-		url: "https://sourceforge.net/projects/yajsw/files/yajsw/yajsw-stable-13.12/yajsw-stable-13.12.zip",
-		sha256: "c6fc59815d3800d14ec977926a8afd3f606a0ebd74d2cfd60601677466edeaa2",
+		name: "yajsw-stable-13.13.zip",
+		url: "https://sourceforge.net/projects/yajsw/files/yajsw/yajsw-stable-13.13/yajsw-stable-13.13.zip",
+		sha256: "13fd79f2bdbf91dd9ac0c157c86440240eced2f21488a056e0843d2b8d459772",
 		destination: file("${DEPS_DIR}/GhidraServer")
 	],
 	[
@@ -170,15 +170,15 @@ ext.deps = [
 		destination: file("${DEPS_DIR}/Debugger-rmi-trace/")
 	],
 	[
-		name: "setuptools-68.0.0-py3-none-any.whl",
-		url: "https://files.pythonhosted.org/packages/c7/42/be1c7bbdd83e1bfb160c94b9cafd8e25efc7400346cf7ccdbdb452c467fa/setuptools-68.0.0-py3-none-any.whl",
-		sha256: "11e52c67415a381d10d6b462ced9cfb97066179f0e871399e006c4ab101fc85f",
+		name: "setuptools-75.8.0-py3-none-any.whl",
+		url: "https://files.pythonhosted.org/packages/69/8a/b9dc7678803429e4a3bc9ba462fa3dd9066824d3c607490235c6a796be5a/setuptools-75.8.0-py3-none-any.whl",
+		sha256: "e3982f444617239225d675215d51f6ba05f845d4eec313da4418fdbb56fb27e3",
 		destination: [file("${DEPS_DIR}/Debugger-rmi-trace/"), file("${DEPS_DIR}/PyGhidra/")]
 	],
 	[
-		name: "wheel-0.37.1-py2.py3-none-any.whl",
-		url: "https://files.pythonhosted.org/packages/27/d6/003e593296a85fd6ed616ed962795b2f87709c3eee2bca4f6d0fe55c6d00/wheel-0.37.1-py2.py3-none-any.whl",
-		sha256: "4bdcd7d840138086126cd09254dc6195fb4fc6f01c050a1d7236f2630db1d22a",
+		name: "wheel-0.45.1-py3-none-any.whl",
+		url: "https://files.pythonhosted.org/packages/0b/2c/87f3254fd8ffd29e4c02732eee68a83a1d3c346ae39bc6822dcbcb697f2b/wheel-0.45.1-py3-none-any.whl",
+		sha256: "708e7481cc80179af0e556bbf0cc00b8444c7321e2700b8d8580231d13017248",
 		destination: [file("${DEPS_DIR}/Debugger-rmi-trace/"), file("${DEPS_DIR}/PyGhidra/")]
 	],
 	[
