diff --git a/config.layout b/config.layout
index a29351ef14..d519be64ad 100644
--- a/config.layout
+++ b/config.layout
@@ -315,7 +315,7 @@
     cgidir:        ${prefix}/usr/lib/cgi-bin
     includedir:    ${exec_prefix}/include/apache2
     localstatedir: ${prefix}/var/lock/apache2
-    runtimedir:    ${prefix}/var/run/apache2
+    runtimedir:    ${prefix}/run/apache2
     logfiledir:    ${prefix}/var/log/apache2
     proxycachedir: ${prefix}/var/cache/apache2/proxy
     infodir:       ${exec_prefix}/share/info
