From 60619321ef24c68d9a3c65cf547ad80780e3a091 Mon Sep 17 00:00:00 2001
From: Batuhan Apaydin <batuhan.apaydin@chainguard.dev>
Date: Mon, 21 Aug 2023 19:18:59 +0300
Subject: [PATCH] disable copyWebJavadocPython

Signed-off-by: Batuhan Apaydin <batuhan.apaydin@chainguard.dev>

:100644 100644 b88a5cb 641593a M	py4j-java/build.gradle

diff --git a/py4j-java/build.gradle b/py4j-java/build.gradle
index b88a5cb..641593a 100644
--- a/py4j-java/build.gradle
+++ b/py4j-java/build.gradle
@@ -346,7 +346,7 @@ task copyJavaPython(type: Copy, dependsOn: ['copyMainJar', 'javadoc', 'createPom
 }
 
 // Python Source Dist
-task buildPythonSrc(type: Exec, dependsOn: ['copyWebJavadocPython', 'copyJavaPython']) {
+task buildPythonSrc(type: Exec, dependsOn: [ 'copyJavaPython']) {
     description 'Builds python source distribution.'
     workingDir pythonDir;
     executable 'python'
-- 
2.39.2 (Apple Git-143)

