diff --git a/Makefile b/Makefile
index b329e7a1..ad9baa59 100644
--- a/Makefile
+++ b/Makefile
@@ -147,7 +147,7 @@ output/windows_amd64/test/bin/%.exe: $(PKG_SOURCES)
 
 output/linux_amd64/bin/%: $(PKG_SOURCES)
 	GOOS=linux GOARCH=amd64 CGO_ENABLED=$(CGO_ENABLED) \
-	  CC=x86_64-linux-gnu-gcc go build \
+	  go build \
 		-o $@ \
 		-ldflags '-X $(PKG)/pkg/version.version=$(VERSION)' \
 		-tags "$(LINUX_BUILD_TAGS)" \
@@ -157,14 +157,14 @@ output/linux_amd64/bin/%: $(PKG_SOURCES)
 output/linux_amd64/test/bin/%: $(PKG_SOURCES)
 	cd test && \
 	GOOS=linux GOARCH=amd64 CGO_ENABLED=$(CGO_ENABLED) \
-	  CC=x86_64-linux-gnu-gcc go build \
+	  go build \
 		-o ../$@ \
 		-tags "$(LINUX_BUILD_TAGS)" \
 		./e2e/$(subst -,,$*)
 
 output/linux_arm64/bin/%: $(PKG_SOURCES)
 	GOOS=linux GOARCH=arm64 CGO_ENABLED=$(CGO_ENABLED) \
-	  CC=aarch64-linux-gnu-gcc go build \
+	 go build \
 		-o $@ \
 		-ldflags '-X $(PKG)/pkg/version.version=$(VERSION)' \
 		-tags "$(LINUX_BUILD_TAGS)" \
@@ -174,7 +174,7 @@ output/linux_arm64/bin/%: $(PKG_SOURCES)
 output/linux_arm64/test/bin/%: $(PKG_SOURCES)
 	cd test && \
 	GOOS=linux GOARCH=arm64 CGO_ENABLED=$(CGO_ENABLED) \
-	  CC=aarch64-linux-gnu-gcc go build \
+	 go build \
 		-o ../$@ \
 		-tags "$(LINUX_BUILD_TAGS)" \
 		./e2e/$(subst -,,$*)
