diff --git a/Makefile b/Makefile
index eb297f7..fe6e3c1 100644
--- a/Makefile
+++ b/Makefile
@@ -148,7 +148,7 @@ output/windows_amd64/test/bin/%.exe: $(PKG_SOURCES)
 
 output/linux_amd64/bin/%: $(PKG_SOURCES)
 	GOOS=linux GOARCH=amd64 CGO_ENABLED=$(CGO_ENABLED) GO111MODULE=on \
-	  CC=x86_64-linux-gnu-gcc go build \
+	  go build \
 		-mod vendor \
 		-o $@ \
 		-ldflags '-X $(PKG)/pkg/version.version=$(VERSION)' \
@@ -158,7 +158,7 @@ output/linux_amd64/bin/%: $(PKG_SOURCES)
 
 output/linux_amd64/test/bin/%: $(PKG_SOURCES)
 	GOOS=linux GOARCH=amd64 CGO_ENABLED=$(CGO_ENABLED) GO111MODULE=on \
-	  CC=x86_64-linux-gnu-gcc go build \
+	  go build \
 		-mod vendor \
 		-o $@ \
 		-tags "$(LINUX_BUILD_TAGS)" \
@@ -166,7 +166,7 @@ output/linux_amd64/test/bin/%: $(PKG_SOURCES)
 
 output/linux_arm64/bin/%: $(PKG_SOURCES)
 	GOOS=linux GOARCH=arm64 CGO_ENABLED=$(CGO_ENABLED) GO111MODULE=on \
-	  CC=aarch64-linux-gnu-gcc go build \
+	  go build \
 		-mod vendor \
 		-o $@ \
 		-ldflags '-X $(PKG)/pkg/version.version=$(VERSION)' \
@@ -176,7 +176,7 @@ output/linux_arm64/bin/%: $(PKG_SOURCES)
 
 output/linux_arm64/test/bin/%: $(PKG_SOURCES)
 	GOOS=linux GOARCH=arm64 CGO_ENABLED=$(CGO_ENABLED) GO111MODULE=on \
-	  CC=aarch64-linux-gnu-gcc go build \
+	  go build \
 		-mod vendor \
 		-o $@ \
 		-tags "$(LINUX_BUILD_TAGS)" \
