diff --git a/toolchain/internal/configure.bzl b/toolchain/internal/configure.bzl
index e0ad59e..9f7b9b9 100644
--- a/toolchain/internal/configure.bzl
+++ b/toolchain/internal/configure.bzl
@@ -369,13 +369,6 @@ def _cc_toolchain_str(
         toolchain_path_prefix + "include/{}/c++/v1".format(target_system_name),
     ]

-    # TODO(zbarsky): Not sure if these lib64 paths are actually needed for system toolchains?
-    if use_absolute_paths_llvm:
-        cxx_builtin_include_directories.extend([
-            toolchain_path_prefix + "lib64/clang/{}/include".format(llvm_version),
-            toolchain_path_prefix + "lib64/clang/{}/include".format(major_llvm_version),
-        ])
-
     sysroot_prefix = ""
     if sysroot_path:
         sysroot_prefix = "%sysroot%"
