diff --git a/toolchain/internal/configure.bzl b/toolchain/internal/configure.bzl
index 45d2055..9cfda79 100644
--- a/toolchain/internal/configure.bzl
+++ b/toolchain/internal/configure.bzl
@@ -366,13 +366,6 @@ def _cc_toolchain_str(
         toolchain_path_prefix + "include/{}/c++/v1".format(target_system_name),
     ]
 
-    # TODO(zbarsky): Not sure if these lib64 paths are actually needed for system toolchains?
-    if use_absolute_paths_llvm:
-        cxx_builtin_include_directories.extend([
-            toolchain_path_prefix + "lib64/clang/{}/include".format(llvm_version),
-            toolchain_path_prefix + "lib64/clang/{}/include".format(major_llvm_version),
-        ])
-
     sysroot_prefix = ""
     if sysroot_path:
         sysroot_prefix = "%sysroot%"
diff --git a/toolchain/internal/llvm_distributions.bzl b/toolchain/internal/llvm_distributions.bzl
index 415f1a0..6aded99 100644
--- a/toolchain/internal/llvm_distributions.bzl
+++ b/toolchain/internal/llvm_distributions.bzl
@@ -889,11 +889,13 @@ def _get_all_llvm_distributions(*, llvm_distributions, extra_llvm_distributions,
 
 _UBUNTU_NAMES = [
     "arch",
+    "chainguard",
     "linuxmint",
     "manjaro",
     "nixos",
     "pop",
     "ubuntu",
+    "wolfi",
 ]
 
 _UBUNTU_VERSIONS = [
@@ -1399,6 +1401,7 @@ def _distributions_test_writer_impl(ctx):
             struct(name = "arch", version = ANY_VERSION),
             struct(name = "centos", version = "6"),
             struct(name = "centos", version = "7"),
+            struct(name = "chainguard", version = ANY_VERSION),
             struct(name = "debian", version = "0"),
             struct(name = "debian", version = "8"),
             struct(name = "debian", version = "9"),
@@ -1432,6 +1435,7 @@ def _distributions_test_writer_impl(ctx):
             struct(name = "ubuntu", version = "20.10"),
             struct(name = "ubuntu", version = "22.04"),
             struct(name = "ubuntu", version = "24.04"),
+            struct(name = "wolfi", version = ANY_VERSION),
         ],
     }
 
