From 7bf18ce49c110a1e6135f2d0c24e95aabb0b056c Mon Sep 17 00:00:00 2001
From: Srishti Hegde <srishti.hegde@chainguard.dev>
Date: Tue, 30 Apr 2024 16:16:09 -0700
Subject: [PATCH] version requirements

---
 setup.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index 1ecf23f284..f91ec6365a 100644
--- a/setup.py
+++ b/setup.py
@@ -93,7 +93,7 @@ setup(
         "flask-wtf>=1.1.0, <2.0",
         "func_timeout",
         "geopy",
-        "gunicorn>=21.2.0, <22.0; sys_platform != 'win32'",
+        "gunicorn>=21.2.0",
         "hashids>=1.3.1, <2",
         "holidays>=0.25, <0.26",
         "humanize",
@@ -127,7 +127,7 @@ setup(
         "sqlalchemy>=1.4, <2",
         "sqlalchemy-utils>=0.38.3, <0.39",
         "sqlglot>=23.0.2,<24",
-        "sqlparse>=0.4.4, <0.5",
+        "sqlparse>=0.4.4",
         "tabulate>=0.8.9, <0.9",
         "typing-extensions>=4, <5",
         "waitress; sys_platform == 'win32'",
-- 
2.39.3 (Apple Git-146)

