diff --git a/storage/myisam/myisamchk.c b/storage/myisam/myisamchk.c
index 956bcda536e..80f7e8c9054 100644
--- a/storage/myisam/myisamchk.c
+++ b/storage/myisam/myisamchk.c
@@ -698,8 +698,6 @@ get_one_option(const struct my_option *opt,
     break;
   case 'V':
     print_version();
-    free_defaults(default_argv);
-    my_end(MY_CHECK_ERROR);
     my_exit(0);
   case OPT_CORRECT_CHECKSUM:
     if (argument == disabled_my_option)
