diff --git a/scripts/build.sh b/scripts/build.sh
index 40b4f4d8..d796e964 100755
--- a/scripts/build.sh
+++ b/scripts/build.sh
@@ -13,7 +13,7 @@ fi
 # Install node modules per package-lock.json
 yarn
 (cd $SQLPAD_CLIENT_DIR && yarn)
-(cd $SQLPAD_SERVER_DIR && yarn)
+(cd $SQLPAD_SERVER_DIR && yarn --production)
 
 # Build front-end
 (cd $SQLPAD_CLIENT_DIR && yarn build)
@@ -24,4 +24,4 @@ mkdir ${SQLPAD_SERVER_DIR}/public
 cp -r ${SQLPAD_CLIENT_DIR}/build/* ${SQLPAD_SERVER_DIR}/public
 
 # Build test deb for test cases and to use during dev
-node ${SQLPAD_SERVER_DIR}/generate-test-db-fixture.js
\ No newline at end of file
+node ${SQLPAD_SERVER_DIR}/generate-test-db-fixture.js
