diff --git a/client/src/common/IconButton.tsx b/client/src/common/IconButton.tsx
index 62ce444d..a82e5374 100644
--- a/client/src/common/IconButton.tsx
+++ b/client/src/common/IconButton.tsx
@@ -60,7 +60,6 @@ const IconButton = React.forwardRef<Ref, ButtonProps & LinkProps>(
     if (to && !disabled && !onClick) {
       button = (
         <div style={{ display: 'inline', lineHeight: 'initial' }}>
-          {/* @ts-expect-error anchor and buttons types are fighting */}
           <Link to={to} className={classNames.join(' ')} {...rest}>
             {React.Children.map(children, (child) => {
               if (React.isValidElement(child)) {
