--- /usr/lib/spark/sbin/spark-daemon.sh
+++ /usr/lib/spark/sbin/spark-daemon.sh
@@ -132,7 +132,7 @@ fi

 execute_command() {
   if [ -z ${SPARK_NO_DAEMONIZE+set} ]; then
-      nohup -- "$@" >> $log 2>&1 < /dev/null &
+      /usr/bin/nohup -- "$@" >> $log 2>&1 < /dev/null &
       newpid="$!"

       echo "$newpid" > "$pid"
@@ -140,7 +140,7 @@ execute_command() {
       # Poll for up to 5 seconds for the java process to start
       for i in {1..10}
       do
-        if [[ $(ps -p "$newpid" -o comm=) =~ "java" ]]; then
+        if [[ $(/usr/bin/ps -p "$newpid" -o comm=) =~ "java" ]]; then
            break
         fi
         sleep 0.5
@@ -148,7 +148,7 @@ execute_command() {

       sleep 2
       # Check if the process has died; in that case we'll tail the log so the user can see
-      if [[ ! $(ps -p "$newpid" -o comm=) =~ "java" ]]; then
+      if [[ ! $(/usr/bin/ps -p "$newpid" -o comm=) =~ "java" ]]; then
         echo "failed to launch: $@"
         tail -10 "$log" | sed 's/^/  /'
         echo "full log in $log"
@@ -166,7 +166,7 @@ run_command() {

   if [ -f "$pid" ]; then
     TARGET_ID="$(cat "$pid")"
-    if [[ $(ps -p "$TARGET_ID" -o comm=) =~ "java" ]]; then
+    if [[ $(/usr/bin/ps -p "$TARGET_ID" -o comm=) =~ "java" ]]; then
       echo "$command running as process $TARGET_ID.  Stop it first."
       exit 1
     fi
@@ -211,7 +211,7 @@ case $option in

     if [ -f $pid ]; then
       TARGET_ID="$(cat "$pid")"
-      if [[ $(ps -p "$TARGET_ID" -o comm=) =~ "java" ]]; then
+      if [[ $(/usr/bin/ps -p "$TARGET_ID" -o comm=) =~ "java" ]]; then
         echo "stopping $command"
         kill "$TARGET_ID" && rm -f "$pid"
       else
@@ -226,7 +226,7 @@ case $option in

     if [ -f $pid ]; then
       TARGET_ID="$(cat "$pid")"
-      if [[ $(ps -p "$TARGET_ID" -o comm=) =~ "java" ]]; then
+      if [[ $(/usr/bin/ps -p "$TARGET_ID" -o comm=) =~ "java" ]]; then
         echo "decommissioning $command"
         kill -s SIGPWR "$TARGET_ID"
       else
@@ -241,7 +241,7 @@ case $option in

     if [ -f $pid ]; then
       TARGET_ID="$(cat "$pid")"
-      if [[ $(ps -p "$TARGET_ID" -o comm=) =~ "java" ]]; then
+      if [[ $(/usr/bin/ps -p "$TARGET_ID" -o comm=) =~ "java" ]]; then
         echo $command is running.
         exit 0
       else