diff --git a/lib_pypy/_sysconfigdata.py b/lib_pypy/_sysconfigdata.py
index 77b05ecf41..08f1e44df8 100644
--- a/lib_pypy/_sysconfigdata.py
+++ b/lib_pypy/_sysconfigdata.py
@@ -37,6 +37,7 @@ build_time_vars = {
     'Py_DEBUG': 0,  # cpyext never uses this
     'Py_ENABLE_SHARED': 0,  # if 1, will add python so to link like -lpython3.7
     'SIZEOF_VOID_P': struct.calcsize("P"),
+    'WHEEL_PKG_DIR': "/usr/share/python-wheels",
 }
 
 # LIBDIR should point to where the libpypy3.9-c.so file lives, on CPython
