From dd714f1b72aa8c341e5a27aa9e968b4ecdaf1abb Mon Sep 17 00:00:00 2001
From: Cedric Nugteren <web@cedricnugteren.nl>
Date: Fri, 27 Jun 2025 08:37:11 +0200
Subject: [PATCH] Fixed compatibility with CMake 4.0 (#575)

Origin: backport, https://github.com/CNugteren/CLBlast/commit/dd714f1b72aa8c341e5a27aa9e968b4ecdaf1abb

---
 CHANGELOG      | 1 +
 CMakeLists.txt | 3 ++-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4a4b92c2..d218d813 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,7 +9,8 @@
 #
 # ==================================================================================================
 
-cmake_minimum_required(VERSION 2.8.11)
+# Technically the minimum is 2.8.11 but to be compatible with CMake 4 this is set to 3.10
+cmake_minimum_required(VERSION 3.10)
 
 # Overrides for MSVC static runtime
 option(OVERRIDE_MSVC_FLAGS_TO_MT "Override compiler flags for MSVC to build with a static runtime (/MT instead of /MD)" ON)
