diff --git a/logstash-core/build.gradle b/logstash-core/build.gradle
index 14dd90e81..f41d2be77 100644
--- a/logstash-core/build.gradle
+++ b/logstash-core/build.gradle
@@ -245,6 +245,7 @@ dependencies {
         exclude group: 'com.google.guava', module: 'guava'
     }
     implementation 'org.javassist:javassist:3.30.2-GA'
+    implementation 'io.netty:netty-codec:4.1.125.Final'
     testImplementation "org.apache.logging.log4j:log4j-core:${log4jVersion}:tests"
     testImplementation 'org.hamcrest:hamcrest:2.2'
     testImplementation 'org.hamcrest:hamcrest-library:2.2'
