diff --git a/src/cpp/session/CMakeLists.txt b/src/cpp/session/CMakeLists.txt
index 571b3a1..c1efd8f 100644
--- a/src/cpp/session/CMakeLists.txt
+++ b/src/cpp/session/CMakeLists.txt
@@ -716,10 +716,10 @@ if(NOT RSTUDIO_SESSION_WIN32 AND NOT RSESSION_ALTERNATE_BUILD)
            DESTINATION "${RSTUDIO_INSTALL_SUPPORTING}/resources")
 
    # install node
-   install(
-      DIRECTORY "${RSTUDIO_DEPENDENCIES_DIR}/common/node/${RSTUDIO_INSTALLED_NODE_VERSION}-patched/"
-      DESTINATION "${RSTUDIO_INSTALL_BIN}/node"
-      USE_SOURCE_PERMISSIONS)
+   #install(
+   #   DIRECTORY "${RSTUDIO_DEPENDENCIES_DIR}/common/node/${RSTUDIO_INSTALLED_NODE_VERSION}-patched/"
+   #   DESTINATION "${RSTUDIO_INSTALL_BIN}/node"
+   #   USE_SOURCE_PERMISSIONS)
 
    # install quarto (or pandoc if quarto disabled)
    if(QUARTO_ENABLED)
