diff --git a/cmake/globals.cmake b/cmake/globals.cmake
index 3601ee47b9..a213c9655d 100644
--- a/cmake/globals.cmake
+++ b/cmake/globals.cmake
@@ -237,7 +237,7 @@ set(PANDOC_VERSION "3.2" CACHE INTERNAL "Pandoc version")
 set(RSTUDIO_NODE_VERSION "22.13.1" CACHE INTERNAL "Node version for building")
 
 # node version installed with the product
-set(RSTUDIO_INSTALLED_NODE_VERSION "22.21.1" CACHE INTERNAL "Node version installed with product")
+set(RSTUDIO_INSTALLED_NODE_VERSION "22.22.0" CACHE INTERNAL "Node version installed with product")
 
 # quarto support
 
diff --git a/dependencies/tools/rstudio-tools.cmd b/dependencies/tools/rstudio-tools.cmd
index 339d42c71e..aef13da41e 100644
--- a/dependencies/tools/rstudio-tools.cmd
+++ b/dependencies/tools/rstudio-tools.cmd
@@ -57,7 +57,7 @@ set WIN32_SDK_VERSION=1.0.20348.0
 set RSTUDIO_NODE_VERSION=22.13.1
 
 :: Node version installed with the product
-set RSTUDIO_INSTALLED_NODE_VERSION=22.21.1
+set RSTUDIO_INSTALLED_NODE_VERSION=22.22.0
 
 :: The base URL where RStudio build tools are available for download.
 set RSTUDIO_BUILDTOOLS=https://rstudio-buildtools.s3.amazonaws.com
diff --git a/dependencies/tools/rstudio-tools.sh b/dependencies/tools/rstudio-tools.sh
index 43f7a4ce88..79ab0000ec 100755
--- a/dependencies/tools/rstudio-tools.sh
+++ b/dependencies/tools/rstudio-tools.sh
@@ -67,7 +67,7 @@ export RSTUDIO_NODE_VERSION="22.13.1"
 #
 # In addition to updating the version here, search the entire repo for other instances of
 # RSTUDIO_INSTALLED_NODE_VERSION and update to match.
-export RSTUDIO_INSTALLED_NODE_VERSION="22.21.1"
+export RSTUDIO_INSTALLED_NODE_VERSION="22.22.0"
 
 # actual directory name of the node installation used
 # mainly relevant for cases like macOS, where we have an -arm64 variant installed
diff --git a/dependencies/tools/upload-node.sh b/dependencies/tools/upload-node.sh
index c28a55f76a..b8964c93c3 100755
--- a/dependencies/tools/upload-node.sh
+++ b/dependencies/tools/upload-node.sh
@@ -10,7 +10,7 @@
 # installed with the RStudio IDE by RSTUDIO_INSTALLED_NODE_VERSION.
 
 # Modify to set the node.js version to upload
-NODE_VERSION="v22.21.1"
+NODE_VERSION="v22.22.0"
 
 # Check that we're logged in with AWS
 aws sts get-caller-identity || aws sso login
diff --git a/version/news/NEWS-2026.01.0-apple-blossom.md b/version/news/NEWS-2026.01.0-apple-blossom.md
index 625958e1eb..dec0f0eaa0 100644
--- a/version/news/NEWS-2026.01.0-apple-blossom.md
+++ b/version/news/NEWS-2026.01.0-apple-blossom.md
@@ -85,7 +85,7 @@
 ### Dependencies
 - Copilot Language Server 1.399.0
 - Electron 38.7.2
-- Node.js 22.21.1 (copilot completions)
+- Node.js 22.22.0 (copilot completions)
 - Node.js 22.20.0 (vscode server)
 - Quarto 1.8.25
 - Launcher 2.21.0
