diff --git a/src/cpp/server/CMakeLists.txt b/src/cpp/server/CMakeLists.txt
index dda1851..ec2d16a 100644
--- a/src/cpp/server/CMakeLists.txt
+++ b/src/cpp/server/CMakeLists.txt
@@ -251,10 +251,10 @@ if (UNIX AND NOT APPLE)
            DESTINATION ${RSERVER_SYSTEMD_DIR})

    # install node
-   install(
-      DIRECTORY "${RSTUDIO_DEPENDENCIES_DIR}/common/node/${RSTUDIO_NODE_VERSION}/"
-      DESTINATION "${RSTUDIO_INSTALL_BIN}/node"
-      USE_SOURCE_PERMISSIONS)
+   #install(
+    #  DIRECTORY "${RSTUDIO_DEPENDENCIES_DIR}/common/node/${RSTUDIO_NODE_VERSION}/"
+     # DESTINATION "${RSTUDIO_INSTALL_BIN}/node"
+      #USE_SOURCE_PERMISSIONS)

 elseif(APPLE)
