diff --git a/pkg/runtimes/docker/container.go b/pkg/runtimes/docker/container.go
index 98910a4..7f0d724 100644
--- a/pkg/runtimes/docker/container.go
+++ b/pkg/runtimes/docker/container.go
@@ -27,11 +27,11 @@ import (
 	"fmt"
 	"io"
 
-	"github.com/docker/docker/api/types"
-	"github.com/docker/docker/api/types/container"
+	"github.com/moby/moby/api/types"
+	"github.com/moby/moby/api/types/container"
 	"github.com/docker/docker/api/types/filters"
-	dockerimage "github.com/docker/docker/api/types/image"
-	"github.com/docker/docker/client"
+	dockerimage "github.com/moby/moby/api/types/image"
+	"github.com/moby/moby/client"
 	l "github.com/k3d-io/k3d/v5/pkg/logger"
 	k3d "github.com/k3d-io/k3d/v5/pkg/types"
 	"github.com/sirupsen/logrus"
diff --git a/pkg/runtimes/docker/image.go b/pkg/runtimes/docker/image.go
index beb86b7..e6aca3a 100644
--- a/pkg/runtimes/docker/image.go
+++ b/pkg/runtimes/docker/image.go
@@ -25,7 +25,7 @@ import (
 	"context"
 	"fmt"
 
-	"github.com/docker/docker/api/types/image"
+	"github.com/moby/moby/api/types/image"
 )
 
 // GetImages returns a list of images present in the runtime
diff --git a/pkg/runtimes/docker/network.go b/pkg/runtimes/docker/network.go
index 567dce6..092f485 100644
--- a/pkg/runtimes/docker/network.go
+++ b/pkg/runtimes/docker/network.go
@@ -28,7 +28,7 @@ import (
 	"strings"
 
 	"github.com/docker/docker/api/types/filters"
-	"github.com/docker/docker/api/types/network"
+	"github.com/moby/moby/api/types/network"
 
 	l "github.com/k3d-io/k3d/v5/pkg/logger"
 	runtimeErr "github.com/k3d-io/k3d/v5/pkg/runtimes/errors"
diff --git a/pkg/runtimes/docker/node.go b/pkg/runtimes/docker/node.go
index 5de0e45..4c3339a 100644
--- a/pkg/runtimes/docker/node.go
+++ b/pkg/runtimes/docker/node.go
@@ -31,8 +31,8 @@ import (
 	"io"
 	"time"
 
-	"github.com/docker/docker/api/types"
-	"github.com/docker/docker/api/types/container"
+	"github.com/moby/moby/api/types"
+	"github.com/moby/moby/api/types/container"
 	"github.com/docker/docker/api/types/filters"
 	l "github.com/k3d-io/k3d/v5/pkg/logger"
 	runtimeErr "github.com/k3d-io/k3d/v5/pkg/runtimes/errors"
diff --git a/pkg/runtimes/docker/translate.go b/pkg/runtimes/docker/translate.go
index 9ef825e..98b2148 100644
--- a/pkg/runtimes/docker/translate.go
+++ b/pkg/runtimes/docker/translate.go
@@ -30,9 +30,9 @@ import (
 	"strconv"
 	"strings"
 
-	"github.com/docker/docker/api/types"
-	docker "github.com/docker/docker/api/types/container"
-	"github.com/docker/docker/api/types/network"
+	"github.com/moby/moby/api/types"
+	docker "github.com/moby/moby/api/types/container"
+	"github.com/moby/moby/api/types/network"
 	"github.com/docker/go-connections/nat"
 	l "github.com/k3d-io/k3d/v5/pkg/logger"
 	runtimeErr "github.com/k3d-io/k3d/v5/pkg/runtimes/errors"
diff --git a/pkg/runtimes/docker/types.go b/pkg/runtimes/docker/types.go
index e80113e..608e166 100644
--- a/pkg/runtimes/docker/types.go
+++ b/pkg/runtimes/docker/types.go
@@ -22,8 +22,8 @@ THE SOFTWARE.
 package docker
 
 import (
-	"github.com/docker/docker/api/types/container"
-	"github.com/docker/docker/api/types/network"
+	"github.com/moby/moby/api/types/container"
+	"github.com/moby/moby/api/types/network"
 )
 
 // NodeInDocker represents everything that we need to represent a k3d node in docker
diff --git a/pkg/runtimes/docker/util.go b/pkg/runtimes/docker/util.go
index 5ece578..ef7ceb5 100644
--- a/pkg/runtimes/docker/util.go
+++ b/pkg/runtimes/docker/util.go
@@ -31,12 +31,12 @@ import (
 	"regexp"
 	"strings"
 
-	"github.com/docker/docker/api/types/container"
+	"github.com/moby/moby/api/types/container"
 
 	"github.com/docker/cli/cli/command"
 	"github.com/docker/cli/cli/flags"
 	"github.com/docker/docker/api/types/filters"
-	"github.com/docker/docker/client"
+	"github.com/moby/moby/client"
 	"github.com/docker/docker/pkg/archive"
 	l "github.com/k3d-io/k3d/v5/pkg/logger"
 	runtimeErrors "github.com/k3d-io/k3d/v5/pkg/runtimes/errors"
diff --git a/pkg/runtimes/docker/volume.go b/pkg/runtimes/docker/volume.go
index 04dfcbb..bc8c7e5 100644
--- a/pkg/runtimes/docker/volume.go
+++ b/pkg/runtimes/docker/volume.go
@@ -26,7 +26,7 @@ import (
 	"fmt"
 
 	"github.com/docker/docker/api/types/filters"
-	"github.com/docker/docker/api/types/volume"
+	"github.com/moby/moby/api/types/volume"
 	runtimeErrors "github.com/k3d-io/k3d/v5/pkg/runtimes/errors"
 	k3d "github.com/k3d-io/k3d/v5/pkg/types"
 )
