diff --git a/pkg/runtimes/docker/util.go b/pkg/runtimes/docker/util.go
index 5ece578..85efb5a 100644
--- a/pkg/runtimes/docker/util.go
+++ b/pkg/runtimes/docker/util.go
@@ -33,8 +33,6 @@ import (
 
 	"github.com/docker/docker/api/types/container"
 
-	"github.com/docker/cli/cli/command"
-	"github.com/docker/cli/cli/flags"
 	"github.com/docker/docker/api/types/filters"
 	"github.com/docker/docker/client"
 	"github.com/docker/docker/pkg/archive"
@@ -42,7 +40,6 @@ import (
 	runtimeErrors "github.com/k3d-io/k3d/v5/pkg/runtimes/errors"
 	k3d "github.com/k3d-io/k3d/v5/pkg/types"
 	"github.com/pkg/errors"
-	"github.com/spf13/pflag"
 )
 
 func IsDockerDesktop(os string) bool {
@@ -182,24 +179,7 @@ func (d Docker) ReadFromNode(ctx context.Context, path string, node *k3d.Node) (
 
 // GetDockerClient returns a docker client
 func GetDockerClient() (client.APIClient, error) {
-	dockerCli, err := command.NewDockerCli(command.WithStandardStreams())
-	if err != nil {
-		return nil, fmt.Errorf("failed to create new docker CLI with standard streams: %w", err)
-	}
-
-	newClientOpts := flags.NewClientOptions()
-	newClientOpts.LogLevel = l.Log().GetLevel().String() // this is needed, as the following Initialize() call will set a new log level on the global logrus instance
-
-	flagset := pflag.NewFlagSet("docker", pflag.ContinueOnError)
-	newClientOpts.InstallFlags(flagset)
-	newClientOpts.SetDefaultOptions(flagset)
-
-	err = dockerCli.Initialize(newClientOpts)
-	if err != nil {
-		return nil, fmt.Errorf("failed to initialize docker CLI: %w", err)
-	}
-
-	return dockerCli.Client(), nil
+	return client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
 }
 
 // isAttachedToNetwork return true if node is attached to network
