diff --git a/project/VersionUtil.scala b/project/VersionUtil.scala
index f177278ee5..a7b96aad3c 100644
--- a/project/VersionUtil.scala
+++ b/project/VersionUtil.scala
@@ -3,7 +3,7 @@ package scala.build
 import sbt._
 import Keys._
 
-import java.util.{Date, Locale, Properties, TimeZone}
+import java.util.{Calendar, Date, Locale, Properties, TimeZone}
 import java.io.{File, FileInputStream, StringWriter}
 import java.text.SimpleDateFormat
 import java.time.Instant
@@ -78,7 +78,7 @@ object VersionUtil {
             // Workaround lack of git worktree support in JGit https://bugs.eclipse.org/bugs/show_bug.cgi?id=477475
             val sha = List("git", "rev-parse", "HEAD").!!.trim
             val commitDateIso = List("git", "log", "-1", "--format=%cI", "HEAD").!!.trim
-            val date = Date.from(ISO_DATE_TIME.parse(commitDateIso, Instant.from _))
+            val date = Date.from(ISO_DATE_TIME.parse("1970-01-01T00:00:00Z", Instant.from _))
             (date, sha.substring(0, 7))
           } catch {
             case ex: Exception =>