From 0d34201b0ac11c35681a580cfac80f4c901cda44 Mon Sep 17 00:00:00 2001
From: uti <uti.edun@chainguard.dev>
Date: Thu, 29 May 2025 18:29:46 +0000
Subject: [PATCH] remove unused library dependency  from krb5 build
 configuration

---
 bazel/thirdparty/krb5.BUILD | 1 -
 1 file changed, 1 deletion(-)

diff --git a/bazel/thirdparty/krb5.BUILD b/bazel/thirdparty/krb5.BUILD
index c7aa78b1bc..a899643725 100644
--- a/bazel/thirdparty/krb5.BUILD
+++ b/bazel/thirdparty/krb5.BUILD
@@ -57,7 +57,6 @@ configure_make(
     },
     lib_source = ":srcs",
     out_shared_libs = [
-        "libcom_err.so.3",
         "libgssapi_krb5.so.2",
         "libk5crypto.so.3",
         "libkrb5.so.3",
-- 
2.49.0

