diff --git a/src/openssl.c b/src/openssl.c
index a9708753..af62235b 100644
--- a/src/openssl.c
+++ b/src/openssl.c
@@ -219,75 +219,21 @@ ssl_init (void)
 
   switch (opt.secure_protocol)
     {
-    case secure_protocol_sslv2:
-#if !defined OPENSSL_NO_SSL2 && OPENSSL_VERSION_NUMBER < 0x10100000L
-      meth = SSLv2_client_method ();
-#endif
-      break;
-
-    case secure_protocol_sslv3:
-#ifndef OPENSSL_NO_SSL3_METHOD
-      meth = SSLv3_client_method ();
-#endif
-      break;
-
     case secure_protocol_auto:
     case secure_protocol_pfs:
-      meth = SSLv23_client_method ();
-      ssl_options |= SSL_OP_NO_SSLv2 | SSL_OP_NO_SSLv3;
-      break;
-    case secure_protocol_tlsv1:
-#if !defined(LIBRESSL_VERSION_NUMBER) && (OPENSSL_VERSION_NUMBER >= 0x10100000L)
       meth = TLS_client_method();
-      ssl_proto_version = TLS1_VERSION;
-#else
-      meth = TLSv1_client_method ();
-#endif
-      break;
-
-#if OPENSSL_VERSION_NUMBER >= 0x10001000
-    case secure_protocol_tlsv1_1:
-#if !defined(LIBRESSL_VERSION_NUMBER) && (OPENSSL_VERSION_NUMBER >= 0x10100000L)
-      meth = TLS_client_method();
-      ssl_proto_version = TLS1_1_VERSION;
-#else
-      meth = TLSv1_1_client_method ();
-#endif
+      ssl_options |= SSL_OP_NO_SSLv2 | SSL_OP_NO_SSLv3;
       break;
 
     case secure_protocol_tlsv1_2:
-#if !defined(LIBRESSL_VERSION_NUMBER) && (OPENSSL_VERSION_NUMBER >= 0x10100000L)
       meth = TLS_client_method();
       ssl_proto_version = TLS1_2_VERSION;
-#else
-      meth = TLSv1_2_client_method ();
-#endif
       break;
 
     case secure_protocol_tlsv1_3:
-#if !defined(LIBRESSL_VERSION_NUMBER) && (OPENSSL_VERSION_NUMBER >= 0x10100000L) && defined TLS1_3_VERSION
       meth = TLS_client_method();
       ssl_proto_version = TLS1_3_VERSION;
-#else
-      logprintf (LOG_NOTQUIET, _("Your OpenSSL version is too old to support TLS 1.3\n"));
-      goto error;
-#endif
       break;
-#else
-    case secure_protocol_tlsv1_1:
-      logprintf (LOG_NOTQUIET, _("Your OpenSSL version is too old to support TLSv1.1\n"));
-      goto error;
-
-    case secure_protocol_tlsv1_2:
-      logprintf (LOG_NOTQUIET, _("Your OpenSSL version is too old to support TLSv1.2\n"));
-      goto error;
-
-#endif
-
-    default:
-      logprintf (LOG_NOTQUIET, _("OpenSSL: unimplemented 'secure-protocol' option value %d\n"), opt.secure_protocol);
-      logprintf (LOG_NOTQUIET, _("Please report this issue to bug-wget@gnu.org\n"));
-      abort ();
     }
 
   if (!meth)
