--- a/pkg/scraper/client/resource/decode.go	2026-05-20 16:39:33.796867027 -0400
+++ b/pkg/scraper/client/resource/decode.go	2026-05-20 16:41:21.123961246 -0400
@@ -44,7 +44,7 @@
 	}
 	node := &storage.MetricsPoint{}
 	pods := make(map[apitypes.NamespacedName]storage.PodMetricsPoint)
-	parser, err := textparse.New(b, "text/plain", "", false, false, nil)
+	parser, err := textparse.New(b, "text/plain", nil, textparse.ParserOptions{})
 	if err != nil {
 		return nil, fmt.Errorf("failed to initialize Prometheus parser: %w", err)
 	}
