From: Debasish Biswas (debasish.biswas@chainguard.dev)
Date: 2024-07-05 07:34:04 +0000
Subject: use newer libassuan

check if it works with the new Lib version

--- gnupg-2.4.8.orig/configure
+++ gnupg-2.4.8/configure
@@ -2921,8 +2921,8 @@ NEED_GPGRT_VERSION=1.46
 NEED_LIBGCRYPT_API=1
 NEED_LIBGCRYPT_VERSION=1.9.1
 
-NEED_LIBASSUAN_API=2
-NEED_LIBASSUAN_VERSION=2.5.0
+NEED_LIBASSUAN_API=3
+NEED_LIBASSUAN_VERSION=3.0.0
 
 NEED_KSBA_API=1
 NEED_KSBA_VERSION=1.6.3
