# remove the signing key requirement in gemspec
---
diff --git a/async.gemspec b/async.gemspec
index ded8f15..4a77dd5 100644
--- a/async.gemspec
+++ b/async.gemspec
@@ -11,7 +11,6 @@ Gem::Specification.new do |spec|
 	spec.license = "MIT"
 	
 	spec.cert_chain  = ['release.cert']
-	spec.signing_key = File.expand_path('~/.gem/release.pem')
 	
 	spec.homepage = "https://github.com/socketry/async"
 	
