--- gsm-1.0-pl12/inc/private.h.64bit	2006-04-26 21:17:01.000000000 +0200
+++ gsm-1.0-pl12/inc/private.h	2007-05-11 00:25:23.000000000 +0200
@@ -10,10 +10,10 @@
 #define	PRIVATE_H
 
 typedef short			word;		/* 16 bit signed int	*/
-typedef long			longword;	/* 32 bit signed int	*/
+typedef int			longword;	/* 32 bit signed int	*/
 
 typedef unsigned short		uword;		/* unsigned word	*/
-typedef unsigned long		ulongword;	/* unsigned longword	*/
+typedef unsigned int		ulongword;	/* unsigned longword	*/
 
 struct gsm_state {
 