diff --git a/awx/settings/defaults.py b/awx/settings/defaults.py
index 546f68f50d..f4f24d954b 100644
--- a/awx/settings/defaults.py
+++ b/awx/settings/defaults.py
@@ -20,6 +20,9 @@ SQL_DEBUG = DEBUG
 # Build paths inside the project like this: os.path.join(BASE_DIR, ...)
 BASE_DIR = os.path.dirname(os.path.dirname(__file__))
 
+# Path to the base virtual environment for AWX
+BASE_VENV_PATH = '/var/lib/awx/venv'
+
 # FIXME: it would be nice to cycle back around and allow this to be
 # BigAutoField going forward, but we'd have to be explicit about our
 # existing models.
