diff --git a/setup.py b/setup.py
index 9727ac2..a7a0f7b 100644
--- a/setup.py
+++ b/setup.py
@@ -50,13 +50,13 @@ setuptools.setup(
         'ffwd>=0.0.2',
         'lockfile>=0.12.2',
         'pyOpenSSL==22.0.0',
-        'cryptography<=35.0,>=2.5',
+        'cryptography<=41.0.6,>=2.5',
         'pycryptodome>=3.9.9',
         'retrying>=1.3.3',
         'parallel-ssh==2.2.0',
         'ssh2-python==1.0.0',
         'ssh-python>=0.8.0',
-        'requests==2.22.0',
+        'requests==2.31.0',
         'protobuf==4.24.3',
         'grpcio==1.58.0',
         'grpcio-health-checking==1.58.0',
@@ -68,7 +68,7 @@ setuptools.setup(
         'botocore>=1.13.27',
         'dnspython>=2.2.1',
         'asyncio==3.4.3',
-        'aiohttp==3.8.5',
+        'aiohttp==3.9.0',
         'boto3>=1.28.38',
         'gcloud-aio-storage==8.3.0',
         'azure-core==1.29.4',
