Description: Fix issue with assert call?
Author: James Page <james.page@chainguard.dev>

diff --git a/libmamba/src/util/encoding.cpp b/libmamba/src/util/encoding.cpp
index 08033f3..b32a7aa 100644
--- a/libmamba/src/util/encoding.cpp
+++ b/libmamba/src/util/encoding.cpp
@@ -5,6 +5,7 @@
 // The full license is in the file LICENSE, distributed with this software.
 
 #include <array>
+#include <cassert>
 #include <cstdint>
 #include <cstring>
 #include <ranges>

