diff --git a/libmamba/src/download/curl.cpp b/libmamba/src/download/curl.cpp
index 4c5e50b0..dda8c97d 100644
--- a/libmamba/src/download/curl.cpp
+++ b/libmamba/src/download/curl.cpp
@@ -250,6 +250,7 @@ namespace mamba::download
     {
         rhs.m_handle = nullptr;
         rhs.p_headers = nullptr;
+        std::fill(m_errorbuffer.begin(), m_errorbuffer.end(), '\0');
         std::swap(m_errorbuffer, rhs.m_errorbuffer);
         set_opt(CURLOPT_ERRORBUFFER, m_errorbuffer.data());
     }
