diff --git a/release/platform/platform.go b/release/platform/platform.go
index df1712d..9fed41f 100644
--- a/release/platform/platform.go
+++ b/release/platform/platform.go
@@ -548,6 +548,14 @@ var platforms = []Platform{
 		Repos:     []Repo{RepoEnterprise, RepoOrg},
 		BuildTags: []string{"failpoints", "ssl"},
 	},
+	{
+		Name:      "ubuntu1804",
+		Arch:      ArchAarch64,
+		OS:        OSLinux,
+		Pkg:       PkgDeb,
+		Repos:     []Repo{RepoEnterprise, RepoOrg},
+		BuildTags: []string{"failpoints", "ssl"},
+	},
 	{
 		Name:      "ubuntu1804",
 		Arch:      ArchX86_64,
