--- ./Makefile.orig
+++ ./Makefile
@@ -1,13 +1,37 @@
-all: linenoise-example linenoise-test
+MAJOR_VERSION = 0
+EXTRA_VERSION = .0.0
+SONAME = liblinenoise.so.$(MAJOR_VERSION)
+LIB = $(SONAME)$(EXTRA_VERSION)
+
+export CFLAGS ?= -Os -g
+CFLAGS += -Wall -fpic -DUSE_UTF8
+LIBDIR ?= /usr/lib
+INCLUDEDIR ?= /usr/include
+
+.PHONY: all
+all: $(LIB) linenoise-example linenoise-test
+
+$(LIB): linenoise.o
+	$(CC) $(CFLAGS) -shared -Wl,-soname,$(SONAME) $(LDFLAGS) -o $@ $^

 linenoise-example: linenoise.h linenoise.c example.c
 	$(CC) -Wall -W -Os -g -o linenoise-example linenoise.c example.c

-linenoise-test: linenoise-test.c linenoise-example
+linenoise-test: linenoise-test.c
 	$(CC) -Wall -W -Os -g -o linenoise-test linenoise-test.c

+.PHONY: test
 test: linenoise-test linenoise-example
 	./linenoise-test

+.PHONY: clean
 clean:
 	rm -f linenoise-example linenoise-test
+	rm -f $(LIB) *.o
+
+.PHONY: install
+install: $(LIB)
+	install -m 0755 -d $(DESTDIR)$(INCLUDEDIR)
+	install -m 0644 linenoise.h $(DESTDIR)$(INCLUDEDIR)
+	install -m 0755 -d $(DESTDIR)$(LIBDIR)
+	install -m 0755 $(LIB) $(DESTDIR)$(LIBDIR)
+	ldconfig -n $(DESTDIR)$(LIBDIR)
+	ln -s $(LIB) $(DESTDIR)$(LIBDIR)/liblinenoise.so
