--- src/modules/algorithms/enigma.h.orig	2002-03-09 21:17:08.000000000 +0100
+++ src/modules/algorithms/enigma.h	2019-04-07 13:54:03.000000000 +0200
@@ -3,11 +3,11 @@
 #define MASK 0377
 
 typedef struct crypt_key {
-	char t1[ROTORSZ];
-	char t2[ROTORSZ];
-	char t3[ROTORSZ];
-	char deck[ROTORSZ];
-	char cbuf[13];
+	signed char t1[ROTORSZ];
+	signed char t2[ROTORSZ];
+	signed char t3[ROTORSZ];
+	signed char deck[ROTORSZ];
+	signed char cbuf[13];
 	int n1, n2, nr1, nr2;
 } CRYPT_KEY;
