--- a/src/vmod_bodyaccess.c
+++ b/src/vmod_bodyaccess.c
@@ -33,7 +33,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 
-#include <cache/cache_int.h>
+#include <cache/cache.h>
 #include <vcl.h>
 
 #include "vre.h"
--- a/src/vmod_xkey.c
+++ b/src/vmod_xkey.c
@@ -33,7 +33,7 @@
 #include <stdlib.h>
 #include <string.h>
 
-#include <cache/cache_int.h>
+#include <cache/cache_varnishd.h>
 
 #include "vsha256.h"
 #include "vtree.h"
