diff --git a/admin/webapp/websrc/app/routes/components/global-notifications/global-notifications.component.ts b/admin/webapp/websrc/app/routes/components/global-notifications/global-notifications.component.ts
index 0ff2bc82..79beeca0 100644
--- a/admin/webapp/websrc/app/routes/components/global-notifications/global-notifications.component.ts
+++ b/admin/webapp/websrc/app/routes/components/global-notifications/global-notifications.component.ts
@@ -36,6 +36,7 @@ export class GlobalNotificationsComponent implements OnInit {
   telemetryStatus!: TelemetryStatus | null;
   unUpdateDays!: number;
   get isVersionMismatch() {
+    /*
     return GlobalVariable.summary.component_versions
       ? (GlobalVariable.summary.component_versions.length > 1 &&
           GlobalVariable.summary.component_versions[0] !==
@@ -45,6 +46,8 @@ export class GlobalNotificationsComponent implements OnInit {
               ? GlobalVariable.summary.component_versions[0].substring(1)
               : GlobalVariable.summary.component_versions[0])
       : false;
+    */
+    return false
   }
   get passwordExpiration() {
     return GlobalVariable.user.token.password_days_until_expire;
