diff --git a/sigdump.gemspec b/sigdump.gemspec
index dd34181..813838d 100644
--- a/sigdump.gemspec
+++ b/sigdump.gemspec
@@ -11,7 +11,6 @@ Gem::Specification.new do |gem|
   gem.authors     = ["Sadayuki Furuhashi"]
   gem.email       = ["frsyuki@gmail.com"]
   gem.license     = "MIT"
-  gem.has_rdoc    = false
   gem.files       = `git ls-files`.split("\n")
   gem.test_files  = `git ls-files -- {test,spec,features}/*`.split("\n")
   gem.executables = `git ls-files -- bin/*`.split("\n").map{ |f| File.basename(f) }
