From 4b5408b307f524862a9cf8ce73af0eb595e62f88 Mon Sep 17 00:00:00 2001
From: Srishti Hegde <srishti.hegde@chainguard.dev>
Date: Wed, 15 May 2024 15:31:54 -0700
Subject: [PATCH] build prod providers

Signed-off-by: Srishti Hegde <srishti.hegde@chainguard.dev>
---
 hatch_build.py | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/hatch_build.py b/hatch_build.py
index 705cbf9f65..00eb7d4558 100644
--- a/hatch_build.py
+++ b/hatch_build.py
@@ -40,14 +40,33 @@ GENERATED_PROVIDERS_DEPENDENCIES_FILE = AIRFLOW_ROOT_PATH / "generated" / "provi
 PROVIDER_DEPENDENCIES = json.loads(GENERATED_PROVIDERS_DEPENDENCIES_FILE.read_text())
 
 PRE_INSTALLED_PROVIDERS = [
+    "amazon",
+    "celery",
+    "cncf.kubernetes",
     "common.io",
     "common.sql",
+    "docker",
+    "elasticsearch",
     "fab>=1.0.2dev0",
     "ftp",
+    "google",
+    "grpc",
+    "hashicorp",
     "http",
     "imap",
+    "microsoft.azure",
+    "mysql",
+    "odbc",
+    "openlineage",
+    "postgres",
+    "redis",
+    "sendgrid",
+    "sftp",
+    "slack",
     "smtp",
+    "snowflake",
     "sqlite",
+    "ssh",
 ]
 
 # Those extras are dynamically added by hatch in the build hook to metadata optional dependencies
-- 
2.39.3 (Apple Git-146)

