From 4fb1845778bdd73b1e34a55ce3ffab727767c539 Mon Sep 17 00:00:00 2001
From: Srishti Hegde <srishti.hegde@chainguard.dev>
Date: Wed, 15 May 2024 09:42:53 -0700
Subject: [PATCH] build celery provider

---
 hatch_build.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/hatch_build.py b/hatch_build.py
index 4e0d74b254..b22f8cc596 100644
--- a/hatch_build.py
+++ b/hatch_build.py
@@ -40,6 +40,7 @@ GENERATED_PROVIDERS_DEPENDENCIES_FILE = AIRFLOW_ROOT_PATH / "generated" / "provi
 PROVIDER_DEPENDENCIES = json.loads(GENERATED_PROVIDERS_DEPENDENCIES_FILE.read_text())
 
 PRE_INSTALLED_PROVIDERS = [
+    "celery",
     "common.io",
     "common.sql",
     "fab>=1.0.2",
-- 
2.39.3 (Apple Git-146)

