From 2f5c710f941e340adb6d5de0a627b65d51d186c8 Mon Sep 17 00:00:00 2001
From: Dentrax <furkan.turkal@chainguard.dev>
Date: Mon, 2 Sep 2024 16:33:18 +0300
Subject: [PATCH] fix requests patch

Signed-off-by: Dentrax <furkan.turkal@chainguard.dev>
---
 hatch_build.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/hatch_build.py b/hatch_build.py
index f366ae9..1613d5c 100644
--- a/hatch_build.py
+++ b/hatch_build.py
@@ -474,7 +474,7 @@ DEPENDENCIES = [
     "python-nvd3>=0.15.0",
     "python-slugify>=5.0",
     # Requests 3 if it will be released, will be heavily breaking.
-    "requests>=2.27.0,<3",
+    "requests>=2.32.0,<3",
     "requests-toolbelt>=0.4.0",
     "rfc3339-validator>=0.1.4",
     "rich-argparse>=1.0.0",
 ]
 
 
-- 
2.39.3 (Apple Git-146)

