diff --git a/hatch_build.py b/hatch_build.py
index 705cbf9f65..6c3ec4244e 100644
--- a/hatch_build.py
+++ b/hatch_build.py
@@ -478,7 +478,7 @@ DEPENDENCIES = [
     "python-nvd3>=0.15.0",
     "python-slugify>=5.0",
     # Requests 3 if it will be released, will be heavily breaking.
-    "requests>=2.27.0,<3",
+    "requests>=2.32.0,<3",
     "rfc3339-validator>=0.1.4",
     "rich-argparse>=1.0.0",
     "rich>=12.4.4",
@@ -501,7 +501,7 @@ DEPENDENCIES = [
     "universal-pathlib>=0.2.2",
     # Werkzug 3 breaks Flask-Login 0.6.2, also connexion needs to be updated to >= 3.0
     # we should remove this limitation when FAB supports Flask 2.3 and we migrate connexion to 3+
-    "werkzeug>=2.0,<3",
+    "werkzeug>=2.3.8,<3",
 ]
 
 
