diff --git a/src/bin/tolua.c b/src/bin/tolua.c
index cc9f9ad..0f3152a 100644
--- a/src/bin/tolua.c
+++ b/src/bin/tolua.c
@@ -67,7 +67,7 @@ static void setfield (lua_State* L, int table, char* f, char* v)
 static void add_extra (lua_State* L, char* value) {
 	int len;
 	lua_getglobal(L, "_extra_parameters");
-	len = luaL_getn(L, -1);
+	len = luaL_len(L, -1);
 	lua_pushstring(L, value);
 	lua_rawseti(L, -2, len+1);
 	lua_pop(L, 1);
diff --git a/src/lib/tolua_event.c b/src/lib/tolua_event.c
index 8258867..6dd5e3d 100644
--- a/src/lib/tolua_event.c
+++ b/src/lib/tolua_event.c
@@ -23,12 +23,12 @@
 static void storeatubox (lua_State* L, int lo)
 {
 	#ifdef LUA_VERSION_NUM
-		lua_getfenv(L, lo);
+		lua_getuservalue(L, lo);
 		if (lua_rawequal(L, -1, TOLUA_NOPEER)) {
 			lua_pop(L, 1);
 			lua_newtable(L);
 			lua_pushvalue(L, -1);
-			lua_setfenv(L, lo);	/* stack: k,v,table  */
+			lua_setuservalue(L, lo);	/* stack: k,v,table  */
 		};
 		lua_insert(L, -3);
 		lua_settable(L, -3); /* on lua 5.1, we trade the "tolua_peers" lookup for a settable call */
@@ -141,7 +141,7 @@ static int class_index_event (lua_State* L)
 	{
 		/* Access alternative table */
 		#ifdef LUA_VERSION_NUM /* new macro on version 5.1 */
-		lua_getfenv(L,1);
+		lua_getuservalue(L,1);
 		if (!lua_rawequal(L, -1, TOLUA_NOPEER)) {
 			lua_pushvalue(L, 2); /* key */
 			lua_gettable(L, -2); /* on lua 5.1, we trade the "tolua_peers" lookup for a gettable call */
diff --git a/src/lib/tolua_map.c b/src/lib/tolua_map.c
index d00e706..e89cf06 100644
--- a/src/lib/tolua_map.c
+++ b/src/lib/tolua_map.c
@@ -263,7 +263,7 @@ static int tolua_bnd_setpeer(lua_State* L) {
 		lua_pop(L, 1);
 		lua_pushvalue(L, TOLUA_NOPEER);
 	};
-	lua_setfenv(L, -2);
+	lua_setuservalue(L, -2);
 
 	return 0;
 };
@@ -271,7 +271,7 @@ static int tolua_bnd_setpeer(lua_State* L) {
 static int tolua_bnd_getpeer(lua_State* L) {
 
 	/* stack: userdata */
-	lua_getfenv(L, -1);
+	lua_getuservalue(L, -1);
 	if (lua_rawequal(L, -1, TOLUA_NOPEER)) {
 		lua_pop(L, 1);
 		lua_pushnil(L);
@@ -411,7 +411,7 @@ TOLUA_API void tolua_beginmodule (lua_State* L, const char* name)
 		lua_rawget(L,-2);
 	}
 	else
-	 lua_pushvalue(L,LUA_GLOBALSINDEX);
+        lua_getglobal(L, "_G");
 }
 
 /* End module
@@ -445,7 +445,7 @@ TOLUA_API void tolua_module (lua_State* L, const char* name, int hasvar)
 	else
 	{
 		/* global table */
-		lua_pushvalue(L,LUA_GLOBALSINDEX);
+		lua_getglobal(L, "_G");
 	}
 	if (hasvar)
 	{
diff --git a/src/lib/tolua_push.c b/src/lib/tolua_push.c
index 6394147..08ae751 100644
--- a/src/lib/tolua_push.c
+++ b/src/lib/tolua_push.c
@@ -79,7 +79,7 @@ TOLUA_API void tolua_pushusertype (lua_State* L, void* value, const char* type)
 
    #ifdef LUA_VERSION_NUM
    lua_pushvalue(L, TOLUA_NOPEER);
-   lua_setfenv(L, -2);
+   lua_setuservalue(L, -2);
    #endif
   }
   else
