From 7375f0ab668c8ba97ad236fbb6e926d66630ba06 Mon Sep 17 00:00:00 2001
From: ajayk <ajaykemparaj@gmail.com>
Date: Fri, 9 Feb 2024 11:13:24 -0800
Subject: [PATCH] fixes envoy commit sha256

---
 WORKSPACE | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/WORKSPACE b/WORKSPACE
index 7e335952fa..d6fb484f93 100644
--- a/WORKSPACE
+++ b/WORKSPACE
@@ -37,7 +37,7 @@ bind(
 # Commit date: 2024-01-18
 ENVOY_SHA = "3e6e83d166c2ec70aa07a60be088ac76a60734a9"
 
-ENVOY_SHA256 = "de0e61e908a6dd5e27271e7c1482b1f11d364b7fddb2481adfeed1a1ce4096af"
+ENVOY_SHA256 = "505056137453f5fa57fad9c20d9bfcf2290ad6dc229030c42dd20cbd7d300540"
 
 ENVOY_ORG = "istio"
 
