diff --git a/src/pam/mod.rs b/src/pam/mod.rs
index 9eec40c..d46c795 100644
--- a/src/pam/mod.rs
+++ b/src/pam/mod.rs
@@ -2,6 +2,7 @@ use std::{
     collections::HashMap,
     ffi::{CStr, CString, OsStr, OsString},
     os::unix::prelude::OsStrExt,
+    os::raw::c_char,
 };

 use converse::ConverserData;
@@ -214,7 +215,7 @@ impl<C: Converser> PamContext<C> {
         }

         // unsafe conversion to cstr
-        let cstr = unsafe { CStr::from_ptr(data as *const i8) };
+        let cstr = unsafe { CStr::from_ptr(data as *const c_char) };

         Ok(cstr.to_str()?.to_owned())
     }

