From 1f32fd335681dfbaf82123c589da2e18f307a756 Mon Sep 17 00:00:00 2001
From: Ariadne Conill <ariadne@dereferenced.org>
Date: Thu, 7 Sep 2023 12:33:54 -0700
Subject: [PATCH] tsl: denormal: add missing cstdint include

Signed-off-by: Ariadne Conill <ariadne@dereferenced.org>
---
 tensorflow/tsl/platform/denormal.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tensorflow/tsl/platform/denormal.cc b/tensorflow/tsl/platform/denormal.cc
index d8685c4db26..d8caf5447ed 100644
--- a/tensorflow/tsl/platform/denormal.cc
+++ b/tensorflow/tsl/platform/denormal.cc
@@ -13,6 +13,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 ==============================================================================*/
 
+#include <cstdint>
 #include "tensorflow/tsl/platform/denormal.h"
 
 #include "tensorflow/tsl/platform/cpu_info.h"
-- 
2.39.2 (Apple Git-143)

