From cf74137a5d65ae3bf9b0fa5214b2844f23c85660 Mon Sep 17 00:00:00 2001
From: Ariadne Conill <ariadne@dereferenced.org>
Date: Thu, 7 Sep 2023 11:57:40 -0700
Subject: [PATCH] core: fix compile with GCC 13

Signed-off-by: Ariadne Conill <ariadne@dereferenced.org>
---
 tensorflow/core/lib/io/cache.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tensorflow/core/lib/io/cache.h b/tensorflow/core/lib/io/cache.h
index f894c5916d5..4e509cea05c 100644
--- a/tensorflow/core/lib/io/cache.h
+++ b/tensorflow/core/lib/io/cache.h
@@ -16,6 +16,7 @@ limitations under the License.
 #ifndef TENSORFLOW_core_LIB_IO_CACHE_H_
 #define TENSORFLOW_core_LIB_IO_CACHE_H_
 
+#include <cstdint>
 #include "tensorflow/core/platform/stringpiece.h"
 
 // A Cache is an interface that maps keys to values.  It has internal
-- 
2.39.2 (Apple Git-143)

