diff --git a/tensorflow/tools/pip_package/setup.py.tpl b/tensorflow/tools/pip_package/setup.py.tpl
index 949dc393..3426e7f1 100644
--- a/tensorflow/tools/pip_package/setup.py.tpl
+++ b/tensorflow/tools/pip_package/setup.py.tpl
@@ -110,7 +110,7 @@ REQUIRED_PACKAGES = [
     'tensorboard ~= 2.20.0',
     'keras >= 3.10.0',
     'numpy >= 1.26.0',
-    'h5py >= 3.11.0',
+    'h5py >= 3.13.0',
     'ml_dtypes >= 0.5.1, < 1.0.0',
 ]
