diff --git a/LICENSE-binary b/LICENSE-binary
index 030f62b967..d9416410b3 100644
--- a/LICENSE-binary
+++ b/LICENSE-binary
@@ -206,10 +206,10 @@ This project bundles some components that are also licensed under the Apache
 License Version 2.0:
 
 - caffeine-3.1.1
-- commons-beanutils-1.9.4
+- commons-beanutils-1.11.0
 - commons-collections-3.2.2
 - commons-digester-2.1
-- commons-lang3-3.12.0
+- commons-lang3-3.18.0
 - commons-logging-1.3.2
 - commons-validator-1.9.0
 - error_prone_annotations-2.14.0
diff --git a/build.gradle b/build.gradle
index 388a85aa85..066f133bed 100644
--- a/build.gradle
+++ b/build.gradle
@@ -197,7 +197,10 @@ allprojects {
           // ensure we have a single version in the classpath despite transitive dependencies
           libs.scalaLibrary,
           libs.scalaReflect,
-          libs.jacksonAnnotations
+          libs.jacksonAnnotations,
+          // Enforce the updated commons-beanutils to address CVE
+          libs.commonsBeanutils,
+          libs.commonsLang3,
         )
       }
     }
@@ -1084,6 +1087,7 @@ project(':core') {
     implementation project(':share-coordinator')
 
     implementation libs.argparse4j
+    implementation libs.commonsLang3
     implementation libs.commonsValidator
     implementation libs.jacksonDatabind
     implementation libs.jacksonDataformatCsv
diff --git a/gradle/dependencies.gradle b/gradle/dependencies.gradle
index 66eca369aa..b7a75be8a5 100644
--- a/gradle/dependencies.gradle
+++ b/gradle/dependencies.gradle
@@ -60,6 +60,8 @@ versions += [
   caffeine: "3.1.1",
   bndlib: "7.0.0",
   checkstyle: project.hasProperty('checkstyleVersion') ? checkstyleVersion : "10.20.2",
+  commonsBeanutils: "1.11.0", // Added to address CVE GHSA-wxr5-93ph-8wr9
+  commonsLang3: "3.18.0",
   commonsValidator: "1.9.0",
   classgraph: "4.8.173",
   gradle: "8.10.2",
@@ -148,6 +150,8 @@ libs += [
   bndlib:"biz.aQute.bnd:biz.aQute.bndlib:$versions.bndlib",
   caffeine: "com.github.ben-manes.caffeine:caffeine:$versions.caffeine",
   classgraph: "io.github.classgraph:classgraph:$versions.classgraph",
+  commonsBeanutils: "commons-beanutils:commons-beanutils:$versions.commonsBeanutils",
+  commonsLang3: "org.apache.commons:commons-lang3:$versions.commonsLang3",
   commonsValidator: "commons-validator:commons-validator:$versions.commonsValidator",
   jacksonAnnotations: "com.fasterxml.jackson.core:jackson-annotations:$versions.jackson",
   jacksonDatabind: "com.fasterxml.jackson.core:jackson-databind:$versions.jackson",
