From 39d769de5cbd0c98aae7f9ad668f52740d2f5c82 Mon Sep 17 00:00:00 2001
From: Dentrax <furkan.turkal@chainguard.dev>
Date: Sun, 5 Jan 2025 20:43:32 +0300
Subject: [PATCH] GHSA-r8w9-5wcg-vfj7

Signed-off-by: Dentrax <furkan.turkal@chainguard.dev>
---
 zellij-client/Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/zellij-client/Cargo.toml b/zellij-client/Cargo.toml
index b27f1dd..813f2f8 100644
--- a/zellij-client/Cargo.toml
+++ b/zellij-client/Cargo.toml
@@ -14,7 +14,7 @@ interprocess = { workspace = true }
 lazy_static = { workspace = true }
 libc = { workspace = true }
 log = { workspace = true }
-mio = { version = "0.7.11", default-features = false, features = ['os-ext'] }
+mio = { version = "0.8.11", default-features = false, features = ['os-ext'] }
 nix = { workspace = true }
 notify-debouncer-full = { workspace = true }
 regex = { workspace = true }
