diff --git a/stdlib/exit.c b/stdlib/exit.c
index 114657a8fb..30770c78e1 100644
--- a/stdlib/exit.c
+++ b/stdlib/exit.c
@@ -23,6 +23,12 @@
 #include <set-freeres.h>
 #include "exit.h"
 
+#if defined __has_include
+# if __has_include ("/home/build/.melange.fdo.h")
+#  include "/home/build/.melange.fdo.h"
+# endif
+#endif
+
 /* Initialize the flag that indicates exit function processing
    is complete. See concurrency notes in stdlib/exit.h where
    __exit_funcs_lock is declared.  */
