--- a/vendor/github.com/docker/docker/pkg/archive/archive_deprecated.go
+++ b/vendor/github.com/docker/docker/pkg/archive/archive_deprecated.go
@@ -100,7 +100,7 @@
 // DetectCompression detects the compression algorithm of the source.
 //
 // Deprecated: use [compression.Detect] instead.
-func DetectCompression(source []byte) archive.Compression {
+func DetectCompression(source []byte) Compression {
 	return compression.Detect(source)
 }
 
@@ -156,7 +156,7 @@
 // stream of bytes.
 //
 // Deprecated: use [archive.Tar] instead.
-func Tar(path string, compression archive.Compression) (io.ReadCloser, error) {
+func Tar(path string, compression Compression) (io.ReadCloser, error) {
 	return archive.TarWithOptions(path, &archive.TarOptions{Compression: compression})
 }
 
--- a/vendor/github.com/docker/docker/daemon/graphdriver/fsdiff.go
+++ b/vendor/github.com/docker/docker/daemon/graphdriver/fsdiff.go
@@ -9,6 +9,7 @@
 	"github.com/docker/docker/pkg/ioutils"
 	"github.com/moby/go-archive"
 	"github.com/moby/go-archive/chrootarchive"
+	"github.com/moby/go-archive/compression"
 	"github.com/moby/sys/user"
 )
 
@@ -64,7 +65,7 @@
 	}()
 
 	if parent == "" {
-		tarArchive, err := archive.Tar(layerFs, archive.Uncompressed)
+		tarArchive, err := archive.Tar(layerFs, compression.None)
 		if err != nil {
 			return nil, err
 		}
--- a/vendor/github.com/docker/docker/builder/remotecontext/git.go
+++ b/vendor/github.com/docker/docker/builder/remotecontext/git.go
@@ -8,6 +8,7 @@
 	"github.com/docker/docker/builder"
 	"github.com/docker/docker/builder/remotecontext/git"
 	"github.com/moby/go-archive"
+	"github.com/moby/go-archive/compression"
 )
 
 // MakeGitContext returns a Context from gitURL that is cloned in a temporary directory.
@@ -17,7 +18,7 @@
 		return nil, err
 	}
 
-	c, err := archive.Tar(root, archive.Uncompressed)
+	c, err := archive.Tar(root, compression.None)
 	if err != nil {
 		return nil, err
 	}
