diff --git a/pkg/executor/build.go b/pkg/executor/build.go
# Fixes: https://github.com/chainguard-dev/customer-issues/issues/2579
# Internal Issue to investigate long-term fix: https://github.com/chainguard-dev/internal-dev/issues/21302
index 73b2f0df..75e885f2 100644
--- a/pkg/executor/build.go
+++ b/pkg/executor/build.go
@@ -691,6 +691,8 @@ func CalculateDependencies(stages []config.KanikoStage, opts *config.KanikoOptio
 // DoBuild executes building the Dockerfile
 func DoBuild(opts *config.KanikoOptions) (v1.Image, error) {
 	t := timing.Start("Total Build Time")
+	fixUsrMergeSymlinks()
+    
 	digestToCacheKey := make(map[string]string)
 	stageIdxToDigest := make(map[string]string)
 
@@ -827,6 +829,25 @@ func DoBuild(opts *config.KanikoOptions) (v1.Image, error) {
 	return nil, err
 }

+// remove usrmerge symlinks to avoid circular dependencies
+func fixUsrMergeSymlinks() error {
+	for _, link := range []string{"/bin", "/lib", "/sbin", "/lib64"} {
+		info, err := os.Lstat(link)
+		if err != nil || info.Mode()&os.ModeSymlink == 0 {
+			continue
+		}
+
+		if err := os.Remove(link); err != nil {
+			return err
+		}
+	}
+
+	// Recreate /bin directory and /bin/sh symlink needed by kaniko
+	if err := os.MkdirAll("/bin", 0755); err != nil {
+		return err
+	}
+
+	if err := os.Symlink("/busybox/sh", "/bin/sh"); err != nil {
+		return err
+	}
+
+	return nil
+}
+
 
 // filesToSave returns all the files matching the given pattern in deps.
 // If a file is a symlink, it also returns the target file.
 func filesToSave(deps []string) ([]string, error) {
