From 60e5cc4f940a65fbd5fbf294ad8e8d05877342c2 Mon Sep 17 00:00:00 2001
From: Batuhan Apaydin <batuhan.apaydin@chainguard.dev>
Date: Fri, 26 Apr 2024 12:39:47 +0300
Subject: [PATCH] tls-cipher fix

Signed-off-by: Batuhan Apaydin <batuhan.apaydin@chainguard.dev>
---
 util/tls/tls.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/util/tls/tls.go b/util/tls/tls.go
index 5e18c8eb7..c7925b832 100644
--- a/util/tls/tls.go
+++ b/util/tls/tls.go
@@ -28,7 +28,7 @@ const (
 	DefaultRSABits = 2048
 	// The default TLS cipher suites to provide to clients - see https://cipherlist.eu for updates
 	// Note that for TLS v1.3, cipher suites are not configurable and will be chosen automatically.
-	DefaultTLSCipherSuite = "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384:TLS_RSA_WITH_AES_256_GCM_SHA384"
+	DefaultTLSCipherSuite = "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384"
 	// The default minimum TLS version to provide to clients
 	DefaultTLSMinVersion = "1.2"
 	// The default maximum TLS version to provide to clients
-- 
2.39.3 (Apple Git-146)

