From ada5fb811e09dfe78f3f8bcdf265b94d1dced97c Mon Sep 17 00:00:00 2001
From: Batuhan Apaydin <batuhan.apaydin@chainguard.dev>
Date: Fri, 16 May 2025 20:03:12 +0300
Subject: [PATCH]  Remove modules from bundled gems list to avoid unnecessary
 duplication.

Signed-off-by: Batuhan Apaydin <batuhan.apaydin@chainguard.dev>
---
 gems/bundled_gems | 2 --
 1 file changed, 2 deletions(-)

diff --git a/gems/bundled_gems b/gems/bundled_gems
index a5286d5..d4f50fa 100644
--- a/gems/bundled_gems
+++ b/gems/bundled_gems
@@ -10,10 +10,8 @@ minitest            6.0.0  https://github.com/minitest/minitest
 power_assert        3.0.1   https://github.com/ruby/power_assert
 rake                13.3.1  https://github.com/ruby/rake
 test-unit           3.7.5   https://github.com/test-unit/test-unit
-rexml               3.4.4   https://github.com/ruby/rexml
 rss                 0.3.2   https://github.com/ruby/rss
 net-ftp             0.3.9   https://github.com/ruby/net-ftp
-net-imap            0.6.2   https://github.com/ruby/net-imap
 net-pop             0.1.2   https://github.com/ruby/net-pop
 net-smtp            0.5.1   https://github.com/ruby/net-smtp
 matrix              0.4.3   https://github.com/ruby/matrix
