libreplace is not needed on Linux

--- a/configure.ac
+++ b/configure.ac
@@ -273,8 +273,6 @@ AC_CHECK_FUNCS([strchr strcspn strdup st
 AC_CHECK_FUNCS([strcasecmp strncasecmp strspn strtoul])
 AC_CHECK_FUNCS([gethostbyname inet_ntoa select strrchr strstr strtol])
 
-AC_REPLACE_FUNCS([getopt_long_only strncasecmp])
-AC_REPLACE_FUNCS([__ns_initparse __ns_name_uncompress __ns_msg_getflag])
 dnl Magnus Holmgren says this avoids linking a private function
 AC_LIBOBJ([__ns_get16])
 
@@ -282,7 +280,6 @@ AC_CONFIG_FILES([Makefile
                  Doxyfile
                  src/Makefile
                  src/include/Makefile
-                 src/libreplace/Makefile
                  src/libspf2/Makefile
                  src/spfquery/Makefile
                  src/spftest/Makefile
--- a/src/libspf2/Makefile.am
+++ b/src/libspf2/Makefile.am
@@ -34,10 +34,6 @@ libspf2_la_SOURCES	= \
 	spf_utils.c \
 	spf_win32.c
 
-libspf2_la_LIBADD	= $(top_builddir)/src/libreplace/libreplace.la
-
-
-
 libspf2_la_LDFLAGS	= -version-info 3:0:1
 
 # Copied from the libtool info file:
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -4,5 +4,5 @@ AUTOMAKE_OPTIONS = foreign
 # Attempt to disable parallel processing.
 .NOTPARALLEL:
 .NO_PARALLEL:
-SUBDIRS =	include libreplace libspf2 spfquery spftest spfd spf_example
+SUBDIRS =	include libspf2 spfquery spftest spfd spf_example
 TARGETS =
--- a/aclocal.m4
+++ b/aclocal.m4
@@ -1,4 +1,5 @@
 # generated automatically by aclocal 1.11.1 -*- Autoconf -*-
+# and touched to get build order right
 
 # Copyright (C) 1996, 1997, 1998, 1999, 2000, 2001, 2002, 2003, 2004,
 # 2005, 2006, 2007, 2008, 2009  Free Software Foundation, Inc.
--- a/configure
+++ b/configure
@@ -13602,73 +13602,6 @@ fi
 done
 
 
-ac_fn_c_check_func "$LINENO" "getopt_long_only" "ac_cv_func_getopt_long_only"
-if test "x$ac_cv_func_getopt_long_only" = xyes; then :
-  $as_echo "#define HAVE_GETOPT_LONG_ONLY 1" >>confdefs.h
-
-else
-  case " $LIBOBJS " in
-  *" getopt_long_only.$ac_objext "* ) ;;
-  *) LIBOBJS="$LIBOBJS getopt_long_only.$ac_objext"
- ;;
-esac
-
-fi
-
-ac_fn_c_check_func "$LINENO" "strncasecmp" "ac_cv_func_strncasecmp"
-if test "x$ac_cv_func_strncasecmp" = xyes; then :
-  $as_echo "#define HAVE_STRNCASECMP 1" >>confdefs.h
-
-else
-  case " $LIBOBJS " in
-  *" strncasecmp.$ac_objext "* ) ;;
-  *) LIBOBJS="$LIBOBJS strncasecmp.$ac_objext"
- ;;
-esac
-
-fi
-
-
-ac_fn_c_check_func "$LINENO" "__ns_initparse" "ac_cv_func___ns_initparse"
-if test "x$ac_cv_func___ns_initparse" = xyes; then :
-  $as_echo "#define HAVE___NS_INITPARSE 1" >>confdefs.h
-
-else
-  case " $LIBOBJS " in
-  *" __ns_initparse.$ac_objext "* ) ;;
-  *) LIBOBJS="$LIBOBJS __ns_initparse.$ac_objext"
- ;;
-esac
-
-fi
-
-ac_fn_c_check_func "$LINENO" "__ns_name_uncompress" "ac_cv_func___ns_name_uncompress"
-if test "x$ac_cv_func___ns_name_uncompress" = xyes; then :
-  $as_echo "#define HAVE___NS_NAME_UNCOMPRESS 1" >>confdefs.h
-
-else
-  case " $LIBOBJS " in
-  *" __ns_name_uncompress.$ac_objext "* ) ;;
-  *) LIBOBJS="$LIBOBJS __ns_name_uncompress.$ac_objext"
- ;;
-esac
-
-fi
-
-ac_fn_c_check_func "$LINENO" "__ns_msg_getflag" "ac_cv_func___ns_msg_getflag"
-if test "x$ac_cv_func___ns_msg_getflag" = xyes; then :
-  $as_echo "#define HAVE___NS_MSG_GETFLAG 1" >>confdefs.h
-
-else
-  case " $LIBOBJS " in
-  *" __ns_msg_getflag.$ac_objext "* ) ;;
-  *) LIBOBJS="$LIBOBJS __ns_msg_getflag.$ac_objext"
- ;;
-esac
-
-fi
-
-
 case " $LIBOBJS " in
   *" __ns_get16.$ac_objext "* ) ;;
   *) LIBOBJS="$LIBOBJS __ns_get16.$ac_objext"
@@ -13676,7 +13609,7 @@ case " $LIBOBJS " in
 esac
 
 
-ac_config_files="$ac_config_files Makefile Doxyfile src/Makefile src/include/Makefile src/libreplace/Makefile src/libspf2/Makefile src/spfquery/Makefile src/spftest/Makefile src/spfd/Makefile src/spf_example/Makefile"
+ac_config_files="$ac_config_files Makefile Doxyfile src/Makefile src/include/Makefile src/libspf2/Makefile src/spfquery/Makefile src/spftest/Makefile src/spfd/Makefile src/spf_example/Makefile"
 
 cat >confcache <<\_ACEOF
 # This file is a shell script that caches the results of configure
@@ -14693,7 +14626,6 @@ do
     "Doxyfile") CONFIG_FILES="$CONFIG_FILES Doxyfile" ;;
     "src/Makefile") CONFIG_FILES="$CONFIG_FILES src/Makefile" ;;
     "src/include/Makefile") CONFIG_FILES="$CONFIG_FILES src/include/Makefile" ;;
-    "src/libreplace/Makefile") CONFIG_FILES="$CONFIG_FILES src/libreplace/Makefile" ;;
     "src/libspf2/Makefile") CONFIG_FILES="$CONFIG_FILES src/libspf2/Makefile" ;;
     "src/spfquery/Makefile") CONFIG_FILES="$CONFIG_FILES src/spfquery/Makefile" ;;
     "src/spftest/Makefile") CONFIG_FILES="$CONFIG_FILES src/spftest/Makefile" ;;
--- a/config.h.in
+++ b/config.h.in
@@ -49,9 +49,6 @@
 /* Define to 1 if you have the <getopt.h> header file. */
 #undef HAVE_GETOPT_H
 
-/* Define to 1 if you have the `getopt_long_only' function. */
-#undef HAVE_GETOPT_LONG_ONLY
-
 /* Define to 1 if you have the <grp.h> header file. */
 #undef HAVE_GRP_H
 
--- a/src/libspf2/Makefile.in
+++ b/src/libspf2/Makefile.in
@@ -69,8 +69,7 @@ am__base_list = \
   sed '$$!N;$$!N;$$!N;$$!N;s/\n/ /g'
 am__installdirs = "$(DESTDIR)$(libdir)"
 LTLIBRARIES = $(lib_LTLIBRARIES)
-libspf2_la_DEPENDENCIES =  \
-	$(top_builddir)/src/libreplace/libreplace.la
+libspf2_la_LIBADD =
 am_libspf2_la_OBJECTS = spf_compile.lo spf_dns.lo spf_dns_cache.lo \
 	spf_dns_null.lo spf_dns_resolv.lo spf_dns_rr.lo \
 	spf_dns_test.lo spf_dns_windns.lo spf_dns_zone.lo \
@@ -289,7 +288,6 @@ libspf2_la_SOURCES = \
 	spf_utils.c \
 	spf_win32.c
 
-libspf2_la_LIBADD = $(top_builddir)/src/libreplace/libreplace.la
 libspf2_la_LDFLAGS = -version-info 3:0:1
 all: all-recursive
 
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -151,7 +151,7 @@ SED = @SED@
 SET_MAKE = @SET_MAKE@
 SHELL = @SHELL@
 STRIP = @STRIP@
-SUBDIRS = include libreplace libspf2 spfquery spftest spfd spf_example
+SUBDIRS = include libspf2 spfquery spftest spfd spf_example
 VERSION = @VERSION@
 abs_builddir = @abs_builddir@
 abs_srcdir = @abs_srcdir@
--- a/src/spfd/spfd.c
+++ b/src/spfd/spfd.c
@@ -88,12 +88,8 @@
 #include <grp.h>
 #endif
 
-#ifdef HAVE_GETOPT_LONG_ONLY
 #define _GNU_SOURCE
 #include <getopt.h>
-#else
-#include "libreplace/getopt.h"
-#endif
 
 #include <unistd.h>
 #include <netdb.h>
--- a/src/spfquery/spfquery.c
+++ b/src/spfquery/spfquery.c
@@ -46,8 +46,6 @@
 
 #define SPF_TEST_VERSION  "3.0"
 
-#include "libreplace/win32_config.h"
-
 #ifdef HAVE_CONFIG_H
 # include "config.h"
 #endif
@@ -88,12 +86,8 @@
 # include <arpa/inet.h>	/* in_addr struct */
 #endif
 
-#ifdef HAVE_GETOPT_LONG_ONLY
 #define _GNU_SOURCE
 #include <getopt.h>
-#else
-#include "libreplace/getopt.h"
-#endif
 
 #ifdef _WIN32
 #include "spf_win32.h"

