Index: firefox-148.0/Cargo.lock
===================================================================
--- firefox-148.0.orig/Cargo.lock
+++ firefox-148.0/Cargo.lock
@@ -2804,8 +2804,6 @@ dependencies = [
 [[package]]
 name = "glslopt"
 version = "0.1.12"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "8ba4741358604ca0848c27ecc069d68e62e11cde81e38aac1da3c54b79ab5adf"
 dependencies = [
  "cc",
 ]
Index: firefox-148.0/supply-chain/config.toml
===================================================================
--- firefox-148.0.orig/supply-chain/config.toml
+++ firefox-148.0/supply-chain/config.toml
@@ -64,6 +64,10 @@ notes = "Used for testing."
 dependency-criteria = { tokio-reactor = [], tokio-threadpool = [] }
 notes = "The dependencies on tokio-reactor and tokio-threadpools are just a hack to pin the version used by audioipc-{client,server}. Suppress vetting on those for the same reasons behind the policy entries."
 
+[policy.glslopt]
+audit-as-crates-io = true
+notes = "Patched version of upstream"
+
 [policy.gluesmith]
 criteria = "safe-to-run"
 notes = "Used for fuzzing."
Index: firefox-148.0/build/rust/glslopt/glsl-optimizer/include/c11/threads_posix.h
===================================================================
--- firefox-148.0.orig/build/rust/glslopt/glsl-optimizer/include/c11/threads_posix.h
+++ firefox-148.0/build/rust/glslopt/glsl-optimizer/include/c11/threads_posix.h
@@ -47,11 +47,16 @@ Configuration macro:
 #define EMULATED_THREADS_USE_NATIVE_TIMEDLOCK
 #endif
 
+#ifndef ONCE_FLAG_INIT
+#define ONCE_FLAG_USE_PTHREAD_ONCE
+#endif
 
 #include <pthread.h>
 
 /*---------------------------- macros ----------------------------*/
+#ifdef ONCE_FLAG_USE_PTHREAD_ONCE
 #define ONCE_FLAG_INIT PTHREAD_ONCE_INIT
+#endif
 #ifdef INIT_ONCE_STATIC_INIT
 #define TSS_DTOR_ITERATIONS PTHREAD_DESTRUCTOR_ITERATIONS
 #else
@@ -66,8 +71,9 @@ typedef pthread_cond_t  cnd_t;
 typedef pthread_t       thrd_t;
 typedef pthread_key_t   tss_t;
 typedef pthread_mutex_t mtx_t;
+#ifdef ONCE_FLAG_USE_PTHREAD_ONCE
 typedef pthread_once_t  once_flag;
-
+#endif
 
 /*
 Implementation limits:
@@ -90,11 +96,13 @@ impl_thrd_routine(void *p)
 
 /*--------------- 7.25.2 Initialization functions ---------------*/
 // 7.25.2.1
+#ifdef ONCE_FLAG_USE_PTHREAD_ONCE
 static inline void
 call_once(once_flag *flag, void (*func)(void))
 {
     pthread_once(flag, func);
 }
+#endif
 
 
 /*------------- 7.25.3 Condition variable functions -------------*/
Index: firefox-148.0/Cargo.toml
===================================================================
--- firefox-148.0.orig/Cargo.toml
+++ firefox-148.0/Cargo.toml
@@ -240,6 +240,9 @@ moz_asserts = { path = "mozglue/static/r
 # Workaround for https://github.com/rust-lang/cargo/issues/11232
 rure = { path = "build/rust/rure" }
 
+# Patch glslopt to fix build with glibc 2.43
+glslopt = { path = "build/rust/glslopt" }
+
 # To-be-published changes.
 unicode-bidi = { git = "https://github.com/servo/unicode-bidi", rev = "ca612daf1c08c53abe07327cb3e6ef6e0a760f0c" }
 nss-gk-api = { git = "https://github.com/beurdouche/nss-gk-api", rev = "e48a946811ffd64abc78de3ee284957d8d1c0d63" }
