Index: firefox-148.0/mfbt/tests/TestIntegerRange.cpp
===================================================================
--- firefox-148.0.orig/mfbt/tests/TestIntegerRange.cpp
+++ firefox-148.0/mfbt/tests/TestIntegerRange.cpp
@@ -8,6 +8,7 @@
 #include "mozilla/IntegerRange.h"
 
 #include <stddef.h>
+#include <stdint.h>
 
 using mozilla::IntegerRange;
 using mozilla::Reversed;
Index: firefox-148.0/mfbt/tests/TestSmallPointerArray.cpp
===================================================================
--- firefox-148.0.orig/mfbt/tests/TestSmallPointerArray.cpp
+++ firefox-148.0/mfbt/tests/TestSmallPointerArray.cpp
@@ -4,6 +4,7 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+#include <stdint.h>
 #include "mozilla/SmallPointerArray.h"
 
 #define PTR1 (void*)0x4
