diff --git a/CMakeLists.txt b/CMakeLists.txt
index be804a14765..759bdfeb23a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -455,12 +455,7 @@ endif ()
 
 enable_testing() # Enable for tests without binary
 
-if (ARCH_S390X)
-    set(ENABLE_OPENSSL_DYNAMIC_DEFAULT ON)
-else ()
-    set(ENABLE_OPENSSL_DYNAMIC_DEFAULT OFF)
-endif ()
-option(ENABLE_OPENSSL_DYNAMIC "This option removes SSL from ClickHouse and will link to the OpenSSL version supplied by OS." ${ENABLE_OPENSSL_DYNAMIC_DEFAULT})
+option(ENABLE_OPENSSL_DYNAMIC "This option removes SSL from ClickHouse and will link to the OpenSSL version supplied by OS." ON)
 
 # when installing to /usr - place configs to /etc but for /usr/local place to /usr/local/etc
 if (CMAKE_INSTALL_PREFIX STREQUAL "/usr")
