Update the loopback binary location to be /usr/bin

diff --git a/plugins/cilium-cni/install-plugin.sh b/plugins/cilium-cni/install-plugin.sh
index f3d589acc8..9cd4673fbf 100755
--- a/plugins/cilium-cni/install-plugin.sh
+++ b/plugins/cilium-cni/install-plugin.sh
@@ -30,7 +30,7 @@ install_cni() {
 # Install the CNI loopback driver if not installed already
 if [ ! -f "${CNI_DIR}/bin/loopback" ]; then
 	# Don't fail hard if this fails as it is usually not required
-	install_cni /cni/loopback || true
+	install_cni /usr/bin/loopback || true
 fi
 
 install_cni "/opt/cni/bin/${BIN_NAME}"
