From c98516b5b76e9132eba78a399af9c95ec8d23bd4 Mon Sep 17 00:00:00 2001
From: Andrey Semashev <andrey.semashev@gmail.com>
Date: Sun, 24 Mar 2024 15:43:33 +0300
Subject: [PATCH] Updated list of special dependencies.

---
 add_path.cpp | 16 ++++++----------
 1 file changed, 6 insertions(+), 10 deletions(-)

diff --git a/add_path.cpp b/add_path.cpp
index 747bb8c..9ae43c9 100644
--- a/add_path.cpp
+++ b/add_path.cpp
@@ -196,12 +196,6 @@ void bcp_implementation::add_file(const fs::path& p)
    //
 static const std::pair<fs::path, fs::path>
    specials[] = {
-      std::pair<fs::path, fs::path>("tools/build/src/kernel/modules.jam", "libs/predef/check"),
-      std::pair<fs::path, fs::path>("tools/build/src/kernel/modules.jam", "libs/predef/tools"),
-      std::pair<fs::path, fs::path>("tools/build/src/kernel/modules.jam", "tools/boost_install/boost-install.jam"),
-      std::pair<fs::path, fs::path>("tools/build/src/kernel/modules.jam", "tools/boost_install/boost-install-dirs.jam"),
-      std::pair<fs::path, fs::path>("tools/build/src/kernel/modules.jam", "tools/boost_install/Jamfile"),
-      std::pair<fs::path, fs::path>("tools/build/src/kernel/modules.jam", "libs/headers"),
       std::pair<fs::path, fs::path>("libs/test/build/Jamfile.v2", "libs/timer/src"),
       std::pair<fs::path, fs::path>("libs/test/build/Jamfile.v2", "libs/timer/build"),
       std::pair<fs::path, fs::path>("boost/atomic/capabilities.hpp", "boost/atomic/detail"),
@@ -226,14 +220,14 @@ static const std::pair<fs::path, fs::path>
       std::pair<fs::path, fs::path>("libs/thread/build", "boost/system"),
       std::pair<fs::path, fs::path>("libs/thread/build", "boost/cerrno.hpp"),
       std::pair<fs::path, fs::path>("libs/thread/build", "boost/chrono"),
-      std::pair<fs::path, fs::path>("boost/filesystem/convenience.hpp", "boost/filesystem.hpp"),
+      std::pair<fs::path, fs::path>("boost/filesystem/cstdio.hpp", "boost/filesystem.hpp"),
+      std::pair<fs::path, fs::path>("boost/filesystem/directory.hpp", "boost/filesystem.hpp"),
       std::pair<fs::path, fs::path>("boost/filesystem/exception.hpp", "boost/filesystem.hpp"),
       std::pair<fs::path, fs::path>("boost/filesystem/fstream.hpp", "boost/filesystem.hpp"),
       std::pair<fs::path, fs::path>("boost/filesystem/operations.hpp", "boost/filesystem.hpp"),
+      std::pair<fs::path, fs::path>("boost/filesystem/file_status.hpp", "boost/filesystem.hpp"),
       std::pair<fs::path, fs::path>("boost/filesystem/path.hpp", "boost/filesystem.hpp"),
       std::pair<fs::path, fs::path>("boost/filesystem.hpp", "libs/filesystem/build"),
-      std::pair<fs::path, fs::path>("boost/filesystem.hpp", "libs/filesystem/v2"),
-      std::pair<fs::path, fs::path>("boost/filesystem.hpp", "libs/filesystem/v3"),
       std::pair<fs::path, fs::path>("boost/config.hpp", "boost/config"),
       std::pair<fs::path, fs::path>("tools/build/bootstrap.sh", "libs/config/checks"),
       std::pair<fs::path, fs::path>("tools/build/bootstrap.sh", "libs/config/test"),
@@ -242,6 +236,7 @@ static const std::pair<fs::path, fs::path>
       std::pair<fs::path, fs::path>("tools/build/bootstrap.sh", "tools/boost_install/BoostDetectToolset.cmake"),
       std::pair<fs::path, fs::path>("tools/build/bootstrap.sh", "tools/boost_install/boost-install.jam"),
       std::pair<fs::path, fs::path>("tools/build/bootstrap.sh", "tools/boost_install/boost-install-dirs.jam"),
+      std::pair<fs::path, fs::path>("tools/build/bootstrap.sh", "tools/boost_install/Jamfile"),
       std::pair<fs::path, fs::path>("tools/build/bootstrap.sh", "boostcpp.jam"),
       std::pair<fs::path, fs::path>("tools/build/bootstrap.sh", "project-config.jam"),
       std::pair<fs::path, fs::path>("tools/build/bootstrap.sh", "bootstrap.bat"),
@@ -271,7 +266,8 @@ static const std::pair<fs::path, fs::path>
       std::pair<fs::path, fs::path>("boost/test/detail/config.hpp", "libs/test/src"),
       std::pair<fs::path, fs::path>("boost/test/detail/config.hpp", "libs/test/build"),
       std::pair<fs::path, fs::path>("boost/test/detail/config.hpp", "libs/predef/build.jam"),
-      std::pair<fs::path, fs::path>("boost/test/detail/config.hpp", "libs/predef/check"),
+      std::pair<fs::path, fs::path>("boost/test/detail/config.hpp", "libs/predef/tools/check"),
+      std::pair<fs::path, fs::path>("boost/test/detail/config.hpp", "libs/predef/check"), // libs/predef/check if obsolete, but may still be used
       std::pair<fs::path, fs::path>("boost/typeof.hpp", "boost/typeof/incr_registration_group.hpp"),
       std::pair<fs::path, fs::path>("boost/function_types/detail/pp_loop.hpp", "boost/function_types/detail/pp_cc_loop"),
       std::pair<fs::path, fs::path>("boost/function_types/components.hpp", "boost/function_types/detail/components_impl"),
