diff --git a/build/mvn b/build/mvn
index 3179099304c..a493996a6e4 100755
--- a/build/mvn
+++ b/build/mvn
@@ -129,7 +129,7 @@ install_mvn() {
       if ! curl -L --output /dev/null --silent --head --fail "${APACHE_MIRROR}/${FILE_PATH}${MIRROR_URL_QUERY}" ; then
         # Fall back to archive.apache.org for older Maven
         echo "Falling back to archive.apache.org to download Maven"
-        APACHE_MIRROR="https://archive.apache.org/dist"
+        APACHE_MIRROR="https://dlcdn.apache.org"
         MIRROR_URL_QUERY=""
       fi
     fi
diff --git a/dev/make-distribution.sh b/dev/make-distribution.sh
index ef7c010e930..a3c5ea69ed2 100755
--- a/dev/make-distribution.sh
+++ b/dev/make-distribution.sh
@@ -166,7 +166,7 @@ export MAVEN_OPTS="${MAVEN_OPTS:--Xss128m -Xmx4g -XX:ReservedCodeCacheSize=128m}
 # Store the command as an array because $MVN variable might have spaces in it.
 # Normal quoting tricks don't work.
 # See: http://mywiki.wooledge.org/BashFAQ/050
-BUILD_COMMAND=("$MVN" clean package \
+BUILD_COMMAND=("$MVN" -T$(grep -c processor /proc/cpuinfo) clean package \
     -DskipTests \
     -Dmaven.javadoc.skip=true \
     -Dmaven.scaladoc.skip=true \
