diff --git a/dev/make-distribution.sh b/dev/make-distribution.sh
index ef7c010e930..376ae239a66 100755
--- a/dev/make-distribution.sh
+++ b/dev/make-distribution.sh
@@ -36,7 +36,7 @@ MAKE_TGZ=false
 MAKE_PIP=false
 MAKE_R=false
 NAME=none
-MVN="$SPARK_HOME/build/mvn"
+MVN="mvn"
 
-BUILD_COMMAND=("$MVN" clean package \
+BUILD_COMMAND=("$MVN" -T$(grep -c processor /proc/cpuinfo) clean package \
     -DskipTests \
     -Dmaven.javadoc.skip=true \
     -Dmaven.scaladoc.skip=true \