diff --git a/Mavenfile b/Mavenfile
index 80c88b0..f42101f 100644
--- a/Mavenfile
+++ b/Mavenfile
@@ -12,11 +12,11 @@ properties 'project.build.sourceEncoding' => 'UTF-8',
            # create a pom.xml from this here
            'polyglot.dump.pom' => 'pom.xml'

-jar 'junit:junit', '4.11', :scope => :test
+jar 'junit:junit', '4.13.1', :scope => :test

 jar 'org.jruby:jruby', '9.2.13.0', :scope => :provided

 plugin :compiler, '3.1', :source => '1.8', :target => '1.8',
        :showDeprecation => false,
        :showWarnings => false,
        :executable => '${JAVA_HOME}/bin/javac',
