diff --git a/WORKSPACE b/WORKSPACE
index 5410299..3d1ec39 100644
--- a/WORKSPACE
+++ b/WORKSPACE
@@ -69,6 +69,37 @@ load("@rules_proto//proto:repositories.bzl", "rules_proto_dependencies", "rules_
 rules_proto_dependencies()
 rules_proto_toolchains()
 
+# m4 is used and fetched by zetasql but we want to patch it so load it before calling zetasql_deps_step_2 
+http_archive(
+    name = "m4",
+    build_file_content = """filegroup(name = "all", srcs = glob(["**"]), visibility = ["//visibility:public"])""",
+    strip_prefix = "m4-1.4.18",
+    sha256 = "ab2633921a5cd38e48797bf5521ad259bdc4b979078034a3b790d7fec5493fab",
+    urls = [
+        "https://ftp.gnu.org/gnu/m4/m4-1.4.18.tar.gz",
+        "https://mirrors.kernel.org/gnu/m4/m4-1.4.18.tar.gz",
+    ],
+    patches = [
+        "@com_google_zetasql//bazel:m4.patch",
+        "@//third_party:m4-cstack.patch"
+    ],
+)
+
+
+# Needed by abseil-py by zetasql.
+[
+    http_archive(
+        name = name,
+        urls = [
+            "http://mirror.bazel.build/pypi.python.org/packages/source/s/six/six-1.16.0.tar.gz",
+            "https://pypi.python.org/packages/source/s/six/six-1.16.0.tar.gz",
+        ],
+        sha256 = "1e61c37477a1626458e36f7b1d82aa5c9b094fa4802892072e49de9c60c4c926",
+        build_file = "@//third_party:six.BUILD",
+    )
+    for name in ["six_archive", "six"]
+]
+
 # TODO(b/239095455): Change to using a tfx-bsl workspace macro to load these
 # dependencies.
 # Needed by zetasql.
@@ -83,17 +114,6 @@ http_archive(
     ],
 )
 
-# Needed by abseil-py by zetasql.
-http_archive(
-    name = "six_archive",
-    urls = [
-        "http://mirror.bazel.build/pypi.python.org/packages/source/s/six/six-1.10.0.tar.gz",
-        "https://pypi.python.org/packages/source/s/six/six-1.10.0.tar.gz",
-    ],
-    sha256 = "105f8d68616f8248e24bf0e9372ef04d3cc10104f1980f54d57b2ce73a5ad56a",
-    strip_prefix = "six-1.10.0",
-    build_file = "//third_party:six.BUILD"
-)
 
 COM_GOOGLE_ABSL_COMMIT = "92fdbfb301f8b301b28ab5c99e7361e775c2fb8a" # 2022-08-25 Abseil Logging library first release
 http_archive(
@@ -144,19 +164,25 @@ http_archive(
 
 # Please add all new TensorFlow Data Validation dependencies in workspace.bzl.
 load("//tensorflow_data_validation:workspace.bzl", "tf_data_validation_workspace")
-
 tf_data_validation_workspace()
 
-load("@com_github_tfx_bsl//third_party:python_configure.bzl", "local_python_configure")
-local_python_configure(name = "local_config_python")
 
-PYBIND11_COMMIT = "f1abf5d9159b805674197f6bc443592e631c9130"
 http_archive(
-  name = "pybind11",
-  build_file = "@com_github_tfx_bsl//third_party:pybind11.BUILD",
-  strip_prefix = "pybind11-%s" % PYBIND11_COMMIT,
-  urls = ["https://github.com/pybind/pybind11/archive/%s.zip" % PYBIND11_COMMIT],
-  sha256 = "4972f216f17f35e19d0afe54b0f30fe80ab1a7e57b65328530388285f36c7533",
+  name = "pybind11_bazel",
+  strip_prefix = "pybind11_bazel-23926b00e2b2eb2fc46b17e587cf0c0cfd2f2c4b",
+  urls = ["https://github.com/pybind/pybind11_bazel/archive/23926b00e2b2eb2fc46b17e587cf0c0cfd2f2c4b.zip"],
+  sha256 = "07e529a85cf4c11e1ca1b423149e86e63a3f3859c22efee3b3c5225ca89580f2"
+)
+load("@pybind11_bazel//:python_configure.bzl", "python_configure")
+python_configure(name = "local_config_python")
+
+PYBIND11_COMMIT = "8a099e44b3d5f85b20f05828d919d2332a8de841"
+http_archive(
+    name = "pybind11",
+    build_file = "@pybind11_bazel//:pybind11.BUILD",
+    strip_prefix = "pybind11-%s" % PYBIND11_COMMIT,
+    urls = ["https://github.com/pybind/pybind11/archive/%s.zip" % PYBIND11_COMMIT],
+    sha256 = "8f4b7f28d214e36301435c055076c36186388dc9617117802cba8a059347cb00"
 )
 
 # Specify the minimum required bazel version.
diff --git a/tensorflow_data_validation/workspace.bzl b/tensorflow_data_validation/workspace.bzl
index c0d6f4b..e7bce59 100644
--- a/tensorflow_data_validation/workspace.bzl
+++ b/tensorflow_data_validation/workspace.bzl
@@ -21,6 +21,7 @@ def tf_data_validation_workspace():
 
     git_repository(
         name = "com_github_tfx_bsl",
-        branch = "master",
+        commit = "63ecbd69b30d38a0ff9b4fb26d0ecdfe153162d4", 
+        shallow_since = "1701987236 -0800",
         remote = "https://github.com/tensorflow/tfx-bsl",
     )
diff --git a/third_party/m4-cstack.patch b/third_party/m4-cstack.patch
new file mode 100644
index 0000000..9fedf70
--- /dev/null
+++ b/third_party/m4-cstack.patch
@@ -0,0 +1,40 @@
+--- lib/c-stack.c
++++ lib/c-stack2.c
+@@ -50,14 +50,14 @@
+ #if ! HAVE_STACK_T && ! defined stack_t
+ typedef struct sigaltstack stack_t;
+ #endif
+-#ifndef SIGSTKSZ
+-# define SIGSTKSZ 16384
+-#elif HAVE_LIBSIGSEGV && SIGSTKSZ < 16384
++#ifndef GNULIB_SIGSTKSZ
++# define GNULIB_SIGSTKSZ 16384
++#elif HAVE_LIBSIGSEGV && GNULIB_SIGSTKSZ < 16384
+ /* libsigsegv 2.6 through 2.8 have a bug where some architectures use
+    more than the Linux default of an 8k alternate stack when deciding
+    if a fault was caused by stack overflow.  */
+-# undef SIGSTKSZ
+-# define SIGSTKSZ 16384
++# undef GNULIB_SIGSTKSZ
++# define GNULIB_SIGSTKSZ 16384
+ #endif
+ 
+ #include <stdlib.h>
+@@ -131,7 +131,7 @@
+ /* Storage for the alternate signal stack.  */
+ static union
+ {
+-  char buffer[SIGSTKSZ];
++  char buffer[GNULIB_SIGSTKSZ];
+ 
+   /* These other members are for proper alignment.  There's no
+      standard way to guarantee stack alignment, but this seems enough
+@@ -282,7 +282,7 @@
+ # if SIGALTSTACK_SS_REVERSED
+   /* Irix mistakenly treats ss_sp as the upper bound, rather than
+      lower bound, of the alternate stack.  */
+-  st.ss_sp = alternate_signal_stack.buffer + SIGSTKSZ - sizeof (void *);
++  st.ss_sp = alternate_signal_stack.buffer + GNULIB_SIGSTKSZ - sizeof (void *);
+   st.ss_size = sizeof alternate_signal_stack.buffer - sizeof (void *);
+ # else
+   st.ss_sp = alternate_signal_stack.buffer;
diff --git a/third_party/six.BUILD b/third_party/six.BUILD
index 206a160..5059af2 100644
--- a/third_party/six.BUILD
+++ b/third_party/six.BUILD
@@ -8,9 +8,22 @@ licenses(["notice"])  # MIT
 
 exports_files(["LICENSE"])
 
+load("@rules_python//python:defs.bzl", "py_library")
+
+# Consume `six.py` as `__init__.py` for compatibility
+# with `--incompatible_default_to_explicit_init_py`.
+# https://github.com/protocolbuffers/protobuf/pull/6795#issuecomment-546060749
+# https://github.com/bazelbuild/bazel/issues/10076
+genrule(
+    name = "copy_six",
+    srcs = ["six-1.16.0/six.py"],
+    outs = ["__init__.py"],
+    cmd = "cp $< $(@)",
+)
+
 py_library(
     name = "six",
-    srcs = ["six.py"],
+    srcs = ["__init__.py"],
     srcs_version = "PY2AND3",
     visibility = ["//visibility:public"],
-)
+)
