From 87021bd2cbf98674230b5f2af0ac881a49b802cb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Furkan=20T=C3=BCrkal?= <furkan.turkal@chainguard.dev>
Date: Fri, 17 Nov 2023 11:08:46 +0300
Subject: [PATCH 1/2] bump-abseil-six-rules-foreign-cc-versions
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Furkan Türkal <furkan.turkal@chainguard.dev>
---
 WORKSPACE | 23 ++++++++++++-----------
 1 file changed, 12 insertions(+), 11 deletions(-)

diff --git a/WORKSPACE b/WORKSPACE
index ae7efb2..67cf578 100644
--- a/WORKSPACE
+++ b/WORKSPACE
@@ -2,18 +2,18 @@ workspace(name = "tensorflow_data_validation")
 
 load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
 
-# Install version 0.9.0 of rules_foreign_cc, as default version causes an
+# Install version 0.10.1 of rules_foreign_cc, as default version causes an
 # invalid escape sequence error to be raised, which can't be avoided with
 # the --incompatible_restrict_string_escapes=false flag (flag was removed in
 # Bazel 5.0).
-RULES_FOREIGN_CC_VERSION = "0.9.0"
+RULES_FOREIGN_CC_VERSION = "0.10.1"
 http_archive(
     name = "rules_foreign_cc",
-    sha256 = "2a4d07cd64b0719b39a7c12218a3e507672b82a97b98c6a89d38565894cf7c51",
+    sha256 = "476303bd0f1b04cc311fc258f1708a5f6ef82d3091e53fd1977fa20383425a6a",
     strip_prefix = "rules_foreign_cc-%s" % RULES_FOREIGN_CC_VERSION,
     url = "https://github.com/bazelbuild/rules_foreign_cc/archive/refs/tags/%s.tar.gz" % RULES_FOREIGN_CC_VERSION,
-    patch_tool = "patch",
-    patches = ["//third_party:rules_foreign_cc.patch",],
+    # patch_tool = "patch",
+    # patches = ["//third_party:rules_foreign_cc.patch",],
 )
 
 load("@rules_foreign_cc//foreign_cc:repositories.bzl", "rules_foreign_cc_dependencies")
@@ -87,19 +87,18 @@ http_archive(
 http_archive(
     name = "six_archive",
     urls = [
-        "http://mirror.bazel.build/pypi.python.org/packages/source/s/six/six-1.10.0.tar.gz",
-        "https://pypi.python.org/packages/source/s/six/six-1.10.0.tar.gz",
+        "https://pypi.python.org/packages/source/s/six/six-1.16.0.tar.gz",
     ],
-    sha256 = "105f8d68616f8248e24bf0e9372ef04d3cc10104f1980f54d57b2ce73a5ad56a",
-    strip_prefix = "six-1.10.0",
+    sha256 = "1e61c37477a1626458e36f7b1d82aa5c9b094fa4802892072e49de9c60c4c926",
+    strip_prefix = "six-1.16.0",
     build_file = "//third_party:six.BUILD"
 )
 
-COM_GOOGLE_ABSL_COMMIT = "92fdbfb301f8b301b28ab5c99e7361e775c2fb8a" # 2022-08-25 Abseil Logging library first release
+COM_GOOGLE_ABSL_COMMIT = "fb3621f4f897824c0dbe0615fa94543df6192f30" # Abseil LTS branch, Aug 2023, Patch 1
 http_archive(
   name = "com_google_absl",
   url = "https://github.com/abseil/abseil-cpp/archive/%s.tar.gz" % COM_GOOGLE_ABSL_COMMIT,
-  sha256 = "71d38c5f44997a5ccbc338f904c8682b40c25cad60b9cbaf27087a917228d5fa",
+  sha256 = "0320586856674d16b0b7a4d4afb22151bdc798490bb7f295eddd8f6a62b46fea",
   strip_prefix = "abseil-cpp-%s" % COM_GOOGLE_ABSL_COMMIT
 )
 
@@ -109,6 +108,8 @@ http_archive(
     urls = ["https://github.com/google/zetasql/archive/%s.zip" % ZETASQL_COMMIT],
     strip_prefix = "zetasql-%s" % ZETASQL_COMMIT,
     sha256 = "651a768cd51627f58aa6de7039aba9ddab22f4b0450521169800555269447840",
+    patch_tool = "patch",
+    patches = ["//third_party:bump-six.patch"],
 )
 
 load("@com_google_zetasql//bazel:zetasql_deps_step_1.bzl", "zetasql_deps_step_1")
-- 
2.39.2 (Apple Git-143)

