Description: Disable Clang >= 20 compiler options until packaged
Author: James Page <james.page@chainguard.dev>

--- a/build/config/compiler/BUILD.gn.orig	2025-05-12 16:11:29.949768459 +0100
+++ b/build/config/compiler/BUILD.gn	2025-05-12 16:12:14.385296607 +0100
@@ -312,7 +312,6 @@ config("compiler") {
     ":compiler_cpu_abi",
     ":compiler_codegen",
     ":compiler_deterministic",
-    ":clang_warning_suppression",
   ]
 
   # Here we enable -fno-delete-null-pointer-checks, which makes various nullptr
