Unknown argument as of clang 22.1.5.

diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index 0223ddf..baaf709 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -1910,12 +1910,6 @@ config("sanitize_return") {
     cflags = [
       "-fsanitize=return",
       "-fsanitize-trap=return",
-
-      # Some code users feature detection to determine if UBSAN (or any
-      # sanitizer) is enabled, they then do expensive debug like operations. We
-      # want to suppress this behaviour since we want to keep performance costs
-      # as low as possible while having these checks.
-      "-fsanitize-ignore-for-ubsan-feature=return",
     ]
   }
 }
