diff --git a/gems/bundled_gems b/gems/bundled_gems
index c48acb50d8..c667b4ca14 100644
--- a/gems/bundled_gems
+++ b/gems/bundled_gems
@@ -3,10 +3,8 @@ minitest        5.16.3  https://github.com/seattlerb/minitest
 power_assert    2.0.3   https://github.com/ruby/power_assert
 rake            13.0.6  https://github.com/ruby/rake
 test-unit       3.5.7   https://github.com/test-unit/test-unit
-rexml           3.3.9   https://github.com/ruby/rexml
 rss             0.3.1   https://github.com/ruby/rss
 net-ftp         0.2.1   https://github.com/ruby/net-ftp
-net-imap        0.3.4.1   https://github.com/ruby/net-imap
 net-pop         0.1.2   https://github.com/ruby/net-pop
 net-smtp        0.3.4   https://github.com/ruby/net-smtp
 matrix          0.4.2   https://github.com/ruby/matrix
