git diff
diff --git a/build/repositories.bzl b/build/repositories.bzl
index 4004b32f7..08a5d40cd 100644
--- a/build/repositories.bzl
+++ b/build/repositories.bzl
@@ -50,12 +50,9 @@ def github_cli_repositories():
         )

 def kong_github_repositories():
-    maybe(
-        github_release,
+    http_archive(
         name = "kong_admin_gui",
-        repo = "kong/kong-manager",
-        tag = KONG_VAR["KONG_MANAGER"],
-        pattern = "release.tar.gz",
+        url = "https://github.com/Kong/kong-manager/releases/download/v3.9.0.0/release.tar.gz",
         build_file_content = _DIST_BUILD_FILE_CONTENT,
     )
