diff --git a/build/repositories.bzl b/build/repositories.bzl
index 4004b32..875018e 100644
--- a/build/repositories.bzl
+++ b/build/repositories.bzl
@@ -50,12 +50,10 @@ def github_cli_repositories():
         )
 
 def kong_github_repositories():
-    maybe(
-        github_release,
+    http_archive(
         name = "kong_admin_gui",
-        repo = "kong/kong-manager",
-        tag = KONG_VAR["KONG_MANAGER"],
-        pattern = "release.tar.gz",
+        url = "https://github.com/Kong/kong-manager/releases/download/%s/release.tar.gz" % (KONG_VAR["KONG_MANAGER"]),
+        sha256 = "0e5028f8ecfcc611de8d0a015a1985d46fe207d09a87ffe644a2d19b189db70a,
         build_file_content = _DIST_BUILD_FILE_CONTENT,
     )
 
