diff --git a/build/BUILD.bazel b/build/BUILD.bazel
index b83f4f4..3b7810f 100644
--- a/build/BUILD.bazel
+++ b/build/BUILD.bazel
@@ -10,7 +10,7 @@ exports_files([
 # C libraries
 
 clib_deps = [
-    "@openssl",
+    # "@openssl",
     "@libexpat",
     "@snappy",
     "@ada",
@@ -47,7 +47,7 @@ kong_rules_group(
 kong_rules_group(
     name = "cacheable-targets",
     propagates = [
-        "@openssl",
+        # "@openssl",
         "@libexpat",
         "@atc_router",
         "@simdjson_ffi",
@@ -73,7 +73,7 @@ kong_install(
     prefix = "openresty",
     deps = [
         ":install-openresty-luajit",
-        ":install-openssl",
+        # ":install-openssl",
     ],
 )
 
@@ -241,7 +241,7 @@ kong_rules_group(
         ":install-clibs",
         ":install-lualibs",
         ":install-wasmx",
-        ":install-openresty",
+        # ":install-openresty",
         ":install-static-assets",
         ":install-tools",
     ],
diff --git a/build/luarocks/BUILD.luarocks.bazel b/build/luarocks/BUILD.luarocks.bazel
index b23ad96..50006bc 100644
--- a/build/luarocks/BUILD.luarocks.bazel
+++ b/build/luarocks/BUILD.luarocks.bazel
@@ -40,7 +40,7 @@ kong_template_genrule(
     name = "luarocks_exec",
     srcs = [
         "@libexpat",
-        "@openssl",
+        # "@openssl",
     ] + select({
         "@kong//:any-cross": ["@cross_deps_libyaml//:libyaml"],
         "//conditions:default": [
diff --git a/build/luarocks/templates/luarocks_exec.sh b/build/luarocks/templates/luarocks_exec.sh
index 10515f8..437dee6 100644
--- a/build/luarocks/templates/luarocks_exec.sh
+++ b/build/luarocks/templates/luarocks_exec.sh
@@ -27,7 +27,8 @@ ROCKS_CONFIG=$(readlink -f "$ROCKS_DIR/../luarocks_config.lua")
 
 EXPAT_DIR=$root_path/$libexpat_path
 LIBXML2_DIR=$root_path/$libxml2_path
-OPENSSL_DIR=$root_path/$openssl_path
+# OPENSSL_DIR=$root_path/$openssl_path
+OPENSSL_DIR="/usr"
 
 # The Bazel rules doesn't export the `libexpat.so` file,
 # it only exports something like `libexpat.so.1.6.0`,
diff --git a/build/openresty/BUILD.openresty.bazel b/build/openresty/BUILD.openresty.bazel
index 81ad172..e3ddf0f 100644
--- a/build/openresty/BUILD.openresty.bazel
+++ b/build/openresty/BUILD.openresty.bazel
@@ -125,6 +125,7 @@ selects.config_setting_group(
 
 CONFIGURE_OPTIONS = [
     "--with-pcre-jit",
+    "--with-openssl=/usr",
     "--with-http_ssl_module",
     "--with-http_sub_module",
     "--with-http_realip_module",
@@ -147,10 +148,10 @@ CONFIGURE_OPTIONS = [
     "--without-http_rds_csv_module",
     "--with-luajit=$$EXT_BUILD_DEPS/luajit",
     "--with-cc-opt=\"-I$$EXT_BUILD_DEPS/pcre/include\"",
-    "--with-cc-opt=\"-I$$EXT_BUILD_DEPS/openssl/include\"",
+    # "--with-cc-opt=\"-I$$EXT_BUILD_DEPS/openssl/include\"",
     "--with-cc-opt=\"-I$$EXT_BUILD_DEPS/luajit/include\"",
     "--with-ld-opt=\"-L$$EXT_BUILD_DEPS/pcre/lib\"",
-    "--with-ld-opt=\"-L$$EXT_BUILD_DEPS/openssl/lib\"",
+    # "--with-ld-opt=\"-L$$EXT_BUILD_DEPS/openssl/lib\"",
     "--with-ld-opt=\"-L$$EXT_BUILD_DEPS/luajit/lib\"",
     "--with-ld-opt=\"-L$$EXT_BUILD_DEPS/lib\"",
     # Here let's try not having --disable-new-dtags; --disable-new-dtags creates rpath instead of runpath
@@ -326,7 +327,7 @@ configure_make(
     visibility = ["//visibility:public"],
     deps = [
         "@openresty//:luajit",
-        "@openssl",
+        # "@openssl",
         "@pcre",
     ] + select({
         "@kong//:any-cross": [
diff --git a/build/openresty/openssl/BUILD.bazel b/build/openresty/openssl/BUILD.bazel
index 5970c67..3a475c0 100644
--- a/build/openresty/openssl/BUILD.bazel
+++ b/build/openresty/openssl/BUILD.bazel
@@ -1,5 +1,5 @@
-load("@kong//build/openresty/openssl:openssl.bzl", "build_openssl")
+# load("@kong//build/openresty/openssl:openssl.bzl", "build_openssl")
 
-build_openssl(
-    name = "openssl",
-)
+# build_openssl(
+#     name = "openssl",
+# )
diff --git a/build/repositories.bzl b/build/repositories.bzl
index 4004b32..7a00930 100644
--- a/build/repositories.bzl
+++ b/build/repositories.bzl
@@ -50,12 +50,9 @@ def github_cli_repositories():
         )
 
 def kong_github_repositories():
-    maybe(
-        github_release,
+    http_archive(
         name = "kong_admin_gui",
-        repo = "kong/kong-manager",
-        tag = KONG_VAR["KONG_MANAGER"],
-        pattern = "release.tar.gz",
+        url = "https://github.com/Kong/kong-manager/releases/download/nightly/release.tar.gz",
         build_file_content = _DIST_BUILD_FILE_CONTENT,
     )
 

