From 9b6d7c5e8b23e20e60d03d85265e284679d304a6 Mon Sep 17 00:00:00 2001
From: Kyle Steere <kyle.steere@chainguard.dev>
Date: Fri, 24 Jan 2025 08:05:23 -0600
Subject: [PATCH] fix GHSA-c76h-2ccp-4975

---
 package.json | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/package.json b/package.json
index 0c0ec8c22..6508ed552 100644
--- a/package.json
+++ b/package.json
@@ -66,5 +66,8 @@
     "services/*",
     "cli/*",
     "libs/*"
-  ]
+  ],
+  "resolutions": {
+    "undici": "^6.21.1"
+  }
 }
--
2.43.0
