diff --git a/melange-out/kafka-iamguarded-compat-4.2/opt/iamguarded/scripts/libkafka.sh b/melange-out/kafka-iamguarded-compat-4.2/opt/iamguarded/scripts/libkafka.sh
index 9eabf0e13da..3ddf81fbf11 100644
--- a/melange-out/kafka-iamguarded-compat-4.2/opt/iamguarded/scripts/libkafka.sh
+++ b/melange-out/kafka-iamguarded-compat-4.2/opt/iamguarded/scripts/libkafka.sh
@@ -920,7 +920,7 @@ kafka_kraft_storage_initialize() {
             args+=("--add-scram" "SCRAM-SHA-512=[name=${KAFKA_CONTROLLER_USER},password=${KAFKA_CONTROLLER_PASSWORD}]")
         fi
     fi
-    if kafka_is_dynamic_controller_quorum_supported && [[ "${KAFKA_CFG_PROCESS_ROLES:-}" =~ "controller" ]] && [[ "${KAFKA_KRAFT_VERSION:-}" -eq "1" ]]; then
+    if kafka_is_dynamic_controller_quorum_supported && [[ "${KAFKA_CFG_PROCESS_ROLES:-}" =~ "controller" ]] && [[ "${KAFKA_KRAFT_VERSION:-}" -eq "1" ]] && ! grep -q "^controller\.quorum\.voters=" "$KAFKA_CONF_FILE"; then
         args+=("--feature=kraft.version=1")
         if [[ -n "${KAFKA_INITIAL_CONTROLLERS:-}" ]]; then
             args+=("--initial-controllers=${KAFKA_INITIAL_CONTROLLERS}")
