diff --git a/src/python/CMakeLists.txt b/src/python/CMakeLists.txt
index 43c8eb2..5e0cd3c 100644
--- a/src/python/CMakeLists.txt
+++ b/src/python/CMakeLists.txt
@@ -13,7 +13,7 @@ if(NOT Python3_FOUND)
     message(FATAL_ERROR "Could not find Python")
 endif()
 
-find_package(Boost CONFIG REQUIRED COMPONENTS python)
+find_package(Boost CONFIG REQUIRED COMPONENTS python${Python3_VERSION_MAJOR}${Python3_VERSION_MINOR})
 
 if(NOT Boost_FOUND)
     message(FATAL_ERROR "Could not find Boost")
diff --git a/src/python/PyImathTest/CMakeLists.txt b/src/python/PyImathTest/CMakeLists.txt
index af16fc4..b66eb9e 100644
--- a/src/python/PyImathTest/CMakeLists.txt
+++ b/src/python/PyImathTest/CMakeLists.txt
@@ -30,7 +30,7 @@ if("${CMAKE_PROJECT_NAME}" STREQUAL "")
   project(PyImathTest)
 
   find_package(Python3 REQUIRED COMPONENTS Interpreter Development)
-  find_package(Boost REQUIRED COMPONENTS python)
+  find_package(Boost REQUIRED COMPONENTS python${Python3_VERSION_MAJOR}${Python3_VERSION_MINOR})
 
   find_package(Imath COMPONENTS PyImath)
 endif()
diff --git a/website/examples/quickstart.pyimath/CMakeLists.txt b/website/examples/quickstart.pyimath/CMakeLists.txt
index c0d80fd..1e6531b 100644
--- a/website/examples/quickstart.pyimath/CMakeLists.txt
+++ b/website/examples/quickstart.pyimath/CMakeLists.txt
@@ -4,7 +4,7 @@ cmake_minimum_required(VERSION 3.14)
 project(pyimath_program)
 
 find_package(Python3 REQUIRED COMPONENTS Interpreter Development)
-find_package(Boost REQUIRED COMPONENTS python)
+find_package(Boost REQUIRED COMPONENTS python${Python3_VERSION_MAJOR}${Python3_VERSION_MINOR})
 
 find_package(Imath COMPONENTS PyImath)
