diff --git a/versions.props b/versions.props
index 6becf5a93c8..c3d5b412a41 100644
--- a/versions.props
+++ b/versions.props
@@ -1,5 +1,8 @@
 # The lines in this file needs to be lexicographically sorted.
 # Please only add direct dependencies. Overrides of transitive versions should be called out in a comment.
+
+org.apache.hadoop:hadoop-client-runtime=3.3.6
+
 biz.aQute.bnd:biz.aQute.bnd.annotation=6.4.1
 com.adobe.testing:s3mock-junit4=2.17.0
 com.carrotsearch.randomizedtesting:*=2.8.1
@@ -23,7 +26,7 @@ commons-io:commons-io=2.14.0
 io.dropwizard.metrics:*=4.2.20
 io.grpc:grpc-*=1.56.0
 io.jaegertracing:*=1.8.1
-io.netty:*=4.1.99.Final
+io.netty:*=4.1.100.Final
 io.opentelemetry:opentelemetry-bom=1.30.1
 io.opentracing:*=0.33.0
 io.prometheus:*=0.16.0
@@ -52,7 +55,7 @@ org.apache.logging.log4j:*=2.20.0
 org.apache.lucene:*=9.8.0
 org.apache.tika:*=1.28.5
 org.apache.tomcat:annotations-api=6.0.53
-org.apache.zookeeper:*=3.9.0
+org.apache.zookeeper:*=3.9.1
 org.bitbucket.b_c:jose4j=0.9.3
 org.carrot2:carrot2-core=4.5.1
 org.codehaus.woodstox:stax2-api=4.2.1
