diff --git a/rakelib/artifacts.rake b/rakelib/artifacts.rake
index 88e765e93..cb58e45f8 100644
--- a/rakelib/artifacts.rake
+++ b/rakelib/artifacts.rake
@@ -197,8 +197,8 @@ namespace "artifact" do
     #with bundled JDKs
     @bundles_jdk = true
     license_details = ['APACHE-LICENSE-2.0', "-oss", oss_exclude_paths]
-    create_archive_pack(license_details, "x86_64", "linux", "windows", "darwin")
-    create_archive_pack(license_details, "arm64", "linux", "darwin")
+    create_archive_pack(license_details, "x86_64", "linux")
+    create_archive_pack(license_details, "arm64", "linux")
 
     #without JDK
     @bundles_jdk = false
