diff --git a/subprojects/core/src/integTest/groovy/org/gradle/internal/classpath/BuildScriptClasspathIntegrationSpec.groovy b/subprojects/core/src/integTest/groovy/org/gradle/internal/classpath/BuildScriptClasspathIntegrationSpec.groovy
index 62c041ec9d1..97b7089f565 100644
--- a/subprojects/core/src/integTest/groovy/org/gradle/internal/classpath/BuildScriptClasspathIntegrationSpec.groovy
+++ b/subprojects/core/src/integTest/groovy/org/gradle/internal/classpath/BuildScriptClasspathIntegrationSpec.groovy
@@ -311,7 +311,7 @@ class BuildScriptClasspathIntegrationSpec extends AbstractIntegrationSpec implem
             buildscript {
                 ${mavenCentralRepository()}
                 dependencies {
-                    classpath("org.bouncycastle:bcprov-jdk18on:1.77")
+                    classpath("org.bouncycastle:bcprov-jdk18on:1.78")
                 }
             }
 
diff --git a/subprojects/distributions-dependencies/build.gradle.kts b/subprojects/distributions-dependencies/build.gradle.kts
index 07519bb239a..a79756cc902 100644
--- a/subprojects/distributions-dependencies/build.gradle.kts
+++ b/subprojects/distributions-dependencies/build.gradle.kts
@@ -21,7 +21,7 @@ val antVersion = "1.10.13"
 // when upgrading ASM.
 val asmVersion = "9.7"
 val awsS3Version = "1.12.651"
-val bouncycastleVersion = "1.77"
+val bouncycastleVersion = "1.78"
 val jacksonVersion = "2.16.1"
 val jaxbVersion = "3.0.0"
 val junit5Version = "5.8.2"
