diff --git a/webapp/Makefile b/webapp/Makefile
index da56ad1..390690d 100644
--- a/webapp/Makefile
+++ b/webapp/Makefile
@@ -8,12 +8,12 @@ CI ?= false
 # please see optipng-bin Linux arm64 support issue (https://github.com/imagemin/optipng-bin/issues/118) for details:
 ifeq ($(shell uname)/$(shell uname -m),Linux/aarch64)
   LINUX_ARM64 = true
-  CPPFLAGS += " -DPNG_ARM_NEON_OPT=0"
+  CPPFLAGS += -DPNG_ARM_NEON_OPT=0
 endif
 # Exact same issue but for Linux/PPC64
 ifeq ($(findstring Linux/ppc64,$(shell uname)/$(shell uname -m)),Linux/ppc64)
   LINUX_PPC64 = true
-  CPPFLAGS += " -DPNG_POWERPC_VSX_OPT=0"
+  CPPFLAGS += -DPNG_POWERPC_VSX_OPT=0
 endif
 
 .PHONY: run
