diff --git a/datadog_checks_dev/pyproject.toml b/datadog_checks_dev/pyproject.toml
index 260e4dc642..8f66624770 100644
--- a/datadog_checks_dev/pyproject.toml
+++ b/datadog_checks_dev/pyproject.toml
@@ -68,7 +68,7 @@ cli = [
     "pip-tools",
     "pathspec>=0.10.0",
     "platformdirs>=2.0.0a3",
-    "pydantic>=2.0.2",
+    "pydantic>=2.4.0",
     "pysmi==0.3.4",
     "securesystemslib[crypto]==0.28.0",
     "semver>=2.13.0",
