diff --git a/LICENSE-binary b/LICENSE-binary
index 5a0642b4d9..5e966ebd3f 100644
--- a/LICENSE-binary
+++ b/LICENSE-binary
@@ -263,8 +263,8 @@ scala-reflect-2.13.11
 scala-java8-compat_2.13-1.0.2
 snappy-java-1.1.10.4
 swagger-annotations-2.2.8
-zookeeper-3.8.2
-zookeeper-jute-3.8.2
+zookeeper-3.8.3
+zookeeper-jute-3.8.3
 
 ===============================================================================
 This product bundles various third-party components under other open source
diff --git a/gradle/dependencies.gradle b/gradle/dependencies.gradle
index 6994f994e4..3ff1af47b6 100644
--- a/gradle/dependencies.gradle
+++ b/gradle/dependencies.gradle
@@ -148,7 +148,7 @@ versions += [
   swaggerAnnotations: "2.2.8",
   swaggerJaxrs2: "2.2.8",
   zinc: "1.9.2",
-  zookeeper: "3.8.2",
+  zookeeper: "3.8.3",
   zstd: "1.5.5-1"
 ]
 
