This patches a handful of CVEs:

* CVE-2023-36479
* CVE-2023-40167
* CVE-2023-41900

---
diff --git a/LICENSE-binary b/LICENSE-binary
index 233c99ecb4..10ef6d7037 100644
--- a/LICENSE-binary
+++ b/LICENSE-binary
@@ -220,16 +220,16 @@ jackson-module-scala_2.13-2.13.5
 jackson-module-scala_2.12-2.13.5
 jakarta.validation-api-2.0.2
 javassist-3.29.2-GA
-jetty-client-9.4.51.v20230217
-jetty-continuation-9.4.51.v20230217
-jetty-http-9.4.51.v20230217
-jetty-io-9.4.51.v20230217
-jetty-security-9.4.51.v20230217
-jetty-server-9.4.51.v20230217
-jetty-servlet-9.4.51.v20230217
-jetty-servlets-9.4.51.v20230217
-jetty-util-9.4.51.v20230217
-jetty-util-ajax-9.4.51.v20230217
+jetty-client-9.4.52.v20230823
+jetty-continuation-9.4.52.v20230823
+jetty-http-9.4.52.v20230823
+jetty-io-9.4.52.v20230823
+jetty-security-9.4.52.v20230823
+jetty-server-9.4.52.v20230823
+jetty-servlet-9.4.52.v20230823
+jetty-servlets-9.4.52.v20230823
+jetty-util-9.4.52.v20230823
+jetty-util-ajax-9.4.52.v20230823
 jose4j-0.9.3
 lz4-java-1.8.0
 maven-artifact-3.8.8
diff --git a/gradle/dependencies.gradle b/gradle/dependencies.gradle
index 28d17ac138..1509065e22 100644
--- a/gradle/dependencies.gradle
+++ b/gradle/dependencies.gradle
@@ -72,7 +72,7 @@ versions += [
   jacksonDatabind: "2.13.5",
   jacoco: "0.8.10",
   javassist: "3.29.2-GA",
-  jetty: "9.4.51.v20230217",
+  jetty: "9.4.52.v20230823",
   jersey: "2.39.1",
   jline: "3.22.0",
   jmh: "1.36",
