Author: Chris Herborth <chris.herborth@chainguard.dev>
Forwarded: not-needed
Description: Skip downloading Mac/Windows JDKs during builds.
---
diff --git a/build.gradle b/build.gradle
index 483d3b4..0635b72 100644
--- a/build.gradle
+++ b/build.gradle
@@ -318,8 +318,10 @@ task dockerBootstrap {
 
 tasks.register("installDefaultGems") {
     dependsOn bootstrap
+    dependsOn "logstash-core:assemble"
+    dependsOn "installDevelopmentGems"
     doLast {
         rake(projectDir, buildDir, 'plugin:install-default')
     }
 }
 
@@ -1042,9 +1045,9 @@ tasks.register("lint") {
 def getTargetPlatforms() {
     String arch = selectArch()
     if (arch == "x86_64") {
-        return [[os: "linux", arch: "x86_64"], [os: "windows", arch: "x86_64"], [os: "darwin", arch: "x86_64"]]
+        return [[os: "linux", arch: "x86_64"]]
     } else {
-        return [[os: "linux", arch: "arm64"], [os: "darwin", arch: "arm64"]]
+        return [[os: "linux", arch: "arm64"]]
     }
 }
 
