Fix CVE: GHSA-857q-xmph-p2v5, GHSA-52xf-5p2m-9wrv
diff --git a/src/proxy/Cargo.toml b/src/proxy/Cargo.toml
index 0a6d0ad..1d92f38 100644
--- a/src/proxy/Cargo.toml
+++ b/src/proxy/Cargo.toml
@@ -19,9 +19,9 @@ log4rs = { version = "0", features = ["rolling_file_appender", "compound_policy"
 nix = { version = "0.26.2", features = ["signal"]}
 onc-rpc = "0.2.3"
 rand = "0.8.5"
-s2n-tls = "0.0"
-s2n-tls-tokio = "0.0"
-s2n-tls-sys = "0.0"
+s2n-tls = "0.3.0"
+s2n-tls-tokio = "0.3.0"
+s2n-tls-sys = "0.3.0"
 serde = {version="1.0.175",features=["derive"]}
 serde_ini = "0.2.0"
 thiserror = "1.0.44"
