--- lcms-1.19/include/lcms.h.old	2009-10-30 10:57:45.000000000 -0500
+++ lcms-1.19/include/lcms.h	2017-09-08 00:47:09.931999949 -0500
@@ -55,7 +55,7 @@
 
 // Some machines does not have a reliable 'swab' function. Usually
 // leave commented unless the testbed diagnoses the contrary.
-// #define USE_CUSTOM_SWAB   1
+#define USE_CUSTOM_SWAB   1
 
 // Uncomment this if your compiler supports inline
 #define USE_INLINE  1
@@ -64,7 +64,7 @@
 // #define USE_DEFAULT_FLOOR_CONVERSION  1
 
 // Uncomment this line on multithreading environments
-// #define USE_PTHREADS    1
+#define USE_PTHREADS    1
 
 // Uncomment this line if you want lcms to use the black point tag in profile, 
 // if commented, lcms will compute the black point by its own. 
