--- lcms-1.19.orig/include/lcms.h
+++ lcms-1.19/include/lcms.h
@@ -139,12 +139,8 @@
 #   define USE_BIG_ENDIAN      1
 #endif
 
-#if defined(__sgi__) || defined(__sgi) || defined(__powerpc__) || defined(sparc) || defined(__ppc__) || defined(__s390__) || defined(__s390x__)
+#if defined(__sgi__) || defined(__sgi) || defined(sparc) || defined(__s390__) || defined(__s390x__)
 #   define USE_BIG_ENDIAN      1
-#endif
-
-#if TARGET_CPU_PPC
-#   define USE_BIG_ENDIAN   1
 #endif
 
 #if macintosh
