From b070e6956827f0a6d3c6b7dc10dc577f2d0544de Mon Sep 17 00:00:00 2001
From: Aditya Tirmanwar <aditya.tirmanwar@chainguard.dev>
Date: Tue, 25 Nov 2025 22:11:58 +0000
Subject: [PATCH] Fix MD5 usage for FIPS compatibility

---
 pipenv/utils/resolver.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pipenv/utils/resolver.py b/pipenv/utils/resolver.py
index 57545113..be93f5e1 100644
--- a/pipenv/utils/resolver.py
+++ b/pipenv/utils/resolver.py
@@ -753,7 +753,7 @@ def _generate_resolution_cache_key(
     ]
 
     key_string = "|".join(key_components)
-    return hashlib.md5(key_string.encode()).hexdigest()
+    return hashlib.md5(key_string.encode(), usedforsecurity=False).hexdigest()
 
 
 def _should_use_resolution_cache(cache_key, clear):