diff --git a/Gemfile.lock b/Gemfile.lock
index 51f9566..c98a227 100644
--- a/Gemfile.lock
+++ b/Gemfile.lock
@@ -2,13 +2,13 @@ PATH
   remote: .
   specs:
     fluent-plugin-splunk-hec (1.3.2)
-      fluentd (>= 1.5)
-      json-jwt (~> 1.15.0)
-      multi_json (~> 1.13)
+      fluentd (>= 1.6)
+      json-jwt (~> 1.16.0)
+      multi_json (~> 1.15)
       net-http-persistent (~> 4.0)
       openid_connect (~> 1.1.8)
-      prometheus-client (>= 2.1.0)
-      rack-oauth2 (~> 1.19)
+      prometheus-client (>= 4.1.0)
+      rack-oauth2 (~> 2.2)
 
 GEM
   remote: https://rubygems.org/
diff --git a/fluent-plugin-splunk-hec.gemspec b/fluent-plugin-splunk-hec.gemspec
index abee22c..cf2dfff 100644
--- a/fluent-plugin-splunk-hec.gemspec
+++ b/fluent-plugin-splunk-hec.gemspec
@@ -33,13 +33,13 @@ Gem::Specification.new do |spec|
 
   spec.required_ruby_version = '>= 2.3.0'
 
-  spec.add_runtime_dependency 'fluentd', '>= 1.5'
-  spec.add_runtime_dependency 'multi_json', '~> 1.13'
+  spec.add_runtime_dependency 'fluentd', '>= 1.6'
+  spec.add_runtime_dependency 'multi_json', '~> 1.15'
   spec.add_runtime_dependency 'net-http-persistent', '~> 4.0'
   spec.add_runtime_dependency 'openid_connect', '~> 1.1.8'
-  spec.add_runtime_dependency 'prometheus-client', '>= 2.1.0'
-  spec.add_runtime_dependency 'json-jwt', '~> 1.15.0'
-  spec.add_runtime_dependency 'rack-oauth2', '~> 1.19'
+  spec.add_runtime_dependency 'prometheus-client', '>= 4.1.0'
+  spec.add_runtime_dependency 'json-jwt', '~> 1.16.0'
+  spec.add_runtime_dependency 'rack-oauth2', '~> 2.2'
 
 
   spec.add_development_dependency 'bundler', '~> 2.0'
