Upstream evidently don't use the system-provided Go codepath.

diff --git a/doc/scripts/process-native-protocol-specs-in-docker.sh b/doc/scripts/process-native-protocol-specs-in-docker.sh
index 05565c0..56472f3 100755
--- a/doc/scripts/process-native-protocol-specs-in-docker.sh
+++ b/doc/scripts/process-native-protocol-specs-in-docker.sh
@@ -107,7 +107,7 @@ git sparse-checkout set --no-cone /cqlprotodoc
 git checkout
 cd "${TMPDIR}/cassandra-website/cqlprotodoc"
 rm -rf "${TMPDIR}/cqlprotodoc"
-$TMPDIR/go$GO_VERSION/go/bin/go build -o "$TMPDIR"/cqlprotodoc
+go build -o "$TMPDIR"/cqlprotodoc
 
 # Step 2: Process the spec files using the parser
 echo "Processing the .spec files..."
