diff --git a/src/runtime/uv/event_loop.h b/src/runtime/uv/event_loop.h
index 0fd6517f41..c62a9f0986 100644
--- a/src/runtime/uv/event_loop.h
+++ b/src/runtime/uv/event_loop.h
@@ -5,6 +5,7 @@ Released under Apache 2.0 license as described in the file LICENSE.
 Author: Sofia Rodrigues
 */
 #pragma once
+#include <cmath>
 #include <lean/lean.h>
 #include "runtime/io.h"
 #include "runtime/object.h"
diff --git a/src/runtime/uv/net_addr.h b/src/runtime/uv/net_addr.h
index 9bfc2672d8..4415efd6ec 100644
--- a/src/runtime/uv/net_addr.h
+++ b/src/runtime/uv/net_addr.h
@@ -4,6 +4,7 @@ Released under Apache 2.0 license as described in the file LICENSE.
 Author: Henrik Böving
 */
 #pragma once
+#include <cmath>
 #include <lean/lean.h>
 #include "runtime/object.h"
 
diff --git a/src/runtime/uv/timer.h b/src/runtime/uv/timer.h
index 5edeae7af1..5b676cb67a 100644
--- a/src/runtime/uv/timer.h
+++ b/src/runtime/uv/timer.h
@@ -5,6 +5,7 @@ Released under Apache 2.0 license as described in the file LICENSE.
 Author: Sofia Rodrigues, Henrik Böving
 */
 #pragma once
+#include <cmath>
 #include <lean/lean.h>
 #include "runtime/uv/event_loop.h"
 
