From 2a9d601dcaf8aa8f04b257468a3ada3b396a36d9 Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.ledkov@surgut.co.uk>
Date: Sat, 14 Feb 2026 03:08:31 +0000
Subject: [PATCH] uv-patch-dependencies

Patch transitive dependency
---
 Cargo.toml | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/Cargo.toml b/Cargo.toml
index 25c58ea06..69cbd9962 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -154,7 +154,8 @@ ref-cast = { version = "1.0.24" }
 reflink-copy = { version = "0.1.19" }
 regex = { version = "1.10.6" }
 regex-automata = { version = "0.4.8", default-features = false, features = ["dfa-build", "dfa-search", "perf", "std", "syntax"] }
-reqsign = { version = "0.18.1", features = ["aws", "google", "default-context"], default-features = false }
+reqsign = { version = "0.19.1", features = ["aws", "google", "default-context"], default-features = false }
+reqsign-google = { version = "2.0.3" }
 reqwest = { version = "0.12.22", default-features = false, features = ["json", "gzip", "deflate", "zstd", "stream", "system-proxy", "rustls-tls", "rustls-tls-native-roots", "socks", "multipart", "http2", "blocking"] }
 reqwest-middleware = { version = "0.4.2", package = "astral-reqwest-middleware", features = ["multipart"] }
 reqwest-retry = { version = "0.8.0", package = "astral-reqwest-retry" }
@@ -341,3 +342,7 @@ codegen-units = 1
 # The profile that 'cargo dist' will build with.
 [profile.dist]
 inherits = "release"
+
+[patch.crates-io]
+reqsign = { path = "../opendal-reqsign/reqsign/" }
+reqsign-google = { path = "../opendal-reqsign/services/google/" }
-- 
2.51.0

