diff --git a/protocol-http1.gemspec b/protocol-http1.gemspec
index 2d27898..c5c6df1 100644
--- a/protocol-http1.gemspec
+++ b/protocol-http1.gemspec
@@ -11,8 +11,6 @@ Gem::Specification.new do |spec|
 	spec.license = "MIT"
 	
 	spec.cert_chain  = ["release.cert"]
-	spec.signing_key = File.expand_path("~/.gem/release.pem")
-	
 	spec.homepage = "https://github.com/socketry/protocol-http1"
 	
 	spec.metadata = {
