diff --git a/build.gradle b/build.gradle
index 6b2e9464..795fa607 100644
--- a/build.gradle
+++ b/build.gradle
@@ -49,6 +49,14 @@ configurations.all {
         force("com.fasterxml.jackson.core:jackson-databind:" + jacksonVersion)
         force("com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:" + jacksonVersion)
         force("com.fasterxml.jackson.module:jackson-module-scala_" + kafkaScalaVersion + ":" + jacksonVersion)
+        force("commons-beanutils:commons-beanutils:" + beansVersion)
+        force("io.vertx:vertx-core:" + vertxVersion)
+        force("org.apache.commons:commons-compress:" + commonsCompressVersion)
+        force("io.netty:netty-handler:" + nettyVersion)
+        force("org.eclipse.jetty:jetty-http:" + jettyHttpVersion)
+        force("net.minidev:json-smart:2.5.2")
+        force("org.apache.commons:commons-lang3:" + lang3Version)
+        force("com.nimbusds:nimbus-jose-jwt:" + nimbusJoseJwtVersion)
     }
 }

@@ -93,7 +101,7 @@ dependencies {
     implementation("io.micronaut:micronaut-http-server-netty")
     implementation("io.micronaut:micronaut-jackson-databind")
     implementation("io.swagger.core.v3:swagger-annotations")
-    runtimeOnly("ch.qos.logback:logback-classic")
+    runtimeOnly("ch.qos.logback:logback-classic:" + logbackVersion)
     runtimeOnly("org.yaml:snakeyaml")
     implementation("io.micronaut.security:micronaut-security-annotations")
     implementation("io.micronaut.security:micronaut-security-jwt")
@@ -174,6 +182,8 @@ dependencies {
     implementation group: 'io.projectreactor', name: 'reactor-core', version: '3.7.6'

     implementation 'io.jsonwebtoken:jjwt-impl:0.12.6'
+
+    implementation 'io.netty:netty-common:4.1.125.Final'
 }

 /**********************************************************************************************************************\
@@ -295,4 +305,4 @@ shadowJar {

 processResources.dependsOn ":client:installFrontend"
 processResources.dependsOn ":client:assembleFrontend"
-processResources.dependsOn ":client:copyClientResources"
+processResources.dependsOn ":client:copyClientResources"
diff --git a/gradle.properties b/gradle.properties
index ae531b3b..707a3a72 100644
--- a/gradle.properties
+++ b/gradle.properties
@@ -1,6 +1,14 @@
-micronautVersion=4.3.8
+micronautVersion=4.9.3
 confluentVersion=7.4.4
 kafkaVersion=3.6.2
 kafkaScalaVersion=2.13
 lombokVersion=1.18.32
-jacksonVersion=2.17.1
\ No newline at end of file
+jacksonVersion=2.17.1
+lang3Version=3.18.0
+nimbusJoseJwtVersion=10.0.2
+logbackVersion=1.5.19
+commonsCompressVersion=1.26.0
+vertxVersion=4.4.8
+nettyVersion=4.1.125.Final
+jettyHttpVersion=12.0.12
+beansVersion=1.11.0
\ No newline at end of file
