diff --git a/build.gradle b/build.gradle
index 6b2e9464..8fb95475 100644
--- a/build.gradle
+++ b/build.gradle
@@ -39,6 +39,8 @@ configurations.all {
     exclude group: 'org.slf4j', module: 'slf4j-reload4j'
 
     resolutionStrategy {
+        force("org.apache.commons:commons-lang3:" + lang3Version)
+        force("com.nimbusds:nimbus-jose-jwt:" + nimbusJoseJwtVersion)
         force("org.apache.kafka:kafka-clients:" + kafkaVersion)
         force("org.apache.kafka:kafka-streams:" + kafkaVersion)
         force("org.apache.kafka:kafka-server-common:" + kafkaVersion)
@@ -295,4 +297,4 @@ shadowJar {
 
 processResources.dependsOn ":client:installFrontend"
 processResources.dependsOn ":client:assembleFrontend"
-processResources.dependsOn ":client:copyClientResources"
+processResources.dependsOn ":client:copyClientResources"
\ No newline at end of file
diff --git a/gradle.properties b/gradle.properties
index ae531b3b..2a18ffb5 100644
--- a/gradle.properties
+++ b/gradle.properties
@@ -1,3 +1,5 @@
+lang3Version=3.18.0
+nimbusJoseJwtVersion=10.0.2
 micronautVersion=4.3.8
 confluentVersion=7.4.4
 kafkaVersion=3.6.2
