diff --git a/build.gradle b/build.gradle
index f8bb5501..952ef65a 100644
--- a/build.gradle
+++ b/build.gradle
@@ -59,6 +59,7 @@ configurations.all {
         force("com.nimbusds:nimbus-jose-jwt:" + nimbusJoseJwtVersion)
         force("io.netty:netty-codec-http:" + nettyVersion)
         force("org.bitbucket.b_c:jose4j:" + jose4jVersion)
+        force("io.vertx:vertx-core:" + vertxcoreVersion)
     }
 }
 
@@ -307,4 +308,4 @@ shadowJar {
 
 processResources.dependsOn ":client:installFrontend"
 processResources.dependsOn ":client:assembleFrontend"
-processResources.dependsOn ":client:copyClientResources"
+processResources.dependsOn ":client:copyClientResources"
\ No newline at end of file
diff --git a/gradle.properties b/gradle.properties
index 4bd55d28..b8de517d 100644
--- a/gradle.properties
+++ b/gradle.properties
@@ -13,4 +13,5 @@ nettyVersion=4.1.125.Final
 jettyHttpVersion=12.0.12
 beansVersion=1.11.0
 nettyVersion=4.2.8.Final
-jose4jVersion=0.9.6
\ No newline at end of file
+jose4jVersion=0.9.6
+vertxcoreVersion=4.5.24
\ No newline at end of file
