From 4facdf783906f5bb2daa4196f4f8cd8306f657d3 Mon Sep 17 00:00:00 2001
From: Ankush Pathak <ankush.pathak@chainguard.dev>
Date: Fri, 26 Dec 2025 15:15:45 +0000
Subject: [PATCH] Bump jose4j to 0.9.6 to remediate GHSA-3677-xxcr-wjqv

Signed-off-by: Ankush Pathak <ankush.pathak@chainguard.dev>
---
 build.gradle      | 1 +
 gradle.properties | 3 ++-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/build.gradle b/build.gradle
index 4d7280ee..f8bb5501 100644
--- a/build.gradle
+++ b/build.gradle
@@ -58,6 +58,7 @@ configurations.all {
         force("org.apache.commons:commons-lang3:" + lang3Version)
         force("com.nimbusds:nimbus-jose-jwt:" + nimbusJoseJwtVersion)
         force("io.netty:netty-codec-http:" + nettyVersion)
+        force("org.bitbucket.b_c:jose4j:" + jose4jVersion)
     }
 }
 
diff --git a/gradle.properties b/gradle.properties
index d3632ae4..4bd55d28 100644
--- a/gradle.properties
+++ b/gradle.properties
@@ -12,4 +12,5 @@ vertxVersion=4.4.8
 nettyVersion=4.1.125.Final
 jettyHttpVersion=12.0.12
 beansVersion=1.11.0
-nettyVersion=4.2.8.Final
\ No newline at end of file
+nettyVersion=4.2.8.Final
+jose4jVersion=0.9.6
\ No newline at end of file
-- 
2.52.0

