---
 dgraph/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dgraph/Makefile b/dgraph/Makefile
index aed940881..820a67608 100644
--- a/dgraph/Makefile
+++ b/dgraph/Makefile
@@ -63,7 +63,7 @@ ifneq ($(strip $(BUILD_RACE)),)
 endif
 
 # jemalloc stuff
-HAS_JEMALLOC = $(shell test -f /usr/local/lib/libjemalloc.a && echo "jemalloc")
+HAS_JEMALLOC = $(shell test -f /usr/lib/libjemalloc.a && echo "jemalloc")
 JEMALLOC_URL = "https://github.com/jemalloc/jemalloc/releases/download/5.3.0/jemalloc-5.3.0.tar.bz2"
 
 # go install variables
-- 
2.39.3 (Apple Git-146)

