diff --git a/dgraph/Makefile b/dgraph/Makefile
index f7886f8ee..bd197457a 100644
--- a/dgraph/Makefile
+++ b/dgraph/Makefile
@@ -63,7 +63,7 @@ ifneq ($(strip $(BUILD_RACE)),)
 endif
 
 # jemalloc stuff
-HAS_JEMALLOC = $(shell test -f /usr/local/lib/libjemalloc.a && echo "jemalloc")
+HAS_JEMALLOC = $(shell test -f /usr/lib/libjemalloc.a && echo "jemalloc")
 JEMALLOC_URL = "https://github.com/jemalloc/jemalloc/releases/download/5.2.1/jemalloc-5.2.1.tar.bz2"
 
 # go install variables
