Author: Sergio Durigan Junior <sergiodj@chainguard.dev>

diff --git a/src/lib/OpenEXRCore/internal_thread.h b/src/lib/OpenEXRCore/internal_thread.h
index 6c3c1032..08fbc2db 100644
--- a/src/lib/OpenEXRCore/internal_thread.h
+++ b/src/lib/OpenEXRCore/internal_thread.h
@@ -29,13 +29,15 @@ call_once (once_flag* flag, void (*func) (void))
 }
 #    else
 #        include <pthread.h>
-#        define ONCE_FLAG_INIT PTHREAD_ONCE_INIT
+#        ifndef ONCE_FLAG_INIT
+#            define ONCE_FLAG_INIT PTHREAD_ONCE_INIT
 typedef pthread_once_t once_flag;
 static inline void
 call_once (once_flag* flag, void (*func) (void))
 {
     (void) pthread_once (flag, func);
 }
+#        endif
 #    endif
 #else
 #    define ONCE_FLAG_INIT 0
