Author: Sergio Durigan Junior <sergiodj@chainguard.dev>

diff --git a/filesize.gemspec b/filesize.gemspec
index 34bcae4..b75104c 100644
--- a/filesize.gemspec
+++ b/filesize.gemspec
@@ -8,7 +8,6 @@ Gem::Specification.new do |s|
   s.summary = s.description
 
   s.email = "dominikh@fork-bomb.org"
-  s.has_rdoc = "yard"
   s.files = Dir.glob("{lib,spec}/**/*.rb") + ["LICENSE", "README.md"]
   s.homepage = "https://github.com/dominikh/filesize"
   s.required_ruby_version = ">= 1.8.6"
