diff -ru openjdk.orig/jdk/src/solaris/native/java/net/NetworkInterface.c openjdk/jdk/src/solaris/native/java/net/NetworkInterface.c
--- openjdk.orig/jdk/src/solaris/native/java/net/NetworkInterface.c	2014-06-12 20:14:44.000000000 +0000
+++ openjdk/jdk/src/solaris/native/java/net/NetworkInterface.c	2014-07-15 07:19:52.186682096 +0000
@@ -47,7 +47,6 @@
 
 #ifdef __linux__
 #include <sys/ioctl.h>
-#include <bits/ioctls.h>
 #include <sys/utsname.h>
 #include <stdio.h>
 #endif
diff -ru openjdk.orig/jdk/src/solaris/native/java/net/PlainDatagramSocketImpl.c openjdk/jdk/src/solaris/native/java/net/PlainDatagramSocketImpl.c
--- openjdk.orig/jdk/src/solaris/native/java/net/PlainDatagramSocketImpl.c	2014-06-12 20:14:44.000000000 +0000
+++ openjdk/jdk/src/solaris/native/java/net/PlainDatagramSocketImpl.c	2014-07-15 07:19:52.190015524 +0000
@@ -35,7 +35,6 @@
 #endif
 #ifdef __linux__
 #include <unistd.h>
-#include <sys/sysctl.h>
 #include <sys/utsname.h>
 #include <netinet/ip.h>
 
diff -ru openjdk.orig/jdk/src/solaris/native/java/net/PlainSocketImpl.c openjdk/jdk/src/solaris/native/java/net/PlainSocketImpl.c
--- openjdk.orig/jdk/src/solaris/native/java/net/PlainSocketImpl.c	2014-06-12 20:14:44.000000000 +0000
+++ openjdk/jdk/src/solaris/native/java/net/PlainSocketImpl.c	2014-07-15 07:19:52.190015524 +0000
@@ -43,7 +43,6 @@
 #endif
 #ifdef __linux__
 #include <unistd.h>
-#include <sys/sysctl.h>
 #endif
 
 #include "jvm.h"
