--- openjdk.orig/hotspot/make/linux/makefiles/saproc.make	2023-04-26 17:21:11
+++ openjdk/hotspot/make/linux/makefiles/saproc.make	2023-04-26 17:20:39
@@ -66,12 +66,6 @@
 # if $(AGENT_DIR) does not exist, we don't build SA
 # also, we don't build SA on Itanium or zero.
 
-ifneq ($(wildcard $(AGENT_DIR)),)
-ifneq ($(filter-out ia64 zero,$(SRCARCH)),)
-  BUILDLIBSAPROC = $(LIBSAPROC)
-endif
-endif
-
 ifneq ($(ALT_SASRCDIR),)
 ALT_SAINCDIR=-I$(ALT_SASRCDIR)
 else
--- openjdk.orig/hotspot/make/linux/makefiles/sa.make	2023-04-26 17:52:32
+++ openjdk/hotspot/make/linux/makefiles/sa.make	2023-04-26 17:52:39
@@ -61,10 +61,6 @@
 # also, we don't build SA on Itanium or zero.
 
 all: 
-	if [ -d $(AGENT_DIR) -a "$(SRCARCH)" != "ia64" \
-             -a "$(SRCARCH)" != "zero" ] ; then \
-	   $(MAKE) -f sa.make $(GENERATED)/sa-jdi.jar; \
-	fi
 
 $(GENERATED)/sa-jdi.jar:: $(AGENT_FILES)
 	$(QUIETLY) echo "Making $@"
--- openjdk.orig/hotspot/make/linux/makefiles/defs.make	2023-04-26 17:55:00
+++ openjdk/hotspot/make/linux/makefiles/defs.make	2023-04-26 17:56:28
@@ -327,27 +327,9 @@
 
 # Serviceability Binaries
 # No SA Support for PPC, IA64, ARM or zero
-ADD_SA_BINARIES/x86   = $(EXPORT_JRE_LIB_ARCH_DIR)/libsaproc.$(LIBRARY_SUFFIX) \
-                        $(EXPORT_LIB_DIR)/sa-jdi.jar 
-ADD_SA_BINARIES/sparc = $(EXPORT_JRE_LIB_ARCH_DIR)/libsaproc.$(LIBRARY_SUFFIX) \
-                        $(EXPORT_LIB_DIR)/sa-jdi.jar 
-ADD_SA_BINARIES/aarch64 = $(EXPORT_JRE_LIB_ARCH_DIR)/libsaproc.$(LIBRARY_SUFFIX) \
-                        $(EXPORT_LIB_DIR)/sa-jdi.jar
-ifeq ($(ENABLE_FULL_DEBUG_SYMBOLS),1)
-  ifneq ($(ZERO_BUILD), true)
-    ifneq ($(STRIP_POLICY),no_strip)
-      ifeq ($(ZIP_DEBUGINFO_FILES),1)
-        ADD_SA_BINARIES/x86     += $(EXPORT_JRE_LIB_ARCH_DIR)/libsaproc.diz
-        ADD_SA_BINARIES/sparc   += $(EXPORT_JRE_LIB_ARCH_DIR)/libsaproc.diz
-        ADD_SA_BINARIES/aarch64 += $(EXPORT_JRE_LIB_ARCH_DIR)/libsaproc.diz
-      else
-        ADD_SA_BINARIES/x86     += $(EXPORT_JRE_LIB_ARCH_DIR)/libsaproc.debuginfo
-        ADD_SA_BINARIES/sparc   += $(EXPORT_JRE_LIB_ARCH_DIR)/libsaproc.debuginfo
-        ADD_SA_BINARIES/aarch64 += $(EXPORT_JRE_LIB_ARCH_DIR)/libsaproc.debuginfo
-      endif
-    endif
-  endif
-endif
+ADD_SA_BINARIES/x86   =
+ADD_SA_BINARIES/sparc =
+ADD_SA_BINARIES/aarch64 =
 ADD_SA_BINARIES/ppc   = 
 ADD_SA_BINARIES/ppc64 = 
 ADD_SA_BINARIES/ia64  = 
