diff --git a/bazel/toolchains/BUILD b/bazel/toolchains/BUILD
index b806112b6..024d8882e 100644
--- a/bazel/toolchains/BUILD
+++ b/bazel/toolchains/BUILD
@@ -48,6 +48,11 @@ cc_toolchain_config(
     coverage_link_flags = ["--coverage"],
     cpu = "aarch64",
     cxx_builtin_include_directories = [
+        # These aren't how we configure where to look, but which files
+        # Bazel allows us to use in the build. So we don't have to be
+        # super exact and specify the version in the path.
+        "/usr/lib64/gcc/aarch64-unknown-linux-gnu",
+        "/usr/lib/clang",
         "/usr/lib/llvm-15",
         "/usr/aarch64-linux-gnu/include",
         "/usr/include",
@@ -76,18 +81,18 @@ cc_toolchain_config(
     target_libc = "glibc",
     target_system_name = "aarch64-linux-gnu",
     tool_paths = {
-        "ar": "/usr/bin/llvm-ar-15",
-        "compat-ld": "/usr/bin/lld-15",
-        "ld": "/usr/bin/lld-15",
-        "gold": "/usr/bin/lld-15",
+        "ar": "/usr/bin/llvm-ar",
+        "compat-ld": "/usr/bin/lld",
+        "ld": "/usr/bin/lld",
+        "gold": "/usr/bin/lld",
         "cpp": "/usr/bin/clang-cpp-15",
         "gcc": "/usr/bin/clang-15",
-        "dwp": "/usr/bin/llvm-dwp-15",
-        "gcov": "/usr/bin/llvmcov-15",
-        "nm": "/usr/bin/llvm-nm-15",
-        "objcopy": "/usr/bin/llvm-objcopy-15",
-        "objdump": "/usr/bin/llvm-objdump-15",
-        "strip": "/usr/bin/llvm-strip-15",
+        "dwp": "/usr/bin/llvm-dwp",
+        "gcov": "/usr/bin/llvmcov",
+        "nm": "/usr/bin/llvm-nm",
+        "objcopy": "/usr/bin/llvm-objcopy",
+        "objdump": "/usr/bin/llvm-objdump",
+        "strip": "/usr/bin/llvm-strip",
     },
     toolchain_identifier = "linux_aarch64",
     unfiltered_compile_flags = [
@@ -146,6 +151,11 @@ cc_toolchain_config(
     coverage_link_flags = ["--coverage"],
     cpu = "k8",
     cxx_builtin_include_directories = [
+        # These aren't how we configure where to look, but which files
+        # Bazel allows us to use in the build. So we don't have to be
+        # super exact and specify the version in the path.
+        "/usr/lib64/gcc/x86_64-pc-linux-gnu",
+        "/usr/lib/clang",
         "/usr/lib/llvm-15",
         "/usr/x86_64-linux-gnu/include",
         "/usr/include",
@@ -174,18 +184,18 @@ cc_toolchain_config(
     target_libc = "unknown",
     target_system_name = "unknown",
     tool_paths = {
-        "ar": "/usr/bin/llvm-ar-15",
-        "compat-ld": "/usr/bin/lld-15",
-        "ld": "/usr/bin/lld-15",
-        "gold": "/usr/bin/lld-15",
-        "cpp": "/usr/bin/clang-cpp-15",
+        "ar": "/usr/bin/llvm-ar",
+        "compat-ld": "/usr/bin/lld",
+        "ld": "/usr/bin/lld",
+        "gold": "/usr/bin/lld",
+        "cpp": "/usr/bin/clang-cpp",
         "gcc": "/usr/bin/clang-15",
-        "dwp": "/usr/bin/llvm-dwp-15",
-        "gcov": "/usr/bin/llvmcov-15",
-        "nm": "/usr/bin/llvm-nm-15",
-        "objcopy": "/usr/bin/llvm-objcopy-15",
-        "objdump": "/usr/bin/llvm-objdump-15",
-        "strip": "/usr/bin/llvm-strip-15",
+        "dwp": "/usr/bin/llvm-dwp",
+        "gcov": "/usr/bin/llvmcov",
+        "nm": "/usr/bin/llvm-nm",
+        "objcopy": "/usr/bin/llvm-objcopy",
+        "objdump": "/usr/bin/llvm-objdump",
+        "strip": "/usr/bin/llvm-strip",
     },
     toolchain_identifier = "linux_x86_64",
     unfiltered_compile_flags = [
