diff -ru libvpx-1.6.1.orig/build/make/configure.sh libvpx-1.6.1/build/make/configure.sh
--- libvpx-1.6.1.orig/build/make/configure.sh	2017-01-12 22:27:27.000000000 +0200
+++ libvpx-1.6.1/build/make/configure.sh	2018-11-13 13:49:37.998514809 +0200
@@ -680,7 +680,7 @@
       aarch64*)
         tgt_isa=arm64
         ;;
-      armv7*-hardfloat* | armv7*-gnueabihf | arm-*-gnueabihf)
+      armv7*-hardfloat* | armv7*-*eabihf | arm-*-*eabihf)
         tgt_isa=armv7
         float_abi=hard
         ;;
