diff -urN openssh-10.0p1.orig/entropy.c openssh-10.0p1/entropy.c
--- openssh-10.0p1.orig/entropy.c	2025-04-09 09:02:43.000000000 +0200
+++ openssh-10.0p1/entropy.c	2025-09-30 11:53:26.013749303 +0200
@@ -66,12 +66,6 @@
 	/* Initialise libcrypto */
 	ssh_libcrypto_init();
 
-	if (!ssh_compatible_openssl(OPENSSL_VERSION_NUMBER,
-	    OpenSSL_version_num()))
-		fatal("OpenSSL version mismatch. Built against %lx, you "
-		    "have %lx", (u_long)OPENSSL_VERSION_NUMBER,
-		    OpenSSL_version_num());
-
 #ifndef OPENSSL_PRNG_ONLY
 	if (RAND_status() == 1)
 		debug3("RNG is ready, skipping seeding");
