diff --git a/vendor/sigs.k8s.io/custom-metrics-apiserver/pkg/cmd/builder.go b/vendor/sigs.k8s.io/custom-metrics-apiserver/pkg/cmd/builder.go
index 7b4fb40a..3b871055 100644
--- a/vendor/sigs.k8s.io/custom-metrics-apiserver/pkg/cmd/builder.go
+++ b/vendor/sigs.k8s.io/custom-metrics-apiserver/pkg/cmd/builder.go
@@ -264,8 +264,8 @@ func (b *AdapterBase) defaultOpenAPIConfig() *openapicommon.Config {
 	return b.openAPIConfig(genericapiserver.DefaultOpenAPIConfig)
 }
 
-func (b *AdapterBase) defaultOpenAPIV3Config() *openapicommon.Config {
-	return b.openAPIConfig(genericapiserver.DefaultOpenAPIV3Config)
+func (b *AdapterBase) defaultOpenAPIV3Config() *openapicommon.OpenAPIV3Config {
+	return b.defaultOpenAPIV3Config()
 }
 
 // Config fetches the configuration used to ultimately create the custom metrics adapter's
@@ -284,7 +284,7 @@ func (b *AdapterBase) Config() (*apiserver.Config, error) {
 			b.OpenAPIConfig = b.defaultOpenAPIConfig()
 		}
 		b.CustomMetricsAdapterServerOptions.OpenAPIConfig = b.OpenAPIConfig
-		if b.OpenAPIV3Config == nil && utilfeature.DefaultFeatureGate.Enabled(features.OpenAPIV3) {
+		if b.OpenAPIV3Config == nil && utilfeature.DefaultFeatureGate.Enabled(features.OpenAPIEnums) {
 			b.OpenAPIV3Config = b.defaultOpenAPIV3Config()
 		}
 
diff --git a/vendor/sigs.k8s.io/custom-metrics-apiserver/pkg/cmd/options/options.go b/vendor/sigs.k8s.io/custom-metrics-apiserver/pkg/cmd/options/options.go
index 16f943e9..473e8973 100644
--- a/vendor/sigs.k8s.io/custom-metrics-apiserver/pkg/cmd/options/options.go
+++ b/vendor/sigs.k8s.io/custom-metrics-apiserver/pkg/cmd/options/options.go
@@ -25,6 +25,8 @@ import (
 
 	genericapiserver "k8s.io/apiserver/pkg/server"
 	genericoptions "k8s.io/apiserver/pkg/server/options"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
 	openapicommon "k8s.io/kube-openapi/pkg/common"
 )
 
@@ -40,7 +42,7 @@ type CustomMetricsAdapterServerOptions struct {
 	Features       *genericoptions.FeatureOptions
 
 	OpenAPIConfig   *openapicommon.Config
-	OpenAPIV3Config *openapicommon.Config
+	OpenAPIV3Config *openapicommon.OpenAPIV3Config
 	EnableMetrics   bool
 }
 
@@ -99,7 +101,14 @@ func (o *CustomMetricsAdapterServerOptions) ApplyTo(serverConfig *genericapiserv
 	if err := o.Audit.ApplyTo(serverConfig); err != nil {
 		return err
 	}
-	if err := o.Features.ApplyTo(serverConfig); err != nil {
+	kubernetesClient, err := kubernetes.NewForConfig(serverConfig.LoopbackClientConfig)
+	if err != nil {
+		return err
+	}
+
+	sharedInformers := informers.NewSharedInformerFactory(kubernetesClient, serverConfig.LoopbackClientConfig.Timeout)
+
+	if err := o.Features.ApplyTo(serverConfig, kubernetesClient, sharedInformers); err != nil {
 		return err
 	}
 
