--- a/hotspot/src/jdk.aot/unix/native/libjelfshim/jdk_tools_jaotc_jnilibelf_JNILibELFAPI.c
+++ b/hotspot/src/jdk.aot/unix/native/libjelfshim/jdk_tools_jaotc_jnilibelf_JNILibELFAPI.c
@@ -262,7 +262,7 @@
    jlong addr = getNativeAddress(env, ptrObj);
    if (addr != -1) {
        // Call libelf function
-       if ((retPtr = gelf_newehdr((Elf*) addr, elfClass)) == 0) {
+       if ((retPtr = (unsigned long int)gelf_newehdr((Elf*) addr, elfClass)) == 0) {
            errx(EX_SOFTWARE, "gelf_newehdr() failed: %s.", elf_errmsg(-1));
        }
    } else {
@@ -277,7 +277,7 @@
    jlong addr = getNativeAddress(env, ptrObj);
    if (addr != -1) {
        // Call libelf function
-       if ((retPtr = gelf_newphdr((Elf*) addr, phnum)) == 0) {
+       if ((retPtr = (unsigned long int)gelf_newphdr((Elf*) addr, phnum)) == 0) {
            errx(EX_SOFTWARE, "gelf_newphdr() failed: %s.", elf_errmsg(-1));
        }
    } else {
