diff --git a/configure.ac b/configure.ac
index 3a795267..2f1b6683 100644
--- a/configure.ac
+++ b/configure.ac
@@ -137,10 +137,10 @@ AC_SUBST(MOD_DIR)
 
 dnl boost
 AX_BOOST_BASE([[1.58.0]])
-AX_BOOST_PYTHON
-if test "$BOOST_PYTHON_LIB" = ""; then
-   AC_MSG_ERROR([No usable boost::python found])
-fi
+# AX_BOOST_PYTHON
+# if test "$BOOST_PYTHON_LIB" = ""; then
+#    AC_MSG_ERROR([No usable boost::python found])
+# fi
 AX_BOOST_IOSTREAMS
 if test "$BOOST_IOSTREAMS_LIB" = ""; then
    AC_MSG_ERROR([No usable boost::iostreams found])
@@ -167,7 +167,7 @@ if test "$BOOST_THREAD_LIB" = ""; then
    AC_MSG_ERROR([No usable boost::thread found])
 fi
 
-[BOOST_LIBS="${BOOST_LDFLAGS} ${BOOST_IOSTREAMS_LIB} -l${BOOST_PYTHON_LIB} ${BOOST_REGEX_LIB} ${BOOST_CONTEXT_LIB} ${BOOST_COROUTINE_LIB} ${BOOST_THREAD_LIB}"]
+[BOOST_LIBS="${BOOST_LDFLAGS} ${BOOST_IOSTREAMS_LIB} ${BOOST_REGEX_LIB} ${BOOST_CONTEXT_LIB} ${BOOST_COROUTINE_LIB} ${BOOST_THREAD_LIB}"]
 
 dnl GNU MP library (needed by CGAL)
 AC_CHECK_LIB(gmp, __gmpz_init, ,
@@ -375,7 +375,7 @@ AC_SUBST(MOD_LDFLAGS)
 AC_DEFINE([BOOST_BIND_GLOBAL_PLACEHOLDERS], [1], [Suppress Boost warning])
 
 AX_CREATE_PKGCONFIG_INFO([graph-tool-py${PYTHON_VERSION}.pc], [],
-                         [${PYTHON_LIBS} -l${BOOST_PYTHON_LIB}],
+                         [${PYTHON_LIBS} ],
                          [graph-tool Python library],
                          [-I${MOD_DIR}/include -I${MOD_DIR}/include/boost-workaround -I${MOD_DIR}/include/pcg-cpp ${PYTHON_CPPFLAGS} ${BOOST_CPPFLAGS} ${SPARSEHASH_CFLAGS} ${NUMPY_CPPFLAGS}],
                          [])

