From 6eb781b3339b6c89fda40da950be53cb9ed95411 Mon Sep 17 00:00:00 2001
From: Batuhan Apaydin <batuhan.apaydin@chainguard.dev>
Date: Wed, 17 Jan 2024 17:41:57 +0300
Subject: [PATCH] test

Signed-off-by: Batuhan Apaydin <batuhan.apaydin@chainguard.dev>
---
 setup.cfg | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/setup.cfg b/setup.cfg
index 7c6af765c..c27b36111 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -66,11 +66,6 @@ console_scripts =
 SCons.Tool.docbook = *.*
 
 
-[options.data_files]
-. = scons.1
-    scons-time.1
-    sconsign.1
-
 [sdist]
 dist_dir=build/dist
 
-- 
2.39.3 (Apple Git-145)

