diff --git a/MODULE.bazel b/MODULE.bazel
index f55aeae..04ef3b7 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -110,11 +110,24 @@ python = use_extension("@rules_python//python/extensions:python.bzl", "python")
 python.toolchain(
     is_default = True,
     python_version = "3.10.19",
+    ignore_root_user_error = True,
+)
+python.toolchain(
+    python_version = "3.11.14",
+    ignore_root_user_error = True,
+)
+python.toolchain(
+    python_version = "3.12.12",
+    ignore_root_user_error = True,
+)
+python.toolchain(
+    python_version = "3.13.9",
+    ignore_root_user_error = True,
+)
+python.toolchain(
+    python_version = "3.14.0",
+    ignore_root_user_error = True,
 )
-python.toolchain(python_version = "3.11.14")
-python.toolchain(python_version = "3.12.12")
-python.toolchain(python_version = "3.13.9")
-python.toolchain(python_version = "3.14.0")
 use_repo(python, "pythons_hub")
 
 pip = use_extension("@rules_python//python/extensions:pip.bzl", "pip")
