diff --git a/MODULE.bazel b/MODULE.bazel
index 0228c3235b..611cd8215c 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -124,11 +124,24 @@ python = use_extension("@rules_python//python/extensions:python.bzl", "python")
 python.toolchain(
     is_default = True,
     python_version = "3.10",
+    ignore_root_user_error = True,
+)
+python.toolchain(
+    python_version = "3.11",
+    ignore_root_user_error = True,
+)
+python.toolchain(
+    python_version = "3.12",
+    ignore_root_user_error = True,
+)
+python.toolchain(
+    python_version = "3.13",
+    ignore_root_user_error = True,
+)
+python.toolchain(
+    python_version = "3.14",
+    ignore_root_user_error = True,
 )
-python.toolchain(python_version = "3.11")
-python.toolchain(python_version = "3.12")
-python.toolchain(python_version = "3.13")
-python.toolchain(python_version = "3.14")
 use_repo(python, "pythons_hub")
 
 pip = use_extension("@rules_python//python/extensions:pip.bzl", "pip")
