diff --git a/MODULE.bazel b/MODULE.bazel
index 43611ccd9e..cdc848c409 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -98,10 +98,20 @@
 python.toolchain(
     is_default = True,
     python_version = "3.8",
+    ignore_root_user_error = True,
 )
-python.toolchain(python_version = "3.9")
-python.toolchain(python_version = "3.10")
-python.toolchain(python_version = "3.11")
+python.toolchain(
+    python_version = "3.9",
+    ignore_root_user_error = True,
+)
+python.toolchain(
+    python_version = "3.10",
+    ignore_root_user_error = True,
+)
+python.toolchain(
+    python_version = "3.11",
+    ignore_root_user_error = True,
+)
 use_repo(python, "pythons_hub")

 pip = use_extension("@rules_python//python/extensions:pip.bzl", "pip")
