diff --git a/MODULE.bazel b/MODULE.bazel
index 9a98a3348b..4644abd07c 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -182,13 +182,13 @@ maven.install(
         "dev.failsafe:failsafe:3.3.2",
         "io.grpc:grpc-context:1.68.1",
         "io.lettuce:lettuce-core:6.5.0.RELEASE",
-        "io.netty:netty-buffer:4.1.115.Final",
-        "io.netty:netty-codec-http:4.1.115.Final",
-        "io.netty:netty-codec-http2:4.1.115.Final",
-        "io.netty:netty-common:4.1.115.Final",
-        "io.netty:netty-handler:4.1.115.Final",
-        "io.netty:netty-handler-proxy:4.1.115.Final",
-        "io.netty:netty-transport:4.1.115.Final",
+        "io.netty:netty-buffer:4.1.125.Final",
+        "io.netty:netty-codec-http:4.1.125.Final",
+        "io.netty:netty-codec-http2:4.1.125.Final",
+        "io.netty:netty-common:4.1.125.Final",
+        "io.netty:netty-handler:4.1.125.Final",
+        "io.netty:netty-handler-proxy:4.1.125.Final",
+        "io.netty:netty-transport:4.1.125.Final",
         "io.opentelemetry:opentelemetry-api:1.44.1",
         "io.opentelemetry:opentelemetry-context:1.44.1",
         "io.opentelemetry:opentelemetry-exporter-logging:1.44.1",
