diff --git a/docker-entrypoint.sh b/docker-entrypoint.sh
index 94d68b1..5f52bca 100755
--- a/docker-entrypoint.sh
+++ b/docker-entrypoint.sh
@@ -108,7 +108,7 @@ if [ -n "$KUBERNETES_SERVICE_HOST" ] && [ -z "$START_DELAY" ]; then
 fi
 [ -n "$START_DELAY" ] && sleep "$START_DELAY"
 
-RQLITED=/bin/rqlited
+RQLITED=/usr/bin/rqlited
 rqlited_commands="$RQLITED $node_id $http_addr $http_adv_addr $raft_addr $raft_adv_addr $extensions_path_flag"
 
 data_dir="${DATA_DIR:-/rqlite/file/data}"
