From e80d57e7756aa119ca161487cad255462ae559c8 Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.ledkov@chainguard.dev>
Date: Thu, 21 Aug 2025 12:46:43 +0100
Subject: [PATCH] Work without fips sha1

---
 src/apk.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/apk.c b/src/apk.c
index 3b4ce40..c9f9202 100644
--- a/src/apk.c
+++ b/src/apk.c
@@ -432,6 +432,9 @@ static void fini_openssl(void)
 
 static void init_openssl(void)
 {
+	// Preffer, but do not require FIPS algorithms, CRC usage of
+	// SHA1 is needed
+	EVP_set_default_properties(NULL, "?fips=yes");
 	atexit(fini_openssl);
 	OpenSSL_add_all_algorithms();
 #ifndef OPENSSL_NO_ENGINE
-- 
2.48.1

