From b5eec48c333fc8841018dfc43bfca87682f37ced Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.ledkov@chainguard.dev>
Date: Thu, 21 Aug 2025 12:46:43 +0100
Subject: [PATCH] Work without fips sha1

---
 src/apk.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/apk.c b/src/apk.c
index 164acea..6929a80 100644
--- a/src/apk.c
+++ b/src/apk.c
@@ -509,6 +509,10 @@ int main(int argc, char **argv)
 		if (applet->update_cache) dbopts.cache_max_age = 0;
 	}
 
+	// Prefer, but do not require FIPS algorithms, CRC usage of
+	// SHA1 is needed
+	EVP_set_default_properties(NULL, "?fips=yes");
+
 	setup_automatic_flags();
 	fetchTimeout = 60;
 	fetchRedirectMethod = fetch_redirect;
-- 
2.51.0

