diff --git a/src/crypto_openssl.c b/src/crypto_openssl.c
index b6079f8..37e81cf 100644
--- a/src/crypto_openssl.c
+++ b/src/crypto_openssl.c
@@ -194,7 +194,7 @@ void apk_crypto_init(void)
 {
 	// Least popular OpenSSLv3+ only API
 #ifdef EVP_MD_fetch
-	sha1 = EVP_MD_fetch(NULL, "sha1", NULL);
+	sha1 = EVP_MD_fetch(NULL, "sha1", "?fips=yes");
 	sha256 = EVP_MD_fetch(NULL, "sha256", NULL);
 	sha512 = EVP_MD_fetch(NULL, "sha512", NULL);
 #else
