diff --git a/src/apk.c b/src/apk.c
index bdad463..9f91f79 100644
--- a/src/apk.c
+++ b/src/apk.c
@@ -522,7 +522,7 @@ static int load_config(struct apk_ctx *ac)
 	int r;
 
 	is = apk_istream_from_file(AT_FDCWD, getenv("APK_CONFIG") ?: "/etc/apk/config");
-	if (is == ERR_PTR(-ENOENT)) is = apk_istream_from_file(AT_FDCWD, "/lib/apk/config");
+	if (is == ERR_PTR(-ENOENT)) is = apk_istream_from_file(AT_FDCWD, "/usr/lib/apk/config");
 	if (IS_ERR(is)) return PTR_ERR(is);
 
 	while (apk_istream_get_delim(is, newline, &line) == 0) {
diff --git a/src/app_convdb.c b/src/app_convdb.c
index 3369ae9..30d9ed5 100644
--- a/src/app_convdb.c
+++ b/src/app_convdb.c
@@ -209,18 +209,18 @@ static int conv_main(void *pctx, struct apk_ctx *ac, struct apk_string_array *ar
 	adb_wo_alloca(&ctx->pkgs, &schema_package_adb_array, &ctx->dbi);
 
 	apk_tar_parse(
-		apk_istream_from_file(root_fd, "lib/apk/db/scripts.tar"),
+		apk_istream_from_file(root_fd, "usr/lib/apk/db/scripts.tar"),
 		read_script, ctx, apk_ctx_get_id_cache(ac));
 
-	read_triggers(ctx, apk_istream_from_file(root_fd, "lib/apk/db/triggers"));
+	read_triggers(ctx, apk_istream_from_file(root_fd, "usr/lib/apk/db/triggers"));
 
-	convert_idb(ctx, apk_istream_from_file(root_fd, "lib/apk/db/installed"));
+	convert_idb(ctx, apk_istream_from_file(root_fd, "usr/lib/apk/db/installed"));
 
 	adb_wo_obj(&idb, ADBI_IDB_PACKAGES, &ctx->pkgs);
 	adb_w_rootobj(&idb);
 
 	r = adb_c_create(
-		//apk_ostream_to_file(db->root_fd, "lib/apk/db/installed.adb", 0644),
+		//apk_ostream_to_file(db->root_fd, "usr/lib/apk/db/installed.adb", 0644),
 		adb_compress(apk_ostream_to_file(AT_FDCWD, "installed.adb", 0644), &ac->compspec),
 		&ctx->dbi, apk_ctx_get_trust(ac));
 	if (r == 0) {
diff --git a/src/commit.c b/src/commit.c
index d4a844a..019523f 100644
--- a/src/commit.c
+++ b/src/commit.c
@@ -309,7 +309,7 @@ static int run_commit_hooks(struct apk_database *db, int type)
 	return apk_dir_foreach_config_file(db->root_fd,
 		run_commit_hook, &hook, apk_filename_is_hidden,
 		"etc/apk/commit_hooks.d",
-		"lib/apk/commit_hooks.d",
+		"usr/lib/apk/commit_hooks.d",
 		NULL);
 }
 
diff --git a/src/context.c b/src/context.c
index e04b4ed..b562211 100644
--- a/src/context.c
+++ b/src/context.c
@@ -156,7 +156,7 @@ struct apk_trust *apk_ctx_get_trust(struct apk_ctx *ac)
 				__apk_ctx_load_pubkey, &ac->trust,
 				apk_filename_is_hidden,
 				"etc/apk/keys",
-				"lib/apk/keys",
+				"usr/lib/apk/keys",
 				NULL);
 		} else {
 			apk_dir_foreach_file(ac->root_fd, ac->keys_dir,
diff --git a/src/database.c b/src/database.c
index 76c39ef..599ce11 100644
--- a/src/database.c
+++ b/src/database.c
@@ -44,7 +44,7 @@
 static const char * const apk_static_cache_dir = "var/cache/apk";
 static const char * const apk_world_file = "etc/apk/world";
 static const char * const apk_arch_file = "etc/apk/arch";
-static const char * const apk_lock_file = "lib/apk/db/lock";
+static const char * const apk_lock_file = "usr/lib/apk/db/lock";
 
 static struct apk_db_acl *apk_default_acl_dir, *apk_default_acl_file;
 
@@ -1917,8 +1917,8 @@ static int setup_cache(struct apk_database *db)
 const char *apk_db_layer_name(int layer)
 {
 	switch (layer) {
-	case APK_DB_LAYER_ROOT: return "lib/apk/db";
-	case APK_DB_LAYER_UVOL: return "lib/apk/db-uvol";
+	case APK_DB_LAYER_ROOT: return "usr/lib/apk/db";
+	case APK_DB_LAYER_UVOL: return "usr/lib/apk/db-uvol";
 	default:
 		assert(!"invalid layer");
 		return 0;
@@ -2130,7 +2130,7 @@ int apk_db_open(struct apk_database *db)
 				add_repos_from_file, db,
 				file_not_dot_list,
 				"etc/apk/repositories.d",
-				"lib/apk/repositories.d",
+				"usr/lib/apk/repositories.d",
 				NULL);
 		} else {
 			add_repos_from_file(db, AT_FDCWD, NULL, ac->repositories_file);
@@ -2273,7 +2273,7 @@ int apk_db_write_config(struct apk_database *db)
 		return 0;
 
 	if (db->ctx->open_flags & APK_OPENF_CREATE) {
-		apk_make_dirs(db->root_fd, "lib/apk/db", 0755, 0755);
+		apk_make_dirs(db->root_fd, "usr/lib/apk/db", 0755, 0755);
 		apk_make_dirs(db->root_fd, "etc/apk", 0755, 0755);
 	} else if (db->lock_fd == 0) {
 		apk_err(out, "Refusing to write db without write lock!");
diff --git a/src/package.c b/src/package.c
index 451cdfb..4f7aa20 100644
--- a/src/package.c
+++ b/src/package.c
@@ -719,7 +719,7 @@ int apk_ipkg_run_script(struct apk_installed_package *ipkg,
 {
 	// When memfd_create is not available store the script in /lib/apk/exec
 	// and hope it allows executing.
-	static const char script_exec_dir[] = "lib/apk/exec";
+	static const char script_exec_dir[] = "usr/lib/apk/exec";
 	struct apk_out *out = &db->ctx->out;
 	struct apk_package *pkg = ipkg->pkg;
 	const char *reason = "failed to execute: ";
