commit 65c066c3db62b75cf5659e216aa4fb838db58ec9 (HEAD)
Author: Scott Moser <smoser@brickies.net>
Date:   Thu Mar 27 16:53:48 2025 -0400

    try initdb in usrmerge-lib world

diff --git a/src/database.c b/src/database.c
index dc5e4fd..a37123c 100644
--- a/src/database.c
+++ b/src/database.c
@@ -1288,7 +1288,7 @@ static int add_protected_paths_from_file(void *ctx, int dirfd, const char *file)
 
 static int apk_db_create(struct apk_database *db)
 {
-	int fd;
+	int fd, r;
 
 	mkdirat(db->root_fd, "tmp", 01777);
 	mkdirat(db->root_fd, "dev", 0755);
@@ -1299,6 +1299,13 @@ static int apk_db_create(struct apk_database *db)
 	mknodat(db->root_fd, "dev/console", S_IFCHR | 0600, makedev(5, 1));
 	mkdirat(db->root_fd, "etc", 0755);
 	mkdirat(db->root_fd, "etc/apk", 0755);
+	r = symlinkat("usr/lib", db->root_fd, "lib");
+	if (r == -1) {
+	apk_error("failed to symlink lib -> usr/lib");
+		return -errno;
+	}
+	mkdirat(db->root_fd, "usr", 0755);
+	mkdirat(db->root_fd, "usr/lib", 0755);
 	mkdirat(db->root_fd, "lib", 0755);
 	mkdirat(db->root_fd, "lib/apk", 0755);
 	mkdirat(db->root_fd, "lib/apk/db", 0755);
