diff --git a/build.gradle b/build.gradle
index 4433815..1c98868 100644
--- a/build.gradle
+++ b/build.gradle
@@ -495,6 +495,10 @@ task buildRca() {
     }
 
     doLast {
+        exec {
+            workingDir("$rcaProjectDir")
+            commandLine 'git', 'apply', '../../analyzer-commons-io.patch'
+        }
         exec {
             workingDir("$rcaProjectDir")
             if (buildVersionQualifier == null || buildVersionQualifier == '' || buildVersionQualifier == 'null') {
