diff --git a/plugins/repository-azure/build.gradle b/plugins/repository-azure/build.gradle
index c2bf9e07e29..700d3c06adc 100644
--- a/plugins/repository-azure/build.gradle
+++ b/plugins/repository-azure/build.gradle
@@ -57,11 +57,11 @@ dependencies {
   api "io.netty:netty-transport-native-unix-common:${versions.netty}"
   implementation project(':modules:transport-netty4')
   api 'com.azure:azure-storage-blob:12.23.0'
-  api 'com.azure:azure-identity:1.11.4'
+  api 'com.azure:azure-identity:1.13.0'
   // Start of transitive dependencies for azure-identity
-  api 'com.microsoft.azure:msal4j-persistence-extension:1.2.0'
+  api 'com.microsoft.azure:msal4j-persistence-extension:1.3.0'
   api "net.java.dev.jna:jna-platform:${versions.jna}"
-  api 'com.microsoft.azure:msal4j:1.14.3'
+  api 'com.microsoft.azure:msal4j:1.15.1'
   api 'com.nimbusds:oauth2-oidc-sdk:11.9.1'
   api 'com.nimbusds:nimbus-jose-jwt:9.37.3'
   api 'com.nimbusds:content-type:2.3'
diff --git a/plugins/repository-azure/licenses/azure-identity-1.11.4.jar.sha1 b/plugins/repository-azure/licenses/azure-identity-1.11.4.jar.sha1
deleted file mode 100644
index c8d98ba9c8a..00000000000
--- a/plugins/repository-azure/licenses/azure-identity-1.11.4.jar.sha1
+++ /dev/null
@@ -1 +0,0 @@
-59b5ce48888f638b80d85ef5aa0e22a265d3dc89
\ No newline at end of file
diff --git a/plugins/repository-azure/licenses/msal4j-1.14.3.jar.sha1 b/plugins/repository-azure/licenses/msal4j-1.14.3.jar.sha1
deleted file mode 100644
index 2a6e42e3f2b..00000000000
--- a/plugins/repository-azure/licenses/msal4j-1.14.3.jar.sha1
+++ /dev/null
@@ -1 +0,0 @@
-117b28c41bd760f979ed1b6467c5ec491f0d4d60
\ No newline at end of file
diff --git a/plugins/repository-azure/licenses/msal4j-persistence-extension-1.2.0.jar.sha1 b/plugins/repository-azure/licenses/msal4j-persistence-extension-1.2.0.jar.sha1
deleted file mode 100644
index cfcf7548b76..00000000000
--- a/plugins/repository-azure/licenses/msal4j-persistence-extension-1.2.0.jar.sha1
+++ /dev/null
@@ -1 +0,0 @@
-1111a95878de8745ddc9de132df18ebd9ca7024d
\ No newline at end of file
