diff --git a/ml-algorithms/build.gradle b/ml-algorithms/build.gradle
index 251ea79a..6907a113 100644
--- a/ml-algorithms/build.gradle
+++ b/ml-algorithms/build.gradle
@@ -42,22 +42,22 @@ dependencies {
     testImplementation group: 'org.mockito', name: 'mockito-core', version: '5.7.0'
     implementation group: 'com.google.guava', name: 'guava', version: '32.1.2-jre'
     implementation group: 'com.google.code.gson', name: 'gson', version: '2.10.1'
-    implementation platform("ai.djl:bom:0.21.0")
-    implementation group: 'ai.djl.pytorch', name: 'pytorch-model-zoo', version: '0.21.0'
+    implementation platform("ai.djl:bom:0.28.0")
+    implementation group: 'ai.djl.pytorch', name: 'pytorch-model-zoo', version: '0.28.0'
     implementation group: 'ai.djl', name: 'api'
     implementation group: 'ai.djl.huggingface', name: 'tokenizers'
-    implementation("ai.djl.onnxruntime:onnxruntime-engine:0.21.0") {
+    implementation("ai.djl.onnxruntime:onnxruntime-engine:0.28.0") {
         exclude group: "com.microsoft.onnxruntime", module: "onnxruntime"
     }
     def os = DefaultNativePlatform.currentOperatingSystem
     //arm/macos doesn't support GPU
     if (os.macOsX || System.getProperty("os.arch") == "aarch64") {
         dependencies {
-            implementation "com.microsoft.onnxruntime:onnxruntime:1.14.0"
+            implementation "com.microsoft.onnxruntime:onnxruntime:1.17.1"
         }
     } else {
         dependencies {
-            implementation "com.microsoft.onnxruntime:onnxruntime_gpu:1.14.0"
+            implementation "com.microsoft.onnxruntime:onnxruntime_gpu:1.17.1"
         }
     }
 
@@ -77,6 +77,7 @@ lombok {
 configurations.all {
     resolutionStrategy.force 'com.google.protobuf:protobuf-java:3.21.9'
     resolutionStrategy.force 'org.apache.commons:commons-compress:1.26.0'
+    resolutionStrategy.force 'com.microsoft.onnxruntime:onnxruntime:1.17.1'
 }
 
 jacocoTestReport {
