diff --git a/ml-algorithms/build.gradle b/ml-algorithms/build.gradle
index 3561472..050ea67 100644
--- a/ml-algorithms/build.gradle
+++ b/ml-algorithms/build.gradle
@@ -62,7 +62,7 @@ dependencies {
     implementation 'software.amazon.awssdk:auth'
     implementation 'software.amazon.awssdk:apache-client'
     implementation 'com.amazonaws:aws-encryption-sdk-java:2.4.1'
-    implementation 'com.jayway.jsonpath:json-path:2.8.0'
+    implementation 'com.jayway.jsonpath:json-path:2.9.0'
     implementation group: 'org.json', name: 'json', version: '20231013'
 }

diff --git a/ml-algorithms/build.gradle b/ml-algorithms/build.gradle
index 35614721..74b0acbf 100644
--- a/ml-algorithms/build.gradle
+++ b/ml-algorithms/build.gradle
@@ -68,6 +68,7 @@ dependencies {
 
 configurations.all {
     resolutionStrategy.force 'com.google.protobuf:protobuf-java:3.21.9'
+    resolutionStrategy.force 'org.apache.commons:commons-compress:1.25.0'
 }
 
 jacocoTestReport {
diff --git a/plugin/build.gradle b/plugin/build.gradle
index af976e6f..3dc408a8 100644
--- a/plugin/build.gradle
+++ b/plugin/build.gradle
@@ -330,6 +330,7 @@ configurations.all {
     resolutionStrategy.force 'org.apache.httpcomponents:httpclient:4.5.14'
     resolutionStrategy.force 'commons-codec:commons-codec:1.15'
     resolutionStrategy.force 'org.slf4j:slf4j-api:1.7.36'
+    resolutionStrategy.force 'org.apache.commons:commons-compress:1.25.0'
 }
 
 apply plugin: 'com.netflix.nebula.ospackage'
