From: Daniel Watkins <daniel.watkins@chainguard.dev>
Subject: [PATCH] Remediate GHSA-j288-q9x7-2f5v

Bump commons-lang3 to v3.18.0
---
diff --git a/packaging/distributions-dependencies/build.gradle.kts b/packaging/distributions-dependencies/build.gradle.kts
index 4195cda5..52a57a6d 100644
--- a/packaging/distributions-dependencies/build.gradle.kts
+++ b/packaging/distributions-dependencies/build.gradle.kts
@@ -79,7 +79,7 @@ dependencies {
         api(libs.commonsCompress)       { version { strictly("1.26.1") } }
         api(libs.commonsHttpclient)     { version { strictly("4.5.14") } }
         api(libs.commonsIo)             { version { strictly("2.15.1") }}
-        api(libs.commonsLang)           { version { strictly("3.17.0") }}
+        api(libs.commonsLang)           { version { strictly("3.18.0") }}
         api(libs.commonsMath)           { version { strictly("3.6.1") }}
         api(libs.eclipseSisuPlexus)     { version { strictly("0.3.5"); because("transitive dependency of Maven modules to process POM metadata") }}
         api(libs.errorProneAnnotations) { version { strictly("2.36.0") } } // don't forget to upgrade errorprone in gradlebuild.code-quality.gradle.kts
