diff --git a/subprojects/core/src/integTest/groovy/org/gradle/internal/classpath/BuildScriptClasspathIntegrationSpec.groovy b/subprojects/core/src/integTest/groovy/org/gradle/internal/classpath/BuildScriptClasspathIntegrationSpec.groovy
index 62c041ec9d1..97b7089f565 100644
--- a/subprojects/core/src/integTest/groovy/org/gradle/internal/classpath/BuildScriptClasspathIntegrationSpec.groovy
+++ b/subprojects/core/src/integTest/groovy/org/gradle/internal/classpath/BuildScriptClasspathIntegrationSpec.groovy
@@ -311,7 +311,7 @@ class BuildScriptClasspathIntegrationSpec extends AbstractIntegrationSpec implem
             buildscript {
                 ${mavenCentralRepository()}
                 dependencies {
-                    classpath("org.bouncycastle:bcprov-jdk18on:1.77")
+                    classpath("org.bouncycastle:bcprov-jdk18on:1.78")
                 }
             }
 

